{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "694bbe16",
   "metadata": {},
   "source": [
    "## Install Software\n",
    "You may choose to create or load a virtual environment of your choice. The following command will install cwltool in your current working environment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "cc06adaf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: cwltool in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (3.0.20210124104916)\n",
      "Requirement already satisfied: setuptools in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from cwltool) (52.0.0.post20210125)\n",
      "Requirement already satisfied: bagit>=1.6.4 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from cwltool) (1.8.1)\n",
      "Requirement already satisfied: rdflib<5.1.0,>=4.2.2 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from cwltool) (5.0.0)\n",
      "Requirement already satisfied: shellescape<3.5,>=3.4.1 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from cwltool) (3.4.1)\n",
      "Requirement already satisfied: schema-salad<8,>=7.0.20210124093443 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from cwltool) (7.0.20210124093443)\n",
      "Requirement already satisfied: psutil in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from cwltool) (5.8.0)\n",
      "Requirement already satisfied: pydot>=1.4.1 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from cwltool) (1.4.1)\n",
      "Requirement already satisfied: argcomplete in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from cwltool) (1.12.2)\n",
      "Requirement already satisfied: ruamel.yaml<=0.16.5,>=0.12.4 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from cwltool) (0.16.5)\n",
      "Requirement already satisfied: typing-extensions in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from cwltool) (3.7.4.3)\n",
      "Requirement already satisfied: coloredlogs in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from cwltool) (14.0)\n",
      "Requirement already satisfied: requests>=2.6.1 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from cwltool) (2.25.1)\n",
      "Requirement already satisfied: mypy-extensions in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from cwltool) (0.4.3)\n",
      "Requirement already satisfied: prov==1.5.1 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from cwltool) (1.5.1)\n",
      "Requirement already satisfied: lxml in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from prov==1.5.1->cwltool) (4.6.2)\n",
      "Requirement already satisfied: six>=1.9.0 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from prov==1.5.1->cwltool) (1.16.0)\n",
      "Requirement already satisfied: networkx in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from prov==1.5.1->cwltool) (2.5)\n",
      "Requirement already satisfied: python-dateutil in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from prov==1.5.1->cwltool) (2.8.2)\n",
      "Requirement already satisfied: pyparsing>=2.1.4 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from pydot>=1.4.1->cwltool) (2.4.7)\n",
      "Requirement already satisfied: isodate in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from rdflib<5.1.0,>=4.2.2->cwltool) (0.6.0)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from requests>=2.6.1->cwltool) (2021.5.30)\n",
      "Requirement already satisfied: idna<3,>=2.5 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from requests>=2.6.1->cwltool) (2.10)\n",
      "Requirement already satisfied: chardet<5,>=3.0.2 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from requests>=2.6.1->cwltool) (4.0.0)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from requests>=2.6.1->cwltool) (1.26.6)\n",
      "Requirement already satisfied: rdflib-jsonld<0.6.0,>=0.3.0 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from schema-salad<8,>=7.0.20210124093443->cwltool) (0.5.0)\n",
      "Requirement already satisfied: CacheControl<0.12,>=0.11.7 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from schema-salad<8,>=7.0.20210124093443->cwltool) (0.11.7)\n",
      "Requirement already satisfied: mistune<0.9,>=0.8.1 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from schema-salad<8,>=7.0.20210124093443->cwltool) (0.8.4)\n",
      "Requirement already satisfied: lockfile>=0.9 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from schema-salad<8,>=7.0.20210124093443->cwltool) (0.12.2)\n",
      "Requirement already satisfied: humanfriendly>=7.1 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from coloredlogs->cwltool) (8.2)\n",
      "Requirement already satisfied: decorator>=4.3.0 in /Users/jwalker/opt/miniconda3/lib/python3.8/site-packages (from networkx->prov==1.5.1->cwltool) (5.0.9)\n"
     ]
    }
   ],
   "source": [
    "!pip install cwltool"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "504b622c",
   "metadata": {},
   "source": [
    "## Download Input Files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "474d9829",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-10-18 10:40:51--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes.fa\n",
      "Resolving storage.googleapis.com (storage.googleapis.com)... 2607:f8b0:4009:808::2010, 2607:f8b0:4009:809::2010, 2607:f8b0:4009:80a::2010, ...\n",
      "Connecting to storage.googleapis.com (storage.googleapis.com)|2607:f8b0:4009:808::2010|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 258297825 (246M) [application/octet-stream]\n",
      "Saving to: ‘hla_and_brca_genes.fa’\n",
      "\n",
      "hla_and_brca_genes. 100%[===================>] 246.33M  26.4MB/s    in 9.6s    \n",
      "\n",
      "2021-10-18 10:41:01 (25.7 MB/s) - ‘hla_and_brca_genes.fa’ saved [258297825/258297825]\n",
      "\n",
      "--2021-10-18 10:41:01--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes.fa.amb\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 129 [application/octet-stream]\n",
      "Saving to: ‘hla_and_brca_genes.fa.amb’\n",
      "\n",
      "hla_and_brca_genes. 100%[===================>]     129  --.-KB/s    in 0s      \n",
      "\n",
      "2021-10-18 10:41:01 (15.4 MB/s) - ‘hla_and_brca_genes.fa.amb’ saved [129/129]\n",
      "\n",
      "--2021-10-18 10:41:01--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes.fa.ann\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 79 [application/octet-stream]\n",
      "Saving to: ‘hla_and_brca_genes.fa.ann’\n",
      "\n",
      "hla_and_brca_genes. 100%[===================>]      79  --.-KB/s    in 0s      \n",
      "\n",
      "2021-10-18 10:41:01 (37.7 MB/s) - ‘hla_and_brca_genes.fa.ann’ saved [79/79]\n",
      "\n",
      "--2021-10-18 10:41:01--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes.fa.bwt\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 254063496 (242M) [application/octet-stream]\n",
      "Saving to: ‘hla_and_brca_genes.fa.bwt’\n",
      "\n",
      "hla_and_brca_genes. 100%[===================>] 242.29M  26.6MB/s    in 9.3s    \n",
      "\n",
      "2021-10-18 10:41:10 (26.0 MB/s) - ‘hla_and_brca_genes.fa.bwt’ saved [254063496/254063496]\n",
      "\n",
      "--2021-10-18 10:41:10--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes.fa.fai\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 54 [application/octet-stream]\n",
      "Saving to: ‘hla_and_brca_genes.fa.fai’\n",
      "\n",
      "hla_and_brca_genes. 100%[===================>]      54  --.-KB/s    in 0s      \n",
      "\n",
      "2021-10-18 10:41:11 (51.5 MB/s) - ‘hla_and_brca_genes.fa.fai’ saved [54/54]\n",
      "\n",
      "--2021-10-18 10:41:11--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes.fa.pac\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 63515857 (61M) [application/octet-stream]\n",
      "Saving to: ‘hla_and_brca_genes.fa.pac’\n",
      "\n",
      "hla_and_brca_genes. 100%[===================>]  60.57M  24.1MB/s    in 2.5s    \n",
      "\n",
      "2021-10-18 10:41:13 (24.1 MB/s) - ‘hla_and_brca_genes.fa.pac’ saved [63515857/63515857]\n",
      "\n",
      "--2021-10-18 10:41:13--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes.fa.sa\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 127031760 (121M) [application/octet-stream]\n",
      "Saving to: ‘hla_and_brca_genes.fa.sa’\n",
      "\n",
      "hla_and_brca_genes. 100%[===================>] 121.15M  26.4MB/s    in 4.7s    \n",
      "\n",
      "2021-10-18 10:41:18 (26.0 MB/s) - ‘hla_and_brca_genes.fa.sa’ saved [127031760/127031760]\n",
      "\n",
      "--2021-10-18 10:41:18--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes.dict\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 315 [application/octet-stream]\n",
      "Saving to: ‘hla_and_brca_genes.dict’\n",
      "\n",
      "hla_and_brca_genes. 100%[===================>]     315  --.-KB/s    in 0s      \n",
      "\n",
      "2021-10-18 10:41:18 (300 MB/s) - ‘hla_and_brca_genes.dict’ saved [315/315]\n",
      "\n",
      "--2021-10-18 10:41:18--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_bait.interval_list\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 2645 (2.6K) [application/octet-stream]\n",
      "Saving to: ‘hla_and_brca_genes_bait.interval_list’\n",
      "\n",
      "hla_and_brca_genes_ 100%[===================>]   2.58K  --.-KB/s    in 0s      \n",
      "\n",
      "2021-10-18 10:41:18 (2.46 GB/s) - ‘hla_and_brca_genes_bait.interval_list’ saved [2645/2645]\n",
      "\n",
      "--2021-10-18 10:41:18--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_target.interval_list\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 2124 (2.1K) [application/octet-stream]\n",
      "Saving to: ‘hla_and_brca_genes_target.interval_list’\n",
      "\n",
      "hla_and_brca_genes_ 100%[===================>]   2.07K  --.-KB/s    in 0s      \n",
      "\n",
      "2021-10-18 10:41:19 (1013 MB/s) - ‘hla_and_brca_genes_target.interval_list’ saved [2124/2124]\n",
      "\n",
      "--2021-10-18 10:41:19--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_known_indels.vcf.gz\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 22217 (22K) [application/x-gzip]\n",
      "Saving to: ‘hla_and_brca_genes_known_indels.vcf.gz’\n",
      "\n",
      "hla_and_brca_genes_ 100%[===================>]  21.70K  --.-KB/s    in 0.001s  \n",
      "\n",
      "2021-10-18 10:41:19 (23.5 MB/s) - ‘hla_and_brca_genes_known_indels.vcf.gz’ saved [22217/22217]\n",
      "\n",
      "--2021-10-18 10:41:19--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_known_indels.vcf.gz.tbi\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 409 [application/octet-stream]\n",
      "Saving to: ‘hla_and_brca_genes_known_indels.vcf.gz.tbi’\n",
      "\n",
      "hla_and_brca_genes_ 100%[===================>]     409  --.-KB/s    in 0s      \n",
      "\n",
      "2021-10-18 10:41:19 (390 MB/s) - ‘hla_and_brca_genes_known_indels.vcf.gz.tbi’ saved [409/409]\n",
      "\n",
      "--2021-10-18 10:41:19--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_mills.vcf.gz\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 9584 (9.4K) [application/x-gzip]\n",
      "Saving to: ‘hla_and_brca_genes_mills.vcf.gz’\n",
      "\n",
      "hla_and_brca_genes_ 100%[===================>]   9.36K  --.-KB/s    in 0s      \n",
      "\n",
      "2021-10-18 10:41:19 (179 MB/s) - ‘hla_and_brca_genes_mills.vcf.gz’ saved [9584/9584]\n",
      "\n",
      "--2021-10-18 10:41:19--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_mills.vcf.gz.tbi\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 398 [application/octet-stream]\n",
      "Saving to: ‘hla_and_brca_genes_mills.vcf.gz.tbi’\n",
      "\n",
      "hla_and_brca_genes_ 100%[===================>]     398  --.-KB/s    in 0s      \n",
      "\n",
      "2021-10-18 10:41:19 (380 MB/s) - ‘hla_and_brca_genes_mills.vcf.gz.tbi’ saved [398/398]\n",
      "\n",
      "--2021-10-18 10:41:19--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_dbsnp.vcf.gz\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 340475 (332K) [application/x-gzip]\n",
      "Saving to: ‘hla_and_brca_genes_dbsnp.vcf.gz’\n",
      "\n",
      "hla_and_brca_genes_ 100%[===================>] 332.50K  --.-KB/s    in 0.02s   \n",
      "\n",
      "2021-10-18 10:41:19 (13.4 MB/s) - ‘hla_and_brca_genes_dbsnp.vcf.gz’ saved [340475/340475]\n",
      "\n",
      "--2021-10-18 10:41:19--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_dbsnp.vcf.gz.tbi\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 455 [application/octet-stream]\n",
      "Saving to: ‘hla_and_brca_genes_dbsnp.vcf.gz.tbi’\n",
      "\n",
      "hla_and_brca_genes_ 100%[===================>]     455  --.-KB/s    in 0s      \n",
      "\n",
      "2021-10-18 10:41:19 (434 MB/s) - ‘hla_and_brca_genes_dbsnp.vcf.gz.tbi’ saved [455/455]\n",
      "\n",
      "--2021-10-18 10:41:19--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_omni.vcf.gz\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 2850 (2.8K) [application/x-gzip]\n",
      "Saving to: ‘hla_and_brca_genes_omni.vcf.gz’\n",
      "\n",
      "hla_and_brca_genes_ 100%[===================>]   2.78K  --.-KB/s    in 0s      \n",
      "\n",
      "2021-10-18 10:41:19 (1.33 GB/s) - ‘hla_and_brca_genes_omni.vcf.gz’ saved [2850/2850]\n",
      "\n",
      "--2021-10-18 10:41:19--  https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_omni.vcf.gz.tbi\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 277 [application/octet-stream]\n",
      "Saving to: ‘hla_and_brca_genes_omni.vcf.gz.tbi’\n",
      "\n",
      "hla_and_brca_genes_ 100%[===================>]     277  --.-KB/s    in 0s      \n",
      "\n",
      "2021-10-18 10:41:20 (264 MB/s) - ‘hla_and_brca_genes_omni.vcf.gz.tbi’ saved [277/277]\n",
      "\n",
      "--2021-10-18 10:41:20--  https://storage.googleapis.com/analysis-workflows-example-data/unaligned_subset_bams/normal/2895499331.bam\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 396318 (387K) [application/octet-stream]\n",
      "Saving to: ‘2895499331.bam’\n",
      "\n",
      "2895499331.bam      100%[===================>] 387.03K  --.-KB/s    in 0.05s   \n",
      "\n",
      "2021-10-18 10:41:20 (8.30 MB/s) - ‘2895499331.bam’ saved [396318/396318]\n",
      "\n",
      "--2021-10-18 10:41:20--  https://storage.googleapis.com/analysis-workflows-example-data/unaligned_subset_bams/normal/2895499399.bam\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 363023 (355K) [application/octet-stream]\n",
      "Saving to: ‘2895499399.bam’\n",
      "\n",
      "2895499399.bam      100%[===================>] 354.51K  --.-KB/s    in 0.05s   \n",
      "\n",
      "2021-10-18 10:41:20 (7.64 MB/s) - ‘2895499399.bam’ saved [363023/363023]\n",
      "\n",
      "--2021-10-18 10:41:20--  https://storage.googleapis.com/analysis-workflows-example-data/unaligned_subset_bams/tumor/2895499223.bam\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 769303 (751K) [application/octet-stream]\n",
      "Saving to: ‘2895499223.bam’\n",
      "\n",
      "2895499223.bam      100%[===================>] 751.27K  --.-KB/s    in 0.04s   \n",
      "\n",
      "2021-10-18 10:41:20 (19.1 MB/s) - ‘2895499223.bam’ saved [769303/769303]\n",
      "\n",
      "--2021-10-18 10:41:20--  https://storage.googleapis.com/analysis-workflows-example-data/unaligned_subset_bams/tumor/2895499237.bam\n",
      "Reusing existing connection to [storage.googleapis.com]:443.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 761991 (744K) [application/octet-stream]\n",
      "Saving to: ‘2895499237.bam’\n",
      "\n",
      "2895499237.bam      100%[===================>] 744.13K  --.-KB/s    in 0.03s   \n",
      "\n",
      "2021-10-18 10:41:20 (21.4 MB/s) - ‘2895499237.bam’ saved [761991/761991]\n",
      "\n",
      "FINISHED --2021-10-18 10:41:20--\n",
      "Total wall clock time: 30s\n",
      "Downloaded: 22 files, 673M in 26s (25.6 MB/s)\n"
     ]
    }
   ],
   "source": [
    "!wget --no-clobber https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes.fa \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes.fa.amb \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes.fa.ann \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes.fa.bwt \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes.fa.fai \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes.fa.pac \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes.fa.sa \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes.dict \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_bait.interval_list \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_target.interval_list \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_known_indels.vcf.gz \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_known_indels.vcf.gz.tbi \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_mills.vcf.gz \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_mills.vcf.gz.tbi \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_dbsnp.vcf.gz \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_dbsnp.vcf.gz.tbi \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_omni.vcf.gz \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/somatic_inputs/hla_and_brca_genes_omni.vcf.gz.tbi \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/unaligned_subset_bams/normal/2895499331.bam \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/unaligned_subset_bams/normal/2895499399.bam \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/unaligned_subset_bams/tumor/2895499223.bam \\\n",
    "https://storage.googleapis.com/analysis-workflows-example-data/unaligned_subset_bams/tumor/2895499237.bam\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ada849a6",
   "metadata": {},
   "source": [
    "## CWL Workflow"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2babdc58",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cloning into 'analysis-workflows'...\n",
      "remote: Enumerating objects: 14148, done.\u001b[K\n",
      "remote: Counting objects: 100% (431/431), done.\u001b[K\n",
      "remote: Compressing objects: 100% (216/216), done.\u001b[K\n",
      "remote: Total 14148 (delta 261), reused 337 (delta 211), pack-reused 13717\u001b[K\n",
      "Receiving objects: 100% (14148/14148), 594.49 MiB | 25.69 MiB/s, done.\n",
      "Resolving deltas: 100% (10161/10161), done.\n"
     ]
    }
   ],
   "source": [
    "!git clone https://github.com/genome/analysis-workflows.git"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "edb11b66",
   "metadata": {},
   "source": [
    "## YAML Inputs\n",
    "\n",
    "YAML format files are used to declare all of the input files, variables, parameters, etc. for the CWL workflow we inted to run.\n",
    "\n",
    "For each sample, tumor and normal, create a YAML file using the text editor of your choice, ex. vim."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9161a8f6",
   "metadata": {},
   "source": [
    "<details>\n",
    "<summary>Click for normal_inputs.yaml</summary>\n",
    "\n",
    "```yaml\n",
    "---\n",
    "\n",
    "bait_intervals:\n",
    "  class: File\n",
    "  path: hla_and_brca_genes_bait.interval_list\n",
    "\n",
    "target_intervals:\n",
    "  class: File\n",
    "  path: hla_and_brca_genes_target.interval_list\n",
    "\n",
    "sequence:\n",
    "  - sequence:\n",
    "      bam:\n",
    "        class: File\n",
    "        path: 2895499331.bam\n",
    "    readgroup: \"@RG\\tID:2895499331\\tPU:H7HY2CCXX.3\\tSM:H_NJ-HCC1395-HCC1395_BL\\tLB:H_NJ-HCC1395-HCC1395_BL-lg21-lib1\\tPL:Illumina\\tCN:WUGSC\"\n",
    "  - sequence:\n",
    "      bam:\n",
    "        class: File\n",
    "        path: 2895499399.bam\n",
    "    readgroup: \"@RG\\tID:2895499399\\tPU:H7HY2CCXX.4\\tSM:H_NJ-HCC1395-HCC1395_BL\\tLB:H_NJ-HCC1395-HCC1395_BL-lg21-lib1\\tPL:Illumina\\tCN:WUGSC\"\n",
    "\n",
    "bqsr_known_sites:\n",
    "- class: File\n",
    "  path: hla_and_brca_genes_known_indels.vcf.gz\n",
    "- class: File\n",
    "  path: hla_and_brca_genes_mills.vcf.gz\n",
    "- class: File\n",
    "  path: hla_and_brca_genes_dbsnp.vcf.gz\n",
    "\n",
    "omni_vcf:\n",
    "  class: File\n",
    "  path: hla_and_brca_genes_omni.vcf.gz\n",
    "\n",
    "picard_metric_accumulation_level: LIBRARY\n",
    "\n",
    "reference:\n",
    "  class: File\n",
    "  path: hla_and_brca_genes.fa\n",
    "\n",
    "bqsr_intervals:\n",
    "- chr6\n",
    "- chr17\n",
    "\n",
    "per_base_intervals:\n",
    "- file:\n",
    "    class: File\n",
    "    path: hla_and_brca_genes_target.interval_list\n",
    "  label: clinvar\n",
    "\n",
    "per_target_intervals:\n",
    "- file:\n",
    "    class: File\n",
    "    path: hla_and_brca_genes_target.interval_list\n",
    "  label: acmg_genes\n",
    "\n",
    "summary_intervals: []\n",
    "```\n",
    " \n",
    "</details>   "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6c0a2b96",
   "metadata": {},
   "source": [
    "<details>\n",
    "<summary>Click for tumor_inputs.yaml</summary>\n",
    "\n",
    "```yaml\n",
    "---\n",
    "\n",
    "bait_intervals:\n",
    "  class: File\n",
    "  path: hla_and_brca_genes_bait.interval_list\n",
    "\n",
    "target_intervals:\n",
    "  class: File\n",
    "  path: hla_and_brca_genes_target.interval_list\n",
    "\n",
    "sequence:\n",
    "  - sequence:\n",
    "      bam:\n",
    "        class: File\n",
    "        path: 2895499223.bam\n",
    "    readgroup: \"@RG\\tID:2895499223\\tPU:H7HY2CCXX.3\\tSM:H_NJ-HCC1395-HCC1395\\tLB:H_NJ-HCC1395-HCC1395-lg24-lib1\\tPL:Illumina\\tCN:WUGSC\"\n",
    "  - sequence:\n",
    "      bam:\n",
    "        class: File\n",
    "        path: 2895499237.bam\n",
    "    readgroup: \"@RG\\tID:2895499237\\tPU:H7HY2CCXX.4\\tSM:H_NJ-HCC1395-HCC1395\\tLB:H_NJ-HCC1395-HCC1395-lg24-lib1\\tPL:Illumina\\tCN:WUGSC\"\n",
    "\n",
    "bqsr_known_sites:\n",
    "- class: File\n",
    "  path: hla_and_brca_genes_known_indels.vcf.gz\n",
    "- class: File\n",
    "  path: hla_and_brca_genes_mills.vcf.gz\n",
    "- class: File\n",
    "  path: hla_and_brca_genes_dbsnp.vcf.gz\n",
    "\n",
    "omni_vcf:\n",
    "  class: File\n",
    "  path: hla_and_brca_genes_omni.vcf.gz\n",
    "\n",
    "picard_metric_accumulation_level: LIBRARY\n",
    "\n",
    "reference:\n",
    "  class: File\n",
    "  path: hla_and_brca_genes.fa\n",
    "\n",
    "bqsr_intervals:\n",
    "- chr6\n",
    "- chr17\n",
    "\n",
    "per_base_intervals:\n",
    "- file:\n",
    "    class: File\n",
    "    path: hla_and_brca_genes_target.interval_list\n",
    "  label: clinvar\n",
    "\n",
    "per_target_intervals:\n",
    "- file:\n",
    "    class: File\n",
    "    path: hla_and_brca_genes_target.interval_list\n",
    "  label: acmg_genes\n",
    "\n",
    "summary_intervals: []\n",
    "```\n",
    " \n",
    "</details>   "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "afed2660",
   "metadata": {},
   "source": [
    "# Execute Alignment Workflow\n",
    "\n",
    "## Normal Alignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "cdbefeaa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1;30mINFO\u001b[0m /Users/jwalker/opt/miniconda3/bin/cwltool 3.0.20210124104916\n",
      "\u001b[1;30mINFO\u001b[0m Resolved 'analysis-workflows/definitions/pipelines/alignment_exome.cwl' to 'file:///Users/jwalker/git/bfx_workshop/lectures/week_05/analysis-workflows/definitions/pipelines/alignment_exome.cwl'\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33mWorkflow checker warning:\n",
      "analysis-workflows/definitions/subworkflows/sequence_to_bqsr.cwl:23:9:  Source 'bqsr_intervals' of\n",
      "                                                                        type [\"null\", {\"type\":\n",
      "                                                                        \"array\", \"items\":\n",
      "                                                                        \"string\"}] may be\n",
      "                                                                        incompatible\n",
      "analysis-workflows/definitions/subworkflows/sequence_to_bqsr.cwl:79:13:   with sink 'intervals'\n",
      "                                                                          of type {\"type\": \"array\",\n",
      "                                                                          \"items\": \"string\",\n",
      "                                                                          \"inputBinding\":\n",
      "                                                                          {\"prefix\": \"-L\"}}\u001b[0m\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33mWorkflow checker warning:\n",
      "analysis-workflows/definitions/subworkflows/hs_metrics.cwl:100:26: Source\n",
      "                                                                   'per_base_coverage_metrics' of\n",
      "                                                                   type {\"type\": \"array\", \"items\":\n",
      "                                                                   [\"null\", \"File\"]} may be\n",
      "                                                                   incompatible\n",
      "analysis-workflows/definitions/subworkflows/hs_metrics.cwl:34:9:     with sink\n",
      "                                                                     'per_base_coverage_metrics' of\n",
      "                                                                     type {\"type\": \"array\",\n",
      "                                                                     \"items\": \"File\"}\n",
      "analysis-workflows/definitions/subworkflows/hs_metrics.cwl:126:26: Source\n",
      "                                                                   'per_target_coverage_metrics' of\n",
      "                                                                   type {\"type\": \"array\", \"items\":\n",
      "                                                                   [\"null\", \"File\"]} may be\n",
      "                                                                   incompatible\n",
      "analysis-workflows/definitions/subworkflows/hs_metrics.cwl:40:9:     with sink\n",
      "                                                                     'per_target_coverage_metrics'\n",
      "                                                                     of type {\"type\": \"array\",\n",
      "                                                                     \"items\": \"File\"}\u001b[0m\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33mWorkflow checker warning:\n",
      "analysis-workflows/definitions/pipelines/alignment_exome.cwl:38:9:   Source 'final_name' of type\n",
      "                                                                     [\"null\", \"string\"] may be\n",
      "                                                                     incompatible\n",
      "analysis-workflows/definitions/pipelines/alignment_exome.cwl:109:13:   with sink 'final_name' of\n",
      "                                                                       type \"string\"\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [workflow ] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow ] starting step alignment\n",
      "\u001b[1;30mINFO\u001b[0m [step alignment] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] starting step align\n",
      "\u001b[1;30mINFO\u001b[0m [step align] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow align] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow align] starting step align_and_tag\n",
      "\u001b[1;30mINFO\u001b[0m [step align_and_tag] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job align_and_tag] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job align_and_tag] /private/tmp/docker_tmpgrybeu59$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpgrybeu59,target=/LnCocq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpcajypmlh,target=/tmp \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/2895499331.bam,target=/var/lib/cwl/stgb36ca88f-00fb-46e3-b415-7d3179f90a64/2895499331.bam,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stg7b7b44dd-cbde-40f3-833c-e5a9705b8c51/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stg7b7b44dd-cbde-40f3-833c-e5a9705b8c51/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stg7b7b44dd-cbde-40f3-833c-e5a9705b8c51/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stg7b7b44dd-cbde-40f3-833c-e5a9705b8c51/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stg7b7b44dd-cbde-40f3-833c-e5a9705b8c51/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stg7b7b44dd-cbde-40f3-833c-e5a9705b8c51/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stg7b7b44dd-cbde-40f3-833c-e5a9705b8c51/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stg7b7b44dd-cbde-40f3-833c-e5a9705b8c51/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --workdir=/LnCocq \\\n",
      "    --read-only=true \\\n",
      "    --log-driver=none \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/LnCocq \\\n",
      "    --cidfile=/private/tmp/docker_tmpg9_de78k/20211018105201-599247.cid \\\n",
      "    mgibio/alignment_helper-cwl:1.1.0 \\\n",
      "    /bin/bash \\\n",
      "    sequence_alignment_helper.sh \\\n",
      "    -b \\\n",
      "    /var/lib/cwl/stgb36ca88f-00fb-46e3-b415-7d3179f90a64/2895499331.bam \\\n",
      "    -g \\\n",
      "    '@RG\tID:2895499331\tPU:H7HY2CCXX.3\tSM:H_NJ-HCC1395-HCC1395_BL\tLB:H_NJ-HCC1395-HCC1395_BL-lg21-lib1\tPL:Illumina\tCN:WUGSC' \\\n",
      "    -r \\\n",
      "    /var/lib/cwl/stg7b7b44dd-cbde-40f3-833c-e5a9705b8c51/hla_and_brca_genes.fa \\\n",
      "    -n \\\n",
      "    8 > /private/tmp/docker_tmpgrybeu59/refAlign.bam\n",
      "samblaster: Version 0.1.24\n",
      "samblaster: Inputting from stdin\n",
      "samblaster: Outputting to stdout\n",
      "15:52:06.565 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/opt/picard-2.18.1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Oct 18 15:52:06 UTC 2021] SamToFastq INPUT=/var/lib/cwl/stgb36ca88f-00fb-46e3-b415-7d3179f90a64/2895499331.bam FASTQ=/dev/stdout INTERLEAVE=true INCLUDE_NON_PF_READS=true    OUTPUT_PER_RG=false COMPRESS_OUTPUTS_PER_RG=false RG_TAG=PU RE_REVERSE=true CLIPPING_MIN_LENGTH=0 READ1_TRIM=0 READ2_TRIM=0 INCLUDE_NON_PRIMARY_ALIGNMENTS=false VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Mon Oct 18 15:52:06 UTC 2021] Executing as ?@046f5a89543a on Linux 5.10.47-linuxkit amd64; OpenJDK 64-Bit Server VM 1.8.0_252-8u252-b09-1~16.04-b09; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.18.1-SNAPSHOT\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[Mon Oct 18 15:52:10 UTC 2021] picard.sam.SamToFastq done. Elapsed time: 0.06 minutes.\n",
      "Runtime.totalMemory()=99614720\n",
      "[M::process] read 3544 sequences (535144 bp)...\n",
      "[M::process] 0 single-end sequences; 3544 paired-end sequences\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 1709, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (160, 197, 247)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 421)\n",
      "[M::mem_pestat] mean and std.dev: (207.18, 63.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 508)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_process_seqs] Processed 3544 reads in 0.756 CPU sec, 0.121 real sec\n",
      "samblaster: Loaded 2 header sequence entries.\n",
      "[main] Version: 0.7.15-r1140\n",
      "[main] CMD: /usr/local/bin/bwa mem -K 100000000 -t 8 -Y -p -R @RG\tID:2895499331\tPU:H7HY2CCXX.3\tSM:H_NJ-HCC1395-HCC1395_BL\tLB:H_NJ-HCC1395-HCC1395_BL-lg21-lib1\tPL:Illumina\tCN:WUGSC /var/lib/cwl/stg7b7b44dd-cbde-40f3-833c-e5a9705b8c51/hla_and_brca_genes.fa /dev/stdin\n",
      "[main] Real time: 8.018 sec; CPU: 5.573 sec\n",
      "samblaster: Marked 0 of 1772 (0.00%) read ids as duplicates using 1792k memory in 0.003S CPU seconds and 8S wall time.\n",
      "\u001b[1;30mINFO\u001b[0m [job align_and_tag] Max memory used: 1667MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job align_and_tag] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step align_and_tag] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow align] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step align] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow align_2] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow align_2] starting step align_and_tag_2\n",
      "\u001b[1;30mINFO\u001b[0m [step align_and_tag_2] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job align_and_tag_2] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job align_and_tag_2] /private/tmp/docker_tmp8a6d678y$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp8a6d678y,target=/LnCocq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpyqco_7k7,target=/tmp \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/2895499399.bam,target=/var/lib/cwl/stg09134dcc-82c5-4ab9-8a69-04ce5671bbf4/2895499399.bam,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stgcfecfb27-8f4a-4efc-814a-2e99b6883af6/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stgcfecfb27-8f4a-4efc-814a-2e99b6883af6/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stgcfecfb27-8f4a-4efc-814a-2e99b6883af6/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stgcfecfb27-8f4a-4efc-814a-2e99b6883af6/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stgcfecfb27-8f4a-4efc-814a-2e99b6883af6/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stgcfecfb27-8f4a-4efc-814a-2e99b6883af6/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stgcfecfb27-8f4a-4efc-814a-2e99b6883af6/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stgcfecfb27-8f4a-4efc-814a-2e99b6883af6/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --workdir=/LnCocq \\\n",
      "    --read-only=true \\\n",
      "    --log-driver=none \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/LnCocq \\\n",
      "    --cidfile=/private/tmp/docker_tmpehd4husj/20211018105211-750561.cid \\\n",
      "    mgibio/alignment_helper-cwl:1.1.0 \\\n",
      "    /bin/bash \\\n",
      "    sequence_alignment_helper.sh \\\n",
      "    -b \\\n",
      "    /var/lib/cwl/stg09134dcc-82c5-4ab9-8a69-04ce5671bbf4/2895499399.bam \\\n",
      "    -g \\\n",
      "    '@RG\tID:2895499399\tPU:H7HY2CCXX.4\tSM:H_NJ-HCC1395-HCC1395_BL\tLB:H_NJ-HCC1395-HCC1395_BL-lg21-lib1\tPL:Illumina\tCN:WUGSC' \\\n",
      "    -r \\\n",
      "    /var/lib/cwl/stgcfecfb27-8f4a-4efc-814a-2e99b6883af6/hla_and_brca_genes.fa \\\n",
      "    -n \\\n",
      "    8 > /private/tmp/docker_tmp8a6d678y/refAlign.bam\n",
      "samblaster: Version 0.1.24\n",
      "samblaster: Inputting from stdin\n",
      "samblaster: Outputting to stdout\n",
      "15:52:14.852 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/opt/picard-2.18.1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Oct 18 15:52:15 UTC 2021] SamToFastq INPUT=/var/lib/cwl/stg09134dcc-82c5-4ab9-8a69-04ce5671bbf4/2895499399.bam FASTQ=/dev/stdout INTERLEAVE=true INCLUDE_NON_PF_READS=true    OUTPUT_PER_RG=false COMPRESS_OUTPUTS_PER_RG=false RG_TAG=PU RE_REVERSE=true CLIPPING_MIN_LENGTH=0 READ1_TRIM=0 READ2_TRIM=0 INCLUDE_NON_PRIMARY_ALIGNMENTS=false VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Mon Oct 18 15:52:15 UTC 2021] Executing as ?@df2dcfc5afcc on Linux 5.10.47-linuxkit amd64; OpenJDK 64-Bit Server VM 1.8.0_252-8u252-b09-1~16.04-b09; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.18.1-SNAPSHOT\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[Mon Oct 18 15:52:16 UTC 2021] picard.sam.SamToFastq done. Elapsed time: 0.03 minutes.\n",
      "Runtime.totalMemory()=78643200\n",
      "[M::process] read 3272 sequences (494072 bp)...\n",
      "[M::process] 0 single-end sequences; 3272 paired-end sequences\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 1596, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (163, 202, 254)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 436)\n",
      "[M::mem_pestat] mean and std.dev: (212.14, 65.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 527)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 3272 reads in 0.561 CPU sec, 0.074 real sec\n",
      "samblaster: Loaded 2 header sequence entries.\n",
      "[main] Version: 0.7.15-r1140\n",
      "[main] CMD: /usr/local/bin/bwa mem -K 100000000 -t 8 -Y -p -R @RG\tID:2895499399\tPU:H7HY2CCXX.4\tSM:H_NJ-HCC1395-HCC1395_BL\tLB:H_NJ-HCC1395-HCC1395_BL-lg21-lib1\tPL:Illumina\tCN:WUGSC /var/lib/cwl/stgcfecfb27-8f4a-4efc-814a-2e99b6883af6/hla_and_brca_genes.fa /dev/stdin\n",
      "[main] Real time: 3.892 sec; CPU: 1.271 sec\n",
      "samblaster: Marked 0 of 1636 (0.00%) read ids as duplicates using 1768k memory in 0.004S CPU seconds and 4S wall time.\n",
      "\u001b[1;30mINFO\u001b[0m [job align_and_tag_2] Max memory used: 1567MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job align_and_tag_2] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step align_and_tag_2] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow align_2] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step align] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] starting step merge\n",
      "\u001b[1;30mINFO\u001b[0m [step merge] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job merge] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job merge] /private/tmp/docker_tmpsa7_7ajk$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpsa7_7ajk,target=/LnCocq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmparzlggfb,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpgrybeu59/refAlign.bam,target=/var/lib/cwl/stg8d50f2bf-e875-4f48-b002-5f50d310ed9e/refAlign.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp8a6d678y/refAlign.bam,target=/var/lib/cwl/stgeadbf91f-0989-4b89-b4c5-639d82234b39/refAlign.bam,readonly \\\n",
      "    --workdir=/LnCocq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/LnCocq \\\n",
      "    --cidfile=/private/tmp/docker_tmpkd8qhjnq/20211018105217-024332.cid \\\n",
      "    mgibio/bam-merge:0.1 \\\n",
      "    /usr/bin/perl \\\n",
      "    merge.pl \\\n",
      "    -t \\\n",
      "    4 \\\n",
      "    -n \\\n",
      "    final \\\n",
      "    /var/lib/cwl/stg8d50f2bf-e875-4f48-b002-5f50d310ed9e/refAlign.bam \\\n",
      "    /var/lib/cwl/stgeadbf91f-0989-4b89-b4c5-639d82234b39/refAlign.bam\n",
      "15:52:19.616 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/opt/picard-2.18.1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[Mon Oct 18 15:52:19 UTC 2021] MergeSamFiles INPUT=[/var/lib/cwl/stg8d50f2bf-e875-4f48-b002-5f50d310ed9e/refAlign.bam, /var/lib/cwl/stgeadbf91f-0989-4b89-b4c5-639d82234b39/refAlign.bam] OUTPUT=final SORT_ORDER=unsorted ASSUME_SORTED=true USE_THREADING=true VALIDATION_STRINGENCY=LENIENT    MERGE_SEQUENCE_DICTIONARIES=false VERBOSITY=INFO QUIET=false COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Mon Oct 18 15:52:19 UTC 2021] Executing as ?@2b177e0b775f on Linux 5.10.47-linuxkit amd64; OpenJDK 64-Bit Server VM 1.8.0_181-8u181-b13-0ubuntu0.16.04.1-b13; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.18.1-SNAPSHOT\n",
      "Ignoring SAM validation error: ERROR: File /var/lib/cwl/stg8d50f2bf-e875-4f48-b002-5f50d310ed9e/refAlign.bam, Error parsing SAM header. Problem parsing @PG key:value pair ID:2895499331 clashes with ID:bwa. Line:\n",
      "@PG\tID:bwa\tPN:bwa\tVN:0.7.15-r1140\tCL:/usr/local/bin/bwa mem -K 100000000 -t 8 -Y -p -R @RG\tID:2895499331\tPU:H7HY2CCXX.3\tSM:H_NJ-HCC1395-HCC1395_BL\tLB:H_NJ-HCC1395-HCC1395_BL-lg21-lib1\tPL:Illumina\tCN:WUGSC /var/lib/cwl/stg7b7b44dd-cbde-40f3-833c-e5a9705b8c51/hla_and_brca_genes.fa /dev/stdin\n",
      "Ignoring SAM validation error: ERROR: File /var/lib/cwl/stgeadbf91f-0989-4b89-b4c5-639d82234b39/refAlign.bam, Error parsing SAM header. Problem parsing @PG key:value pair ID:2895499399 clashes with ID:bwa. Line:\n",
      "@PG\tID:bwa\tPN:bwa\tVN:0.7.15-r1140\tCL:/usr/local/bin/bwa mem -K 100000000 -t 8 -Y -p -R @RG\tID:2895499399\tPU:H7HY2CCXX.4\tSM:H_NJ-HCC1395-HCC1395_BL\tLB:H_NJ-HCC1395-HCC1395_BL-lg21-lib1\tPL:Illumina\tCN:WUGSC /var/lib/cwl/stgcfecfb27-8f4a-4efc-814a-2e99b6883af6/hla_and_brca_genes.fa /dev/stdin\n",
      "INFO\t2021-10-18 15:52:19\tMergeSamFiles\tInput files are in same order as output so sorting to temp directory is not needed.\n",
      "INFO\t2021-10-18 15:52:20\tMergeSamFiles\tFinished reading inputs.\n",
      "[Mon Oct 18 15:52:20 UTC 2021] picard.sam.MergeSamFiles done. Elapsed time: 0.01 minutes.\n",
      "Runtime.totalMemory()=78643200\n",
      "\u001b[1;30mINFO\u001b[0m [job merge] Max memory used: 217MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job merge] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step merge] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] starting step name_sort\n",
      "\u001b[1;30mINFO\u001b[0m [step name_sort] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job name_sort] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job name_sort] /private/tmp/docker_tmph1p7wtie$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmph1p7wtie,target=/LnCocq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpp97xlpgx,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpsa7_7ajk/final,target=/var/lib/cwl/stg5adc9ba4-e47a-4d72-8727-d9c712eb1e76/final,readonly \\\n",
      "    --workdir=/LnCocq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/LnCocq \\\n",
      "    --cidfile=/private/tmp/docker_tmpth9f0kc9/20211018105220-566597.cid \\\n",
      "    mgibio/sambamba-cwl:0.6.4 \\\n",
      "    /usr/bin/sambamba \\\n",
      "    sort \\\n",
      "    -t \\\n",
      "    8 \\\n",
      "    -m \\\n",
      "    22G \\\n",
      "    -n \\\n",
      "    -o \\\n",
      "    final.NameSorted.bam \\\n",
      "    /var/lib/cwl/stg5adc9ba4-e47a-4d72-8727-d9c712eb1e76/final\n",
      "\u001b[1;30mINFO\u001b[0m [job name_sort] Max memory used: 0MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job name_sort] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step name_sort] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] starting step mark_duplicates_and_sort\n",
      "\u001b[1;30mINFO\u001b[0m [step mark_duplicates_and_sort] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job mark_duplicates_and_sort] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job mark_duplicates_and_sort] /private/tmp/docker_tmp_foenbkw$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp_foenbkw,target=/LnCocq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpa9srw2n2,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmph1p7wtie/final.NameSorted.bam,target=/var/lib/cwl/stgdf173472-ead5-4362-ad89-15f5435ad1c5/final.NameSorted.bam,readonly \\\n",
      "    --workdir=/LnCocq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/LnCocq \\\n",
      "    --cidfile=/private/tmp/docker_tmp0mq0biaj/20211018105221-821971.cid \\\n",
      "    mgibio/mark_duplicates-cwl:1.0.1 \\\n",
      "    /bin/bash \\\n",
      "    markduplicates_helper.sh \\\n",
      "    /var/lib/cwl/stgdf173472-ead5-4362-ad89-15f5435ad1c5/final.NameSorted.bam \\\n",
      "    8 \\\n",
      "    MarkedSorted.bam \\\n",
      "    final.NameSorted.mark_dups_metrics.txt \\\n",
      "    queryname\n",
      "15:52:24.614 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/opt/picard-2.18.1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "INFO\t2021-10-18 15:52:24\tMarkDuplicates\tStart of doWork freeMemory: 72614232; totalMemory: 78643200; maxMemory: 15271460864\n",
      "INFO\t2021-10-18 15:52:24\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2021-10-18 15:52:24\tMarkDuplicates\tWill retain up to 55331379 data points before spilling to disk.\n",
      "INFO\t2021-10-18 15:52:26\tMarkDuplicates\tRead 6821 records. 0 pairs never matched.\n",
      "INFO\t2021-10-18 15:52:26\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 95348104; totalMemory: 543162368; maxMemory: 15271460864\n",
      "INFO\t2021-10-18 15:52:26\tMarkDuplicates\tWill retain up to 477233152 duplicate indices before spilling to disk.\n",
      "INFO\t2021-10-18 15:52:46\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2021-10-18 15:52:46\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2021-10-18 15:52:46\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2021-10-18 15:52:50\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 539314824; totalMemory: 4361551872; maxMemory: 15271460864\n",
      "INFO\t2021-10-18 15:52:50\tMarkDuplicates\tMarking 219 records as duplicates.\n",
      "INFO\t2021-10-18 15:52:50\tMarkDuplicates\tFound 36 optical duplicate clusters.\n",
      "INFO\t2021-10-18 15:52:50\tMarkDuplicates\tReads are assumed to be ordered by: queryname\n",
      "INFO\t2021-10-18 15:52:51\tMarkDuplicates\tBefore output close freeMemory: 4356749152; totalMemory: 4361551872; maxMemory: 15271460864\n",
      "INFO\t2021-10-18 15:52:51\tMarkDuplicates\tAfter output close freeMemory: 4356823848; totalMemory: 4361551872; maxMemory: 15271460864\n",
      "\u001b[1;30mINFO\u001b[0m [job mark_duplicates_and_sort] Max memory used: 14260MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job mark_duplicates_and_sort] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step mark_duplicates_and_sort] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] starting step bqsr\n",
      "\u001b[1;30mINFO\u001b[0m [step bqsr] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job bqsr] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job bqsr] /private/tmp/docker_tmp72ae3zro$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp72ae3zro,target=/LnCocq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpjict9501,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp_foenbkw/MarkedSorted.bam,target=/var/lib/cwl/stgbfcfed8c-8597-4fd3-86e5-d034ee27837e/MarkedSorted.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp_foenbkw/MarkedSorted.bam.bai,target=/var/lib/cwl/stgbfcfed8c-8597-4fd3-86e5-d034ee27837e/MarkedSorted.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_known_indels.vcf.gz,target=/var/lib/cwl/stgb2c86d22-32f1-40c9-a1ba-3ac2955ba8ee/hla_and_brca_genes_known_indels.vcf.gz,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_known_indels.vcf.gz.tbi,target=/var/lib/cwl/stgb2c86d22-32f1-40c9-a1ba-3ac2955ba8ee/hla_and_brca_genes_known_indels.vcf.gz.tbi,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_mills.vcf.gz,target=/var/lib/cwl/stgd6b38a73-8bdf-4933-88fd-1fcde1984f7b/hla_and_brca_genes_mills.vcf.gz,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_mills.vcf.gz.tbi,target=/var/lib/cwl/stgd6b38a73-8bdf-4933-88fd-1fcde1984f7b/hla_and_brca_genes_mills.vcf.gz.tbi,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_dbsnp.vcf.gz,target=/var/lib/cwl/stgf2799cc9-3f48-4be3-accb-9786e4ccb0ce/hla_and_brca_genes_dbsnp.vcf.gz,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_dbsnp.vcf.gz.tbi,target=/var/lib/cwl/stgf2799cc9-3f48-4be3-accb-9786e4ccb0ce/hla_and_brca_genes_dbsnp.vcf.gz.tbi,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stgcdaf15e5-2517-4f45-87ec-6cac55c2f6d6/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stgcdaf15e5-2517-4f45-87ec-6cac55c2f6d6/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stgcdaf15e5-2517-4f45-87ec-6cac55c2f6d6/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stgcdaf15e5-2517-4f45-87ec-6cac55c2f6d6/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stgcdaf15e5-2517-4f45-87ec-6cac55c2f6d6/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stgcdaf15e5-2517-4f45-87ec-6cac55c2f6d6/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stgcdaf15e5-2517-4f45-87ec-6cac55c2f6d6/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stgcdaf15e5-2517-4f45-87ec-6cac55c2f6d6/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --workdir=/LnCocq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/LnCocq \\\n",
      "    --cidfile=/private/tmp/docker_tmprjrklicq/20211018105256-813349.cid \\\n",
      "    broadinstitute/gatk:4.1.8.1 \\\n",
      "    /gatk/gatk \\\n",
      "    --java-options \\\n",
      "    -Xmx16g \\\n",
      "    BaseRecalibrator \\\n",
      "    -O \\\n",
      "    /LnCocq/bqsr.table \\\n",
      "    --known-sites \\\n",
      "    /var/lib/cwl/stgb2c86d22-32f1-40c9-a1ba-3ac2955ba8ee/hla_and_brca_genes_known_indels.vcf.gz \\\n",
      "    --known-sites \\\n",
      "    /var/lib/cwl/stgd6b38a73-8bdf-4933-88fd-1fcde1984f7b/hla_and_brca_genes_mills.vcf.gz \\\n",
      "    -L \\\n",
      "    chr6 \\\n",
      "    -L \\\n",
      "    chr17 \\\n",
      "    --known-sites \\\n",
      "    /var/lib/cwl/stgf2799cc9-3f48-4be3-accb-9786e4ccb0ce/hla_and_brca_genes_dbsnp.vcf.gz \\\n",
      "    -R \\\n",
      "    /var/lib/cwl/stgcdaf15e5-2517-4f45-87ec-6cac55c2f6d6/hla_and_brca_genes.fa \\\n",
      "    -I \\\n",
      "    /var/lib/cwl/stgbfcfed8c-8597-4fd3-86e5-d034ee27837e/MarkedSorted.bam\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15:53:03.229 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/gatk/gatk-package-4.1.8.1-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "Oct 18, 2021 3:53:04 PM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine\n",
      "INFO: Failed to detect whether we are running on Google Compute Engine.\n",
      "15:53:04.196 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
      "15:53:04.197 INFO  BaseRecalibrator - The Genome Analysis Toolkit (GATK) v4.1.8.1\n",
      "15:53:04.197 INFO  BaseRecalibrator - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "15:53:04.198 INFO  BaseRecalibrator - Executing as ?@4d330e02ef18 on Linux v5.10.47-linuxkit amd64\n",
      "15:53:04.198 INFO  BaseRecalibrator - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_242-8u242-b08-0ubuntu3~18.04-b08\n",
      "15:53:04.199 INFO  BaseRecalibrator - Start Date/Time: October 18, 2021 3:53:03 PM GMT\n",
      "15:53:04.199 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
      "15:53:04.199 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
      "15:53:04.200 INFO  BaseRecalibrator - HTSJDK Version: 2.23.0\n",
      "15:53:04.200 INFO  BaseRecalibrator - Picard Version: 2.22.8\n",
      "15:53:04.200 INFO  BaseRecalibrator - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "15:53:04.201 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "15:53:04.201 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "15:53:04.201 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "15:53:04.202 INFO  BaseRecalibrator - Deflater: IntelDeflater\n",
      "15:53:04.202 INFO  BaseRecalibrator - Inflater: IntelInflater\n",
      "15:53:04.202 INFO  BaseRecalibrator - GCS max retries/reopens: 20\n",
      "15:53:04.202 INFO  BaseRecalibrator - Requester pays: disabled\n",
      "15:53:04.203 INFO  BaseRecalibrator - Initializing engine\n",
      "15:53:04.720 INFO  FeatureManager - Using codec VCFCodec to read file file:///var/lib/cwl/stgb2c86d22-32f1-40c9-a1ba-3ac2955ba8ee/hla_and_brca_genes_known_indels.vcf.gz\n",
      "15:53:04.763 INFO  FeatureManager - Using codec VCFCodec to read file file:///var/lib/cwl/stgd6b38a73-8bdf-4933-88fd-1fcde1984f7b/hla_and_brca_genes_mills.vcf.gz\n",
      "15:53:04.789 INFO  FeatureManager - Using codec VCFCodec to read file file:///var/lib/cwl/stgf2799cc9-3f48-4be3-accb-9786e4ccb0ce/hla_and_brca_genes_dbsnp.vcf.gz\n",
      "15:53:04.815 INFO  IntervalArgumentCollection - Processing 254063420 bp from intervals\n",
      "15:53:04.824 INFO  BaseRecalibrator - Done initializing engine\n",
      "15:53:04.836 INFO  BaseRecalibrationEngine - The covariates being used here: \n",
      "15:53:04.836 INFO  BaseRecalibrationEngine - \tReadGroupCovariate\n",
      "15:53:04.836 INFO  BaseRecalibrationEngine - \tQualityScoreCovariate\n",
      "15:53:04.836 INFO  BaseRecalibrationEngine - \tContextCovariate\n",
      "15:53:04.836 INFO  BaseRecalibrationEngine - \tCycleCovariate\n",
      "15:53:04.840 INFO  ProgressMeter - Starting traversal\n",
      "15:53:04.840 INFO  ProgressMeter -        Current Locus  Elapsed Minutes       Reads Processed     Reads/Minute\n",
      "15:53:05.798 INFO  BaseRecalibrator - 13 read(s) filtered by: MappingQualityNotZeroReadFilter \n",
      "0 read(s) filtered by: MappingQualityAvailableReadFilter \n",
      "0 read(s) filtered by: MappedReadFilter \n",
      "0 read(s) filtered by: NotSecondaryAlignmentReadFilter \n",
      "439 read(s) filtered by: NotDuplicateReadFilter \n",
      "0 read(s) filtered by: PassesVendorQualityCheckReadFilter \n",
      "0 read(s) filtered by: WellformedReadFilter \n",
      "452 total reads filtered\n",
      "15:53:05.802 INFO  ProgressMeter -       chr17:43133667              0.0                  6369         398893.5\n",
      "15:53:05.802 INFO  ProgressMeter - Traversal complete. Processed 6369 total reads in 0.0 minutes.\n",
      "15:53:05.928 INFO  BaseRecalibrator - Calculating quantized quality scores...\n",
      "15:53:05.949 INFO  BaseRecalibrator - Writing recalibration report...\n",
      "15:53:48.420 INFO  BaseRecalibrator - ...done!\n",
      "15:53:48.421 INFO  BaseRecalibrator - BaseRecalibrator was able to recalibrate 6369 reads\n",
      "15:53:48.421 INFO  BaseRecalibrator - Shutting down engine\n",
      "[October 18, 2021 3:53:48 PM GMT] org.broadinstitute.hellbender.tools.walkers.bqsr.BaseRecalibrator done. Elapsed time: 0.75 minutes.\n",
      "Runtime.totalMemory()=352845824\n",
      "Tool returned:\n",
      "SUCCESS\n",
      "Using GATK jar /gatk/gatk-package-4.1.8.1-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xmx16g -jar /gatk/gatk-package-4.1.8.1-local.jar BaseRecalibrator -O /LnCocq/bqsr.table --known-sites /var/lib/cwl/stgb2c86d22-32f1-40c9-a1ba-3ac2955ba8ee/hla_and_brca_genes_known_indels.vcf.gz --known-sites /var/lib/cwl/stgd6b38a73-8bdf-4933-88fd-1fcde1984f7b/hla_and_brca_genes_mills.vcf.gz -L chr6 -L chr17 --known-sites /var/lib/cwl/stgf2799cc9-3f48-4be3-accb-9786e4ccb0ce/hla_and_brca_genes_dbsnp.vcf.gz -R /var/lib/cwl/stgcdaf15e5-2517-4f45-87ec-6cac55c2f6d6/hla_and_brca_genes.fa -I /var/lib/cwl/stgbfcfed8c-8597-4fd3-86e5-d034ee27837e/MarkedSorted.bam\n",
      "\u001b[1;30mINFO\u001b[0m [job bqsr] Max memory used: 1351MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job bqsr] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step bqsr] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] starting step apply_bqsr\n",
      "\u001b[1;30mINFO\u001b[0m [step apply_bqsr] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job apply_bqsr] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job apply_bqsr] /private/tmp/docker_tmp6rjhwo5g$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp6rjhwo5g,target=/LnCocq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp35cixp0k,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp_foenbkw/MarkedSorted.bam,target=/var/lib/cwl/stge596a5a0-a6c2-483a-aa6e-0df66afc36de/MarkedSorted.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp_foenbkw/MarkedSorted.bam.bai,target=/var/lib/cwl/stge596a5a0-a6c2-483a-aa6e-0df66afc36de/MarkedSorted.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp72ae3zro/bqsr.table,target=/var/lib/cwl/stg52d14de4-58dc-4f79-8bba-761c7c915d1f/bqsr.table,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stg7ad46966-c22e-4fa4-ba6b-4d9d76eaeea3/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stg7ad46966-c22e-4fa4-ba6b-4d9d76eaeea3/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stg7ad46966-c22e-4fa4-ba6b-4d9d76eaeea3/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stg7ad46966-c22e-4fa4-ba6b-4d9d76eaeea3/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stg7ad46966-c22e-4fa4-ba6b-4d9d76eaeea3/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stg7ad46966-c22e-4fa4-ba6b-4d9d76eaeea3/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stg7ad46966-c22e-4fa4-ba6b-4d9d76eaeea3/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stg7ad46966-c22e-4fa4-ba6b-4d9d76eaeea3/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --workdir=/LnCocq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/LnCocq \\\n",
      "    --cidfile=/private/tmp/docker_tmpkoza_4j0/20211018105348-710249.cid \\\n",
      "    broadinstitute/gatk:4.1.8.1 \\\n",
      "    /gatk/gatk \\\n",
      "    --java-options \\\n",
      "    -Xmx16g \\\n",
      "    ApplyBQSR \\\n",
      "    -O \\\n",
      "    /LnCocq/final.bam \\\n",
      "    --static-quantized-quals \\\n",
      "    10 \\\n",
      "    --static-quantized-quals \\\n",
      "    20 \\\n",
      "    --static-quantized-quals \\\n",
      "    30 \\\n",
      "    -R \\\n",
      "    /var/lib/cwl/stg7ad46966-c22e-4fa4-ba6b-4d9d76eaeea3/hla_and_brca_genes.fa \\\n",
      "    -I \\\n",
      "    /var/lib/cwl/stge596a5a0-a6c2-483a-aa6e-0df66afc36de/MarkedSorted.bam \\\n",
      "    -bqsr \\\n",
      "    /var/lib/cwl/stg52d14de4-58dc-4f79-8bba-761c7c915d1f/bqsr.table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15:53:53.481 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/gatk/gatk-package-4.1.8.1-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "Oct 18, 2021 3:53:54 PM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine\n",
      "INFO: Failed to detect whether we are running on Google Compute Engine.\n",
      "15:53:54.020 INFO  ApplyBQSR - ------------------------------------------------------------\n",
      "15:53:54.021 INFO  ApplyBQSR - The Genome Analysis Toolkit (GATK) v4.1.8.1\n",
      "15:53:54.021 INFO  ApplyBQSR - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "15:53:54.022 INFO  ApplyBQSR - Executing as ?@3fb1391b7db1 on Linux v5.10.47-linuxkit amd64\n",
      "15:53:54.022 INFO  ApplyBQSR - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_242-8u242-b08-0ubuntu3~18.04-b08\n",
      "15:53:54.022 INFO  ApplyBQSR - Start Date/Time: October 18, 2021 3:53:53 PM GMT\n",
      "15:53:54.023 INFO  ApplyBQSR - ------------------------------------------------------------\n",
      "15:53:54.023 INFO  ApplyBQSR - ------------------------------------------------------------\n",
      "15:53:54.023 INFO  ApplyBQSR - HTSJDK Version: 2.23.0\n",
      "15:53:54.024 INFO  ApplyBQSR - Picard Version: 2.22.8\n",
      "15:53:54.024 INFO  ApplyBQSR - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "15:53:54.024 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "15:53:54.024 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "15:53:54.024 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "15:53:54.025 INFO  ApplyBQSR - Deflater: IntelDeflater\n",
      "15:53:54.025 INFO  ApplyBQSR - Inflater: IntelInflater\n",
      "15:53:54.025 INFO  ApplyBQSR - GCS max retries/reopens: 20\n",
      "15:53:54.025 INFO  ApplyBQSR - Requester pays: disabled\n",
      "15:53:54.026 INFO  ApplyBQSR - Initializing engine\n",
      "15:53:54.448 INFO  ApplyBQSR - Done initializing engine\n",
      "15:53:54.485 INFO  ProgressMeter - Starting traversal\n",
      "15:53:54.486 INFO  ProgressMeter -        Current Locus  Elapsed Minutes       Reads Processed     Reads/Minute\n",
      "15:53:55.317 INFO  ApplyBQSR - 0 read(s) filtered by: WellformedReadFilter \n",
      "\n",
      "15:53:55.318 INFO  ProgressMeter -       chr17:43104086              0.0                  6821         491899.0\n",
      "15:53:55.318 INFO  ProgressMeter - Traversal complete. Processed 6821 total reads in 0.0 minutes.\n",
      "15:53:55.377 INFO  ApplyBQSR - Shutting down engine\n",
      "[October 18, 2021 3:53:55 PM GMT] org.broadinstitute.hellbender.tools.walkers.bqsr.ApplyBQSR done. Elapsed time: 0.03 minutes.\n",
      "Runtime.totalMemory()=265814016\n",
      "Using GATK jar /gatk/gatk-package-4.1.8.1-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xmx16g -jar /gatk/gatk-package-4.1.8.1-local.jar ApplyBQSR -O /LnCocq/final.bam --static-quantized-quals 10 --static-quantized-quals 20 --static-quantized-quals 30 -R /var/lib/cwl/stg7ad46966-c22e-4fa4-ba6b-4d9d76eaeea3/hla_and_brca_genes.fa -I /var/lib/cwl/stge596a5a0-a6c2-483a-aa6e-0df66afc36de/MarkedSorted.bam -bqsr /var/lib/cwl/stg52d14de4-58dc-4f79-8bba-761c7c915d1f/bqsr.table\n",
      "\u001b[1;30mINFO\u001b[0m [job apply_bqsr] Max memory used: 1048MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job apply_bqsr] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step apply_bqsr] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] starting step index_bam\n",
      "\u001b[1;30mINFO\u001b[0m [step index_bam] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job index_bam] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job index_bam] /private/tmp/docker_tmpz1yvkp83$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpz1yvkp83,target=/LnCocq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp0d3yk287,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp6rjhwo5g/final.bai,target=/var/lib/cwl/stg141d88a6-51f4-4f9e-8f68-3fb4d8cd7e69/final.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp6rjhwo5g/final.bam,target=/LnCocq/final.bam,readonly \\\n",
      "    --workdir=/LnCocq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/LnCocq \\\n",
      "    --cidfile=/private/tmp/docker_tmpa6w0bt1y/20211018105355-875133.cid \\\n",
      "    quay.io/biocontainers/samtools:1.11--h6270b1f_0 \\\n",
      "    /bin/sh \\\n",
      "    -c \\\n",
      "    '/usr/local/bin/samtools' 'index' '/LnCocq/final.bam' '/LnCocq/final.bam.bai'  &&  'cp' 'final.bam.bai' '/LnCocq/final.bai'\n",
      "\u001b[1;30mINFO\u001b[0m [job index_bam] Max memory used: 0MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job index_bam] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step index_bam] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step alignment] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow ] starting step qc\n",
      "\u001b[1;30mINFO\u001b[0m [step qc] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] starting step collect_detailed_hs_metrics\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_detailed_hs_metrics] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow collect_detailed_hs_metrics] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow collect_detailed_hs_metrics] starting step collect_per_base_hs_metrics\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_per_base_hs_metrics] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job collect_per_base_hs_metrics] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_per_base_hs_metrics] /private/tmp/docker_tmpi237vox7$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpi237vox7,target=/LnCocq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp20x77bj3,target=/tmp \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_target.interval_list,target=/var/lib/cwl/stga01cb5e9-c152-49fc-a341-a6ba081ff562/hla_and_brca_genes_target.interval_list,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp6rjhwo5g/final.bam,target=/var/lib/cwl/stgbd11c673-b53f-49df-9b55-f33b6b0d75f2/final.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpz1yvkp83/final.bam.bai,target=/var/lib/cwl/stgbd11c673-b53f-49df-9b55-f33b6b0d75f2/final.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpz1yvkp83/final.bai,target=/var/lib/cwl/stgbd11c673-b53f-49df-9b55-f33b6b0d75f2/final.bai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stg2904dccf-3634-4e55-b464-9e80d3032c74/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stg2904dccf-3634-4e55-b464-9e80d3032c74/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stg2904dccf-3634-4e55-b464-9e80d3032c74/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stg2904dccf-3634-4e55-b464-9e80d3032c74/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stg2904dccf-3634-4e55-b464-9e80d3032c74/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stg2904dccf-3634-4e55-b464-9e80d3032c74/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stg2904dccf-3634-4e55-b464-9e80d3032c74/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stg2904dccf-3634-4e55-b464-9e80d3032c74/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --workdir=/LnCocq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/LnCocq \\\n",
      "    --cidfile=/private/tmp/docker_tmpbrpb5pl4/20211018105357-165565.cid \\\n",
      "    broadinstitute/picard:2.23.6 \\\n",
      "    /usr/bin/java \\\n",
      "    -Xmx48g \\\n",
      "    -jar \\\n",
      "    /usr/picard/picard.jar \\\n",
      "    CollectHsMetrics \\\n",
      "    O= \\\n",
      "    /LnCocq/final.base-clinvar-HsMetrics.txt \\\n",
      "    BAIT_INTERVALS= \\\n",
      "    /var/lib/cwl/stga01cb5e9-c152-49fc-a341-a6ba081ff562/hla_and_brca_genes_target.interval_list \\\n",
      "    I= \\\n",
      "    /var/lib/cwl/stgbd11c673-b53f-49df-9b55-f33b6b0d75f2/final.bam \\\n",
      "    METRIC_ACCUMULATION_LEVEL= \\\n",
      "    ALL_READS \\\n",
      "    MINIMUM_BASE_QUALITY= \\\n",
      "    1 \\\n",
      "    MINIMUM_MAPPING_QUALITY= \\\n",
      "    1 \\\n",
      "    PER_BASE_COVERAGE= \\\n",
      "    final.base-clinvar-PerBaseCoverage.txt \\\n",
      "    R= \\\n",
      "    /var/lib/cwl/stg2904dccf-3634-4e55-b464-9e80d3032c74/hla_and_brca_genes.fa \\\n",
      "    TARGET_INTERVALS= \\\n",
      "    /var/lib/cwl/stga01cb5e9-c152-49fc-a341-a6ba081ff562/hla_and_brca_genes_target.interval_list\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO\t2021-10-18 15:53:58\tCollectHsMetrics\t\n",
      "\n",
      "********** NOTE: Picard's command line syntax is changing.\n",
      "**********\n",
      "********** For more information, please see:\n",
      "********** https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
      "**********\n",
      "********** The command line looks like this in the new syntax:\n",
      "**********\n",
      "**********    CollectHsMetrics -O  /LnCocq/final.base-clinvar-HsMetrics.txt -BAIT_INTERVALS  /var/lib/cwl/stga01cb5e9-c152-49fc-a341-a6ba081ff562/hla_and_brca_genes_target.interval_list -I  /var/lib/cwl/stgbd11c673-b53f-49df-9b55-f33b6b0d75f2/final.bam -METRIC_ACCUMULATION_LEVEL  ALL_READS -MINIMUM_BASE_QUALITY  1 -MINIMUM_MAPPING_QUALITY  1 -PER_BASE_COVERAGE  final.base-clinvar-PerBaseCoverage.txt -R  /var/lib/cwl/stg2904dccf-3634-4e55-b464-9e80d3032c74/hla_and_brca_genes.fa -TARGET_INTERVALS  /var/lib/cwl/stga01cb5e9-c152-49fc-a341-a6ba081ff562/hla_and_brca_genes_target.interval_list\n",
      "**********\n",
      "\n",
      "\n",
      "15:53:59.441 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/usr/picard/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Oct 18 15:53:59 UTC 2021] CollectHsMetrics BAIT_INTERVALS=[/var/lib/cwl/stga01cb5e9-c152-49fc-a341-a6ba081ff562/hla_and_brca_genes_target.interval_list] TARGET_INTERVALS=[/var/lib/cwl/stga01cb5e9-c152-49fc-a341-a6ba081ff562/hla_and_brca_genes_target.interval_list] INPUT=/var/lib/cwl/stgbd11c673-b53f-49df-9b55-f33b6b0d75f2/final.bam OUTPUT=/LnCocq/final.base-clinvar-HsMetrics.txt METRIC_ACCUMULATION_LEVEL=[ALL_READS] PER_BASE_COVERAGE=final.base-clinvar-PerBaseCoverage.txt MINIMUM_MAPPING_QUALITY=1 MINIMUM_BASE_QUALITY=1 REFERENCE_SEQUENCE=/var/lib/cwl/stg2904dccf-3634-4e55-b464-9e80d3032c74/hla_and_brca_genes.fa    NEAR_DISTANCE=250 CLIP_OVERLAPPING_READS=true INCLUDE_INDELS=false COVERAGE_CAP=200 SAMPLE_SIZE=10000 ALLELE_FRACTION=[0.001, 0.005, 0.01, 0.02, 0.05, 0.1, 0.2, 0.3, 0.5] VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Mon Oct 18 15:53:59 UTC 2021] Executing as ?@618fcb8c3740 on Linux 5.10.47-linuxkit amd64; OpenJDK 64-Bit Server VM 11.0.8+10-post-Debian-1deb10u1; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.23.6-SNAPSHOT\n",
      "INFO\t2021-10-18 15:54:00\tTheoreticalSensitivity\tCreating Roulette Wheel\n",
      "INFO\t2021-10-18 15:54:00\tTheoreticalSensitivity\tCalculating quality sums from quality sampler\n",
      "INFO\t2021-10-18 15:54:00\tTheoreticalSensitivity\t0 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:01\tTheoreticalSensitivity\t1000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:01\tTheoreticalSensitivity\t2000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:02\tTheoreticalSensitivity\t3000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:03\tTheoreticalSensitivity\t4000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:04\tTheoreticalSensitivity\t5000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:05\tTheoreticalSensitivity\t6000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:07\tTheoreticalSensitivity\t7000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:08\tTheoreticalSensitivity\t8000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:09\tTheoreticalSensitivity\t9000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:10\tTheoreticalSensitivity\tCalculating theoretical het sensitivity\n",
      "INFO\t2021-10-18 15:54:10\tTargetMetricsCollector\tCalculating GC metrics\n",
      "[Mon Oct 18 15:54:10 UTC 2021] picard.analysis.directed.CollectHsMetrics done. Elapsed time: 0.19 minutes.\n",
      "Runtime.totalMemory()=159383552\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_per_base_hs_metrics] Max memory used: 679MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_per_base_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_per_base_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow collect_detailed_hs_metrics] starting step collect_summary_hs_metrics\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job step collect_summary_hs_metrics] Notice: scattering over empty input in 'bait_intervals', 'target_intervals', 'output_prefix'.  All outputs will be empty.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_summary_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow collect_detailed_hs_metrics] starting step collect_per_target_hs_metrics\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_per_target_hs_metrics] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job collect_per_target_hs_metrics] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_per_target_hs_metrics] /private/tmp/docker_tmpe8ut_72u$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpe8ut_72u,target=/LnCocq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp_mysj1m4,target=/tmp \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_target.interval_list,target=/var/lib/cwl/stg72b77294-0f53-4a70-a49a-3b91335f8591/hla_and_brca_genes_target.interval_list,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp6rjhwo5g/final.bam,target=/var/lib/cwl/stg2f3572f7-c1ed-404b-9e8b-b180fe48a66a/final.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpz1yvkp83/final.bam.bai,target=/var/lib/cwl/stg2f3572f7-c1ed-404b-9e8b-b180fe48a66a/final.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpz1yvkp83/final.bai,target=/var/lib/cwl/stg2f3572f7-c1ed-404b-9e8b-b180fe48a66a/final.bai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stgb5ade8e9-9fd4-40fc-9c9d-c1a57b5ce651/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stgb5ade8e9-9fd4-40fc-9c9d-c1a57b5ce651/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stgb5ade8e9-9fd4-40fc-9c9d-c1a57b5ce651/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stgb5ade8e9-9fd4-40fc-9c9d-c1a57b5ce651/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stgb5ade8e9-9fd4-40fc-9c9d-c1a57b5ce651/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stgb5ade8e9-9fd4-40fc-9c9d-c1a57b5ce651/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stgb5ade8e9-9fd4-40fc-9c9d-c1a57b5ce651/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stgb5ade8e9-9fd4-40fc-9c9d-c1a57b5ce651/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --workdir=/LnCocq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/LnCocq \\\n",
      "    --cidfile=/private/tmp/docker_tmpodfqsnod/20211018105410-907529.cid \\\n",
      "    broadinstitute/picard:2.23.6 \\\n",
      "    /usr/bin/java \\\n",
      "    -Xmx48g \\\n",
      "    -jar \\\n",
      "    /usr/picard/picard.jar \\\n",
      "    CollectHsMetrics \\\n",
      "    O= \\\n",
      "    /LnCocq/final.target-acmg_genes-HsMetrics.txt \\\n",
      "    BAIT_INTERVALS= \\\n",
      "    /var/lib/cwl/stg72b77294-0f53-4a70-a49a-3b91335f8591/hla_and_brca_genes_target.interval_list \\\n",
      "    I= \\\n",
      "    /var/lib/cwl/stg2f3572f7-c1ed-404b-9e8b-b180fe48a66a/final.bam \\\n",
      "    METRIC_ACCUMULATION_LEVEL= \\\n",
      "    ALL_READS \\\n",
      "    MINIMUM_BASE_QUALITY= \\\n",
      "    1 \\\n",
      "    MINIMUM_MAPPING_QUALITY= \\\n",
      "    1 \\\n",
      "    PER_TARGET_COVERAGE= \\\n",
      "    final.target-acmg_genes-PerTargetCoverage.txt \\\n",
      "    R= \\\n",
      "    /var/lib/cwl/stgb5ade8e9-9fd4-40fc-9c9d-c1a57b5ce651/hla_and_brca_genes.fa \\\n",
      "    TARGET_INTERVALS= \\\n",
      "    /var/lib/cwl/stg72b77294-0f53-4a70-a49a-3b91335f8591/hla_and_brca_genes_target.interval_list\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO\t2021-10-18 15:54:12\tCollectHsMetrics\t\n",
      "\n",
      "********** NOTE: Picard's command line syntax is changing.\n",
      "**********\n",
      "********** For more information, please see:\n",
      "********** https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
      "**********\n",
      "********** The command line looks like this in the new syntax:\n",
      "**********\n",
      "**********    CollectHsMetrics -O  /LnCocq/final.target-acmg_genes-HsMetrics.txt -BAIT_INTERVALS  /var/lib/cwl/stg72b77294-0f53-4a70-a49a-3b91335f8591/hla_and_brca_genes_target.interval_list -I  /var/lib/cwl/stg2f3572f7-c1ed-404b-9e8b-b180fe48a66a/final.bam -METRIC_ACCUMULATION_LEVEL  ALL_READS -MINIMUM_BASE_QUALITY  1 -MINIMUM_MAPPING_QUALITY  1 -PER_TARGET_COVERAGE  final.target-acmg_genes-PerTargetCoverage.txt -R  /var/lib/cwl/stgb5ade8e9-9fd4-40fc-9c9d-c1a57b5ce651/hla_and_brca_genes.fa -TARGET_INTERVALS  /var/lib/cwl/stg72b77294-0f53-4a70-a49a-3b91335f8591/hla_and_brca_genes_target.interval_list\n",
      "**********\n",
      "\n",
      "\n",
      "15:54:12.976 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/usr/picard/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Oct 18 15:54:13 UTC 2021] CollectHsMetrics BAIT_INTERVALS=[/var/lib/cwl/stg72b77294-0f53-4a70-a49a-3b91335f8591/hla_and_brca_genes_target.interval_list] TARGET_INTERVALS=[/var/lib/cwl/stg72b77294-0f53-4a70-a49a-3b91335f8591/hla_and_brca_genes_target.interval_list] INPUT=/var/lib/cwl/stg2f3572f7-c1ed-404b-9e8b-b180fe48a66a/final.bam OUTPUT=/LnCocq/final.target-acmg_genes-HsMetrics.txt METRIC_ACCUMULATION_LEVEL=[ALL_READS] PER_TARGET_COVERAGE=final.target-acmg_genes-PerTargetCoverage.txt MINIMUM_MAPPING_QUALITY=1 MINIMUM_BASE_QUALITY=1 REFERENCE_SEQUENCE=/var/lib/cwl/stgb5ade8e9-9fd4-40fc-9c9d-c1a57b5ce651/hla_and_brca_genes.fa    NEAR_DISTANCE=250 CLIP_OVERLAPPING_READS=true INCLUDE_INDELS=false COVERAGE_CAP=200 SAMPLE_SIZE=10000 ALLELE_FRACTION=[0.001, 0.005, 0.01, 0.02, 0.05, 0.1, 0.2, 0.3, 0.5] VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Mon Oct 18 15:54:13 UTC 2021] Executing as ?@80ffa154d6c0 on Linux 5.10.47-linuxkit amd64; OpenJDK 64-Bit Server VM 11.0.8+10-post-Debian-1deb10u1; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.23.6-SNAPSHOT\n",
      "INFO\t2021-10-18 15:54:13\tTheoreticalSensitivity\tCreating Roulette Wheel\n",
      "INFO\t2021-10-18 15:54:13\tTheoreticalSensitivity\tCalculating quality sums from quality sampler\n",
      "INFO\t2021-10-18 15:54:13\tTheoreticalSensitivity\t0 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:14\tTheoreticalSensitivity\t1000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:15\tTheoreticalSensitivity\t2000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:16\tTheoreticalSensitivity\t3000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:17\tTheoreticalSensitivity\t4000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:18\tTheoreticalSensitivity\t5000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:19\tTheoreticalSensitivity\t6000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:20\tTheoreticalSensitivity\t7000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:21\tTheoreticalSensitivity\t8000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:22\tTheoreticalSensitivity\t9000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:23\tTheoreticalSensitivity\tCalculating theoretical het sensitivity\n",
      "INFO\t2021-10-18 15:54:24\tTargetMetricsCollector\tCalculating GC metrics\n",
      "[Mon Oct 18 15:54:24 UTC 2021] picard.analysis.directed.CollectHsMetrics done. Elapsed time: 0.19 minutes.\n",
      "Runtime.totalMemory()=150994944\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_per_target_hs_metrics] Max memory used: 683MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_per_target_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_per_target_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow collect_detailed_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_detailed_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] starting step collect_insert_size_metrics\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_insert_size_metrics] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job collect_insert_size_metrics] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_insert_size_metrics] /private/tmp/docker_tmp6znkyd6s$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp6znkyd6s,target=/LnCocq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpsj7dmg3j,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp6rjhwo5g/final.bam,target=/var/lib/cwl/stg810ea892-b894-428a-9168-c172bf87fb4e/final.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpz1yvkp83/final.bam.bai,target=/var/lib/cwl/stg810ea892-b894-428a-9168-c172bf87fb4e/final.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpz1yvkp83/final.bai,target=/var/lib/cwl/stg810ea892-b894-428a-9168-c172bf87fb4e/final.bai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stga100c439-0ff3-467d-8c54-473194b007f8/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stga100c439-0ff3-467d-8c54-473194b007f8/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stga100c439-0ff3-467d-8c54-473194b007f8/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stga100c439-0ff3-467d-8c54-473194b007f8/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stga100c439-0ff3-467d-8c54-473194b007f8/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stga100c439-0ff3-467d-8c54-473194b007f8/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stga100c439-0ff3-467d-8c54-473194b007f8/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stga100c439-0ff3-467d-8c54-473194b007f8/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --workdir=/LnCocq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/LnCocq \\\n",
      "    --cidfile=/private/tmp/docker_tmppkta_i7q/20211018105424-526342.cid \\\n",
      "    broadinstitute/picard:2.23.6 \\\n",
      "    /usr/bin/java \\\n",
      "    -Xmx16g \\\n",
      "    -jar \\\n",
      "    /usr/picard/picard.jar \\\n",
      "    CollectInsertSizeMetrics \\\n",
      "    O= \\\n",
      "    /LnCocq/final.InsertSizeMetrics.txt \\\n",
      "    H= \\\n",
      "    /LnCocq/final.InsertSizeHistogram.pdf \\\n",
      "    I= \\\n",
      "    /var/lib/cwl/stg810ea892-b894-428a-9168-c172bf87fb4e/final.bam \\\n",
      "    METRIC_ACCUMULATION_LEVEL= \\\n",
      "    LIBRARY \\\n",
      "    REFERENCE_SEQUENCE= \\\n",
      "    /var/lib/cwl/stga100c439-0ff3-467d-8c54-473194b007f8/hla_and_brca_genes.fa\n",
      "INFO\t2021-10-18 15:54:25\tCollectInsertSizeMetrics\t\n",
      "\n",
      "********** NOTE: Picard's command line syntax is changing.\n",
      "**********\n",
      "********** For more information, please see:\n",
      "********** https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
      "**********\n",
      "********** The command line looks like this in the new syntax:\n",
      "**********\n",
      "**********    CollectInsertSizeMetrics -O  /LnCocq/final.InsertSizeMetrics.txt -H  /LnCocq/final.InsertSizeHistogram.pdf -I  /var/lib/cwl/stg810ea892-b894-428a-9168-c172bf87fb4e/final.bam -METRIC_ACCUMULATION_LEVEL  LIBRARY -REFERENCE_SEQUENCE  /var/lib/cwl/stga100c439-0ff3-467d-8c54-473194b007f8/hla_and_brca_genes.fa\n",
      "**********\n",
      "\n",
      "\n",
      "15:54:26.371 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/usr/picard/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[Mon Oct 18 15:54:26 UTC 2021] CollectInsertSizeMetrics HISTOGRAM_FILE=/LnCocq/final.InsertSizeHistogram.pdf METRIC_ACCUMULATION_LEVEL=[ALL_READS, LIBRARY] INPUT=/var/lib/cwl/stg810ea892-b894-428a-9168-c172bf87fb4e/final.bam OUTPUT=/LnCocq/final.InsertSizeMetrics.txt REFERENCE_SEQUENCE=/var/lib/cwl/stga100c439-0ff3-467d-8c54-473194b007f8/hla_and_brca_genes.fa    DEVIATIONS=10.0 MINIMUM_PCT=0.05 INCLUDE_DUPLICATES=false ASSUME_SORTED=true STOP_AFTER=0 VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Mon Oct 18 15:54:26 UTC 2021] Executing as ?@0a5d711f17da on Linux 5.10.47-linuxkit amd64; OpenJDK 64-Bit Server VM 11.0.8+10-post-Debian-1deb10u1; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.23.6-SNAPSHOT\n",
      "INFO\t2021-10-18 15:54:29\tRExecutor\tExecuting R script via command: Rscript /tmp/?/script290829268003851700.R /LnCocq/final.InsertSizeMetrics.txt /LnCocq/final.InsertSizeHistogram.pdf final.bam\n",
      "INFO\t2021-10-18 15:54:30\tProcessExecutor\tnull device \n",
      "INFO\t2021-10-18 15:54:30\tProcessExecutor\t          1 \n",
      "[Mon Oct 18 15:54:30 UTC 2021] picard.analysis.CollectInsertSizeMetrics done. Elapsed time: 0.07 minutes.\n",
      "Runtime.totalMemory()=264241152\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_insert_size_metrics] Max memory used: 951MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_insert_size_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_insert_size_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] starting step samtools_flagstat\n",
      "\u001b[1;30mINFO\u001b[0m [step samtools_flagstat] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job samtools_flagstat] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job samtools_flagstat] /private/tmp/docker_tmplncx3h_y$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmplncx3h_y,target=/LnCocq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpr5xky9ox,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp6rjhwo5g/final.bam,target=/var/lib/cwl/stg7fc29a57-b2d7-46f3-a528-30b12bdd2825/final.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpz1yvkp83/final.bam.bai,target=/var/lib/cwl/stg7fc29a57-b2d7-46f3-a528-30b12bdd2825/final.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpz1yvkp83/final.bai,target=/var/lib/cwl/stg7fc29a57-b2d7-46f3-a528-30b12bdd2825/final.bai,readonly \\\n",
      "    --workdir=/LnCocq \\\n",
      "    --read-only=true \\\n",
      "    --log-driver=none \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/LnCocq \\\n",
      "    --cidfile=/private/tmp/docker_tmpfcx5e7h1/20211018105430-738824.cid \\\n",
      "    quay.io/biocontainers/samtools:1.11--h6270b1f_0 \\\n",
      "    /usr/local/bin/samtools \\\n",
      "    flagstat \\\n",
      "    /var/lib/cwl/stg7fc29a57-b2d7-46f3-a528-30b12bdd2825/final.bam > /private/tmp/docker_tmplncx3h_y/final.bam.flagstat\n",
      "\u001b[1;30mINFO\u001b[0m [job samtools_flagstat] Max memory used: 0MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job samtools_flagstat] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step samtools_flagstat] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] starting step select_variants\n",
      "\u001b[1;30mINFO\u001b[0m [step select_variants] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job select_variants] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job select_variants] /private/tmp/docker_tmp8ln72nv2$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp8ln72nv2,target=/LnCocq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmplrlpwk3q,target=/tmp \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_target.interval_list,target=/var/lib/cwl/stg596059c4-5ab1-4c8d-8949-5ca9b108fb12/hla_and_brca_genes_target.interval_list,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stg9c8c9eb8-38cb-4f51-a958-f365d5db23d1/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stg9c8c9eb8-38cb-4f51-a958-f365d5db23d1/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stg9c8c9eb8-38cb-4f51-a958-f365d5db23d1/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stg9c8c9eb8-38cb-4f51-a958-f365d5db23d1/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stg9c8c9eb8-38cb-4f51-a958-f365d5db23d1/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stg9c8c9eb8-38cb-4f51-a958-f365d5db23d1/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stg9c8c9eb8-38cb-4f51-a958-f365d5db23d1/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stg9c8c9eb8-38cb-4f51-a958-f365d5db23d1/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_omni.vcf.gz,target=/var/lib/cwl/stg81289fbe-294c-436f-871c-f1cfde5a2a2f/hla_and_brca_genes_omni.vcf.gz,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_omni.vcf.gz.tbi,target=/var/lib/cwl/stg81289fbe-294c-436f-871c-f1cfde5a2a2f/hla_and_brca_genes_omni.vcf.gz.tbi,readonly \\\n",
      "    --workdir=/LnCocq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/LnCocq \\\n",
      "    --cidfile=/private/tmp/docker_tmp9izgt9x2/20211018105431-777149.cid \\\n",
      "    broadinstitute/gatk:4.1.8.1 \\\n",
      "    /gatk/gatk \\\n",
      "    --java-options \\\n",
      "    -Xmx4g \\\n",
      "    SelectVariants \\\n",
      "    -O \\\n",
      "    /LnCocq/select_variants.vcf.gz \\\n",
      "    -R \\\n",
      "    /var/lib/cwl/stg9c8c9eb8-38cb-4f51-a958-f365d5db23d1/hla_and_brca_genes.fa \\\n",
      "    --variant \\\n",
      "    /var/lib/cwl/stg81289fbe-294c-436f-871c-f1cfde5a2a2f/hla_and_brca_genes_omni.vcf.gz \\\n",
      "    -L \\\n",
      "    /var/lib/cwl/stg596059c4-5ab1-4c8d-8949-5ca9b108fb12/hla_and_brca_genes_target.interval_list\n",
      "15:54:36.659 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/gatk/gatk-package-4.1.8.1-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "Oct 18, 2021 3:54:37 PM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine\n",
      "INFO: Failed to detect whether we are running on Google Compute Engine.\n",
      "15:54:37.249 INFO  SelectVariants - ------------------------------------------------------------\n",
      "15:54:37.250 INFO  SelectVariants - The Genome Analysis Toolkit (GATK) v4.1.8.1\n",
      "15:54:37.251 INFO  SelectVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "15:54:37.251 INFO  SelectVariants - Executing as ?@64d0365691af on Linux v5.10.47-linuxkit amd64\n",
      "15:54:37.251 INFO  SelectVariants - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_242-8u242-b08-0ubuntu3~18.04-b08\n",
      "15:54:37.252 INFO  SelectVariants - Start Date/Time: October 18, 2021 3:54:36 PM GMT\n",
      "15:54:37.252 INFO  SelectVariants - ------------------------------------------------------------\n",
      "15:54:37.252 INFO  SelectVariants - ------------------------------------------------------------\n",
      "15:54:37.253 INFO  SelectVariants - HTSJDK Version: 2.23.0\n",
      "15:54:37.254 INFO  SelectVariants - Picard Version: 2.22.8\n",
      "15:54:37.254 INFO  SelectVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "15:54:37.254 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "15:54:37.254 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "15:54:37.254 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "15:54:37.255 INFO  SelectVariants - Deflater: IntelDeflater\n",
      "15:54:37.255 INFO  SelectVariants - Inflater: IntelInflater\n",
      "15:54:37.255 INFO  SelectVariants - GCS max retries/reopens: 20\n",
      "15:54:37.255 INFO  SelectVariants - Requester pays: disabled\n",
      "15:54:37.255 INFO  SelectVariants - Initializing engine\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15:54:37.695 INFO  FeatureManager - Using codec VCFCodec to read file file:///var/lib/cwl/stg81289fbe-294c-436f-871c-f1cfde5a2a2f/hla_and_brca_genes_omni.vcf.gz\n",
      "15:54:37.781 INFO  FeatureManager - Using codec IntervalListCodec to read file file:///var/lib/cwl/stg596059c4-5ab1-4c8d-8949-5ca9b108fb12/hla_and_brca_genes_target.interval_list\n",
      "15:54:37.813 INFO  IntervalArgumentCollection - Processing 10741 bp from intervals\n",
      "15:54:38.816 INFO  SelectVariants - Done initializing engine\n",
      "15:54:38.851 INFO  ProgressMeter - Starting traversal\n",
      "15:54:38.852 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "15:54:38.886 INFO  ProgressMeter -             unmapped              0.0                     2           3636.4\n",
      "15:54:38.886 INFO  ProgressMeter - Traversal complete. Processed 2 total variants in 0.0 minutes.\n",
      "15:54:38.925 INFO  SelectVariants - Shutting down engine\n",
      "[October 18, 2021 3:54:38 PM GMT] org.broadinstitute.hellbender.tools.walkers.variantutils.SelectVariants done. Elapsed time: 0.04 minutes.\n",
      "Runtime.totalMemory()=214958080\n",
      "Using GATK jar /gatk/gatk-package-4.1.8.1-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xmx4g -jar /gatk/gatk-package-4.1.8.1-local.jar SelectVariants -O /LnCocq/select_variants.vcf.gz -R /var/lib/cwl/stg9c8c9eb8-38cb-4f51-a958-f365d5db23d1/hla_and_brca_genes.fa --variant /var/lib/cwl/stg81289fbe-294c-436f-871c-f1cfde5a2a2f/hla_and_brca_genes_omni.vcf.gz -L /var/lib/cwl/stg596059c4-5ab1-4c8d-8949-5ca9b108fb12/hla_and_brca_genes_target.interval_list\n",
      "\u001b[1;30mINFO\u001b[0m [job select_variants] Max memory used: 966MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job select_variants] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step select_variants] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] starting step collect_alignment_summary_metrics\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_alignment_summary_metrics] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job collect_alignment_summary_metrics] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_alignment_summary_metrics] /private/tmp/docker_tmprjf_mcpr$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmprjf_mcpr,target=/LnCocq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpsdx_2r2j,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp6rjhwo5g/final.bam,target=/var/lib/cwl/stg1ddb1612-2073-45f3-9c7d-c9b7ead9341a/final.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpz1yvkp83/final.bam.bai,target=/var/lib/cwl/stg1ddb1612-2073-45f3-9c7d-c9b7ead9341a/final.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpz1yvkp83/final.bai,target=/var/lib/cwl/stg1ddb1612-2073-45f3-9c7d-c9b7ead9341a/final.bai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stg749f6bff-c591-4c5a-84f4-189ab5d2191f/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stg749f6bff-c591-4c5a-84f4-189ab5d2191f/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stg749f6bff-c591-4c5a-84f4-189ab5d2191f/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stg749f6bff-c591-4c5a-84f4-189ab5d2191f/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stg749f6bff-c591-4c5a-84f4-189ab5d2191f/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stg749f6bff-c591-4c5a-84f4-189ab5d2191f/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stg749f6bff-c591-4c5a-84f4-189ab5d2191f/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stg749f6bff-c591-4c5a-84f4-189ab5d2191f/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --workdir=/LnCocq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/LnCocq \\\n",
      "    --cidfile=/private/tmp/docker_tmp8xb8i0j7/20211018105439-242608.cid \\\n",
      "    broadinstitute/picard:2.23.6 \\\n",
      "    /usr/bin/java \\\n",
      "    -Xmx16g \\\n",
      "    -jar \\\n",
      "    /usr/picard/picard.jar \\\n",
      "    CollectAlignmentSummaryMetrics \\\n",
      "    OUTPUT= \\\n",
      "    /LnCocq/final.AlignmentSummaryMetrics.txt \\\n",
      "    INPUT= \\\n",
      "    /var/lib/cwl/stg1ddb1612-2073-45f3-9c7d-c9b7ead9341a/final.bam \\\n",
      "    METRIC_ACCUMULATION_LEVEL= \\\n",
      "    LIBRARY \\\n",
      "    REFERENCE_SEQUENCE= \\\n",
      "    /var/lib/cwl/stg749f6bff-c591-4c5a-84f4-189ab5d2191f/hla_and_brca_genes.fa\n",
      "INFO\t2021-10-18 15:54:40\tCollectAlignmentSummaryMetrics\t\n",
      "\n",
      "********** NOTE: Picard's command line syntax is changing.\n",
      "**********\n",
      "********** For more information, please see:\n",
      "********** https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
      "**********\n",
      "********** The command line looks like this in the new syntax:\n",
      "**********\n",
      "**********    CollectAlignmentSummaryMetrics -OUTPUT  /LnCocq/final.AlignmentSummaryMetrics.txt -INPUT  /var/lib/cwl/stg1ddb1612-2073-45f3-9c7d-c9b7ead9341a/final.bam -METRIC_ACCUMULATION_LEVEL  LIBRARY -REFERENCE_SEQUENCE  /var/lib/cwl/stg749f6bff-c591-4c5a-84f4-189ab5d2191f/hla_and_brca_genes.fa\n",
      "**********\n",
      "\n",
      "\n",
      "15:54:41.517 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/usr/picard/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Oct 18 15:54:41 UTC 2021] CollectAlignmentSummaryMetrics METRIC_ACCUMULATION_LEVEL=[ALL_READS, LIBRARY] INPUT=/var/lib/cwl/stg1ddb1612-2073-45f3-9c7d-c9b7ead9341a/final.bam OUTPUT=/LnCocq/final.AlignmentSummaryMetrics.txt REFERENCE_SEQUENCE=/var/lib/cwl/stg749f6bff-c591-4c5a-84f4-189ab5d2191f/hla_and_brca_genes.fa    MAX_INSERT_SIZE=100000 EXPECTED_PAIR_ORIENTATIONS=[FR] ADAPTER_SEQUENCE=[AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT, AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG, AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT, AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG, AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT, AGATCGGAAGAGCACACGTCTGAACTCCAGTCACNNNNNNNNATCTCGTATGCCGTCTTCTGCTTG] IS_BISULFITE_SEQUENCED=false COLLECT_ALIGNMENT_INFORMATION=true ASSUME_SORTED=true STOP_AFTER=0 VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Mon Oct 18 15:54:41 UTC 2021] Executing as ?@851475edbc94 on Linux 5.10.47-linuxkit amd64; OpenJDK 64-Bit Server VM 11.0.8+10-post-Debian-1deb10u1; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.23.6-SNAPSHOT\n",
      "[Mon Oct 18 15:54:45 UTC 2021] picard.analysis.CollectAlignmentSummaryMetrics done. Elapsed time: 0.07 minutes.\n",
      "Runtime.totalMemory()=264241152\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_alignment_summary_metrics] Max memory used: 1073MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_alignment_summary_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_alignment_summary_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] starting step verify_bam_id\n",
      "\u001b[1;30mINFO\u001b[0m [step verify_bam_id] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job verify_bam_id] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job verify_bam_id] /private/tmp/docker_tmpuqa61xoq$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpuqa61xoq,target=/LnCocq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp1gkenw6z,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp6rjhwo5g/final.bam,target=/var/lib/cwl/stg7fdae000-aeee-461a-a55d-61375a31c589/final.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpz1yvkp83/final.bam.bai,target=/var/lib/cwl/stg7fdae000-aeee-461a-a55d-61375a31c589/final.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpz1yvkp83/final.bai,target=/var/lib/cwl/stg7fdae000-aeee-461a-a55d-61375a31c589/final.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp8ln72nv2/select_variants.vcf.gz,target=/var/lib/cwl/stg1fb5f24e-aee9-45ff-b5aa-024c59a1dfa2/select_variants.vcf.gz,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp8ln72nv2/select_variants.vcf.gz.tbi,target=/var/lib/cwl/stg1fb5f24e-aee9-45ff-b5aa-024c59a1dfa2/select_variants.vcf.gz.tbi,readonly \\\n",
      "    --workdir=/LnCocq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/LnCocq \\\n",
      "    --cidfile=/private/tmp/docker_tmp5kv2zzcc/20211018105445-961439.cid \\\n",
      "    mgibio/verify_bam_id-cwl:1.1.3 \\\n",
      "    /usr/local/bin/verifyBamID \\\n",
      "    --out \\\n",
      "    /LnCocq/final.VerifyBamId \\\n",
      "    --bam \\\n",
      "    /var/lib/cwl/stg7fdae000-aeee-461a-a55d-61375a31c589/final.bam \\\n",
      "    --vcf \\\n",
      "    /var/lib/cwl/stg1fb5f24e-aee9-45ff-b5aa-024c59a1dfa2/select_variants.vcf.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Available Options\n",
      "                             Input Files : --vcf [/var/lib/cwl/stg1fb5f24e-aee9-45ff-b5aa-024c59a1dfa2/select_variants.vcf.gz],\n",
      "                                           --bam [/var/lib/cwl/stg7fdae000-aeee-461a-a55d-61375a31c589/final.bam],\n",
      "                                           --bai [], --subset [], --smID []\n",
      "                    VCF analysis options : --genoError [1.0e-03],\n",
      "                                           --minAF [0.01],\n",
      "                                           --minCallRate [0.50]\n",
      "   Individuals to compare with chip data : --site, --self, --best\n",
      "          Chip-free optimization options : --free-none, --free-mix [ON],\n",
      "                                           --free-refBias, --free-full\n",
      "          With-chip optimization options : --chip-none, --chip-mix [ON],\n",
      "                                           --chip-refBias, --chip-full\n",
      "                    BAM analysis options : --ignoreRG, --ignoreOverlapPair,\n",
      "                                           --noEOF, --precise, --minMapQ [10],\n",
      "                                           --maxDepth [20], --minQ [13],\n",
      "                                           --maxQ [40], --grid [0.05]\n",
      "                 Modeling Reference Bias : --refRef [1.00], --refHet [0.50],\n",
      "                                           --refAlt [0.00]\n",
      "                          Output options : --out [/LnCocq/final.VerifyBamId],\n",
      "                                           --verbose\n",
      "                               PhoneHome : --noPhoneHome,\n",
      "                                           --phoneHomeThinning [50]\n",
      "\n",
      "verifyBamID 1.1.3 -- verify identity and purity of sequence data\n",
      "(c) 2010-2014 Hyun Min Kang, Goo Jun, and Goncalo Abecasis\n",
      "\n",
      "\n",
      "\u0007WARNING - \n",
      "--self option was autotomatically turned on by default. Specify --best option if you wanted to check across all possible samples in the VCF\n",
      "\u001b[1;30mINFO\u001b[0m [job verify_bam_id] Max memory used: 0MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job verify_bam_id] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step verify_bam_id] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] starting step collect_roi_hs_metrics\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_roi_hs_metrics] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job collect_roi_hs_metrics] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_roi_hs_metrics] /private/tmp/docker_tmpbvooflw8$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpbvooflw8,target=/LnCocq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp5239qt98,target=/tmp \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_bait.interval_list,target=/var/lib/cwl/stg5d8f904b-1ed1-4eb2-a5d6-ebc4027a4c9e/hla_and_brca_genes_bait.interval_list,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp6rjhwo5g/final.bam,target=/var/lib/cwl/stg5c22f98d-70b9-4603-9d8e-d3317648578a/final.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpz1yvkp83/final.bam.bai,target=/var/lib/cwl/stg5c22f98d-70b9-4603-9d8e-d3317648578a/final.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpz1yvkp83/final.bai,target=/var/lib/cwl/stg5c22f98d-70b9-4603-9d8e-d3317648578a/final.bai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stg6e95a08a-75fc-4a9f-a591-4eabf54e1ad7/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stg6e95a08a-75fc-4a9f-a591-4eabf54e1ad7/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stg6e95a08a-75fc-4a9f-a591-4eabf54e1ad7/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stg6e95a08a-75fc-4a9f-a591-4eabf54e1ad7/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stg6e95a08a-75fc-4a9f-a591-4eabf54e1ad7/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stg6e95a08a-75fc-4a9f-a591-4eabf54e1ad7/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stg6e95a08a-75fc-4a9f-a591-4eabf54e1ad7/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stg6e95a08a-75fc-4a9f-a591-4eabf54e1ad7/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_target.interval_list,target=/var/lib/cwl/stgcb10bd80-d3f3-4e49-845b-0dd4c299d47b/hla_and_brca_genes_target.interval_list,readonly \\\n",
      "    --workdir=/LnCocq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/LnCocq \\\n",
      "    --cidfile=/private/tmp/docker_tmp47g87kkk/20211018105447-048216.cid \\\n",
      "    broadinstitute/picard:2.23.6 \\\n",
      "    /usr/bin/java \\\n",
      "    -Xmx48g \\\n",
      "    -jar \\\n",
      "    /usr/picard/picard.jar \\\n",
      "    CollectHsMetrics \\\n",
      "    O= \\\n",
      "    /LnCocq/final.roi-HsMetrics.txt \\\n",
      "    BAIT_INTERVALS= \\\n",
      "    /var/lib/cwl/stg5d8f904b-1ed1-4eb2-a5d6-ebc4027a4c9e/hla_and_brca_genes_bait.interval_list \\\n",
      "    I= \\\n",
      "    /var/lib/cwl/stg5c22f98d-70b9-4603-9d8e-d3317648578a/final.bam \\\n",
      "    METRIC_ACCUMULATION_LEVEL= \\\n",
      "    ALL_READS \\\n",
      "    MINIMUM_BASE_QUALITY= \\\n",
      "    1 \\\n",
      "    MINIMUM_MAPPING_QUALITY= \\\n",
      "    1 \\\n",
      "    R= \\\n",
      "    /var/lib/cwl/stg6e95a08a-75fc-4a9f-a591-4eabf54e1ad7/hla_and_brca_genes.fa \\\n",
      "    TARGET_INTERVALS= \\\n",
      "    /var/lib/cwl/stgcb10bd80-d3f3-4e49-845b-0dd4c299d47b/hla_and_brca_genes_target.interval_list\n",
      "INFO\t2021-10-18 15:54:48\tCollectHsMetrics\t\n",
      "\n",
      "********** NOTE: Picard's command line syntax is changing.\n",
      "**********\n",
      "********** For more information, please see:\n",
      "********** https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
      "**********\n",
      "********** The command line looks like this in the new syntax:\n",
      "**********\n",
      "**********    CollectHsMetrics -O  /LnCocq/final.roi-HsMetrics.txt -BAIT_INTERVALS  /var/lib/cwl/stg5d8f904b-1ed1-4eb2-a5d6-ebc4027a4c9e/hla_and_brca_genes_bait.interval_list -I  /var/lib/cwl/stg5c22f98d-70b9-4603-9d8e-d3317648578a/final.bam -METRIC_ACCUMULATION_LEVEL  ALL_READS -MINIMUM_BASE_QUALITY  1 -MINIMUM_MAPPING_QUALITY  1 -R  /var/lib/cwl/stg6e95a08a-75fc-4a9f-a591-4eabf54e1ad7/hla_and_brca_genes.fa -TARGET_INTERVALS  /var/lib/cwl/stgcb10bd80-d3f3-4e49-845b-0dd4c299d47b/hla_and_brca_genes_target.interval_list\n",
      "**********\n",
      "\n",
      "\n",
      "15:54:49.163 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/usr/picard/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Oct 18 15:54:49 UTC 2021] CollectHsMetrics BAIT_INTERVALS=[/var/lib/cwl/stg5d8f904b-1ed1-4eb2-a5d6-ebc4027a4c9e/hla_and_brca_genes_bait.interval_list] TARGET_INTERVALS=[/var/lib/cwl/stgcb10bd80-d3f3-4e49-845b-0dd4c299d47b/hla_and_brca_genes_target.interval_list] INPUT=/var/lib/cwl/stg5c22f98d-70b9-4603-9d8e-d3317648578a/final.bam OUTPUT=/LnCocq/final.roi-HsMetrics.txt METRIC_ACCUMULATION_LEVEL=[ALL_READS] MINIMUM_MAPPING_QUALITY=1 MINIMUM_BASE_QUALITY=1 REFERENCE_SEQUENCE=/var/lib/cwl/stg6e95a08a-75fc-4a9f-a591-4eabf54e1ad7/hla_and_brca_genes.fa    NEAR_DISTANCE=250 CLIP_OVERLAPPING_READS=true INCLUDE_INDELS=false COVERAGE_CAP=200 SAMPLE_SIZE=10000 ALLELE_FRACTION=[0.001, 0.005, 0.01, 0.02, 0.05, 0.1, 0.2, 0.3, 0.5] VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Mon Oct 18 15:54:49 UTC 2021] Executing as ?@5b14426e02a8 on Linux 5.10.47-linuxkit amd64; OpenJDK 64-Bit Server VM 11.0.8+10-post-Debian-1deb10u1; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.23.6-SNAPSHOT\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO\t2021-10-18 15:54:49\tTheoreticalSensitivity\tCreating Roulette Wheel\n",
      "INFO\t2021-10-18 15:54:49\tTheoreticalSensitivity\tCalculating quality sums from quality sampler\n",
      "INFO\t2021-10-18 15:54:49\tTheoreticalSensitivity\t0 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:50\tTheoreticalSensitivity\t1000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:51\tTheoreticalSensitivity\t2000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:52\tTheoreticalSensitivity\t3000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:53\tTheoreticalSensitivity\t4000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:54\tTheoreticalSensitivity\t5000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:55\tTheoreticalSensitivity\t6000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:56\tTheoreticalSensitivity\t7000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:57\tTheoreticalSensitivity\t8000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:58\tTheoreticalSensitivity\t9000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:54:59\tTheoreticalSensitivity\tCalculating theoretical het sensitivity\n",
      "INFO\t2021-10-18 15:55:00\tTargetMetricsCollector\tCalculating GC metrics\n",
      "[Mon Oct 18 15:55:00 UTC 2021] picard.analysis.directed.CollectHsMetrics done. Elapsed time: 0.18 minutes.\n",
      "Runtime.totalMemory()=159383552\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_roi_hs_metrics] Max memory used: 717MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_roi_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_roi_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step qc] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow ] completed success\n",
      "{\n",
      "    \"alignment_summary_metrics\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.AlignmentSummaryMetrics.txt\",\n",
      "        \"basename\": \"final.AlignmentSummaryMetrics.txt\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$d0daa23475ab3f9fabbc6f5408555d3ecbf0f4a4\",\n",
      "        \"size\": 3747,\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.AlignmentSummaryMetrics.txt\"\n",
      "    },\n",
      "    \"bam\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.bam\",\n",
      "        \"basename\": \"final.bam\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$487c9aac5e7420a3a4cb2a201ddc42d63f2e6f3e\",\n",
      "        \"size\": 642139,\n",
      "        \"secondaryFiles\": [\n",
      "            {\n",
      "                \"basename\": \"final.bam.bai\",\n",
      "                \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.bam.bai\",\n",
      "                \"class\": \"File\",\n",
      "                \"checksum\": \"sha1$b4fb874a28557ab7414fe727649761b0db0ee13a\",\n",
      "                \"size\": 36752,\n",
      "                \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.bam.bai\"\n",
      "            },\n",
      "            {\n",
      "                \"basename\": \"final.bai\",\n",
      "                \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.bai\",\n",
      "                \"class\": \"File\",\n",
      "                \"checksum\": \"sha1$b4fb874a28557ab7414fe727649761b0db0ee13a\",\n",
      "                \"size\": 36752,\n",
      "                \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.bai\"\n",
      "            }\n",
      "        ],\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.bam\"\n",
      "    },\n",
      "    \"flagstats\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.bam.flagstat\",\n",
      "        \"basename\": \"final.bam.flagstat\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$56ae2958678dc05035ea9819cd33b5a96c6a7243\",\n",
      "        \"size\": 402,\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.bam.flagstat\"\n",
      "    },\n",
      "    \"hs_metrics\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.roi-HsMetrics.txt\",\n",
      "        \"basename\": \"final.roi-HsMetrics.txt\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$ec20430f2a77ac4a72b67dcd7c88bebda9c9f2dd\",\n",
      "        \"size\": 4313,\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.roi-HsMetrics.txt\"\n",
      "    },\n",
      "    \"insert_size_histogram\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.InsertSizeHistogram.pdf\",\n",
      "        \"basename\": \"final.InsertSizeHistogram.pdf\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$4208ca8332041549d22b0f871c74dc3df5521abf\",\n",
      "        \"size\": 15046,\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.InsertSizeHistogram.pdf\"\n",
      "    },\n",
      "    \"insert_size_metrics\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.InsertSizeMetrics.txt\",\n",
      "        \"basename\": \"final.InsertSizeMetrics.txt\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$381fb2920c03102ceeccf199637cdf601e8a4a80\",\n",
      "        \"size\": 4676,\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.InsertSizeMetrics.txt\"\n",
      "    },\n",
      "    \"mark_duplicates_metrics\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.NameSorted.mark_dups_metrics.txt\",\n",
      "        \"basename\": \"final.NameSorted.mark_dups_metrics.txt\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$e48e739811a0e396edca3b277a74d71e507ac1c4\",\n",
      "        \"size\": 2885,\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.NameSorted.mark_dups_metrics.txt\"\n",
      "    },\n",
      "    \"per_base_coverage_metrics\": [\n",
      "        {\n",
      "            \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.base-clinvar-PerBaseCoverage.txt\",\n",
      "            \"basename\": \"final.base-clinvar-PerBaseCoverage.txt\",\n",
      "            \"class\": \"File\",\n",
      "            \"checksum\": \"sha1$daef4cd565eb9efa7da0de56a1988ecd84838f48\",\n",
      "            \"size\": 264508,\n",
      "            \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.base-clinvar-PerBaseCoverage.txt\"\n",
      "        }\n",
      "    ],\n",
      "    \"per_base_hs_metrics\": [\n",
      "        {\n",
      "            \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.base-clinvar-HsMetrics.txt\",\n",
      "            \"basename\": \"final.base-clinvar-HsMetrics.txt\",\n",
      "            \"class\": \"File\",\n",
      "            \"checksum\": \"sha1$1ec4477823456c270533add83bf60ae18fc44f94\",\n",
      "            \"size\": 4375,\n",
      "            \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.base-clinvar-HsMetrics.txt\"\n",
      "        }\n",
      "    ],\n",
      "    \"per_target_coverage_metrics\": [\n",
      "        {\n",
      "            \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.target-acmg_genes-PerTargetCoverage.txt\",\n",
      "            \"basename\": \"final.target-acmg_genes-PerTargetCoverage.txt\",\n",
      "            \"class\": \"File\",\n",
      "            \"checksum\": \"sha1$a9e164ec1416b8625ccb15c6cc91922602323856\",\n",
      "            \"size\": 4937,\n",
      "            \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.target-acmg_genes-PerTargetCoverage.txt\"\n",
      "        }\n",
      "    ],\n",
      "    \"per_target_hs_metrics\": [\n",
      "        {\n",
      "            \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.target-acmg_genes-HsMetrics.txt\",\n",
      "            \"basename\": \"final.target-acmg_genes-HsMetrics.txt\",\n",
      "            \"class\": \"File\",\n",
      "            \"checksum\": \"sha1$c36d2075038e3056596c8abf7f55a36764e6e0e9\",\n",
      "            \"size\": 4389,\n",
      "            \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.target-acmg_genes-HsMetrics.txt\"\n",
      "        }\n",
      "    ],\n",
      "    \"summary_hs_metrics\": [],\n",
      "    \"verify_bam_id_depth\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.VerifyBamId.depthSM\",\n",
      "        \"basename\": \"final.VerifyBamId.depthSM\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$4121abf4f28a3b184b38098d89502f4233106fee\",\n",
      "        \"size\": 544,\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.VerifyBamId.depthSM\"\n",
      "    },\n",
      "    \"verify_bam_id_metrics\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.VerifyBamId.selfSM\",\n",
      "        \"basename\": \"final.VerifyBamId.selfSM\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$f02f2c93da76b252a96ba4188f2caefb4eb297b4\",\n",
      "        \"size\": 229,\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/normal/final.VerifyBamId.selfSM\"\n",
      "    }\n",
      "}\n",
      "\u001b[1;30mINFO\u001b[0m Final process status is success\n"
     ]
    }
   ],
   "source": [
    "!cwltool --outdir normal analysis-workflows/definitions/pipelines/alignment_exome.cwl normal_inputs.yaml"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4f9b0b95",
   "metadata": {},
   "source": [
    "## Tumor Alignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "a6ed7e4e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1;30mINFO\u001b[0m /Users/jwalker/opt/miniconda3/bin/cwltool 3.0.20210124104916\n",
      "\u001b[1;30mINFO\u001b[0m Resolved 'analysis-workflows/definitions/pipelines/alignment_exome.cwl' to 'file:///Users/jwalker/git/bfx_workshop/lectures/week_05/analysis-workflows/definitions/pipelines/alignment_exome.cwl'\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33mWorkflow checker warning:\n",
      "analysis-workflows/definitions/subworkflows/sequence_to_bqsr.cwl:23:9:  Source 'bqsr_intervals' of\n",
      "                                                                        type [\"null\", {\"type\":\n",
      "                                                                        \"array\", \"items\":\n",
      "                                                                        \"string\"}] may be\n",
      "                                                                        incompatible\n",
      "analysis-workflows/definitions/subworkflows/sequence_to_bqsr.cwl:79:13:   with sink 'intervals'\n",
      "                                                                          of type {\"type\": \"array\",\n",
      "                                                                          \"items\": \"string\",\n",
      "                                                                          \"inputBinding\":\n",
      "                                                                          {\"prefix\": \"-L\"}}\u001b[0m\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33mWorkflow checker warning:\n",
      "analysis-workflows/definitions/subworkflows/hs_metrics.cwl:100:26: Source\n",
      "                                                                   'per_base_coverage_metrics' of\n",
      "                                                                   type {\"type\": \"array\", \"items\":\n",
      "                                                                   [\"null\", \"File\"]} may be\n",
      "                                                                   incompatible\n",
      "analysis-workflows/definitions/subworkflows/hs_metrics.cwl:34:9:     with sink\n",
      "                                                                     'per_base_coverage_metrics' of\n",
      "                                                                     type {\"type\": \"array\",\n",
      "                                                                     \"items\": \"File\"}\n",
      "analysis-workflows/definitions/subworkflows/hs_metrics.cwl:126:26: Source\n",
      "                                                                   'per_target_coverage_metrics' of\n",
      "                                                                   type {\"type\": \"array\", \"items\":\n",
      "                                                                   [\"null\", \"File\"]} may be\n",
      "                                                                   incompatible\n",
      "analysis-workflows/definitions/subworkflows/hs_metrics.cwl:40:9:     with sink\n",
      "                                                                     'per_target_coverage_metrics'\n",
      "                                                                     of type {\"type\": \"array\",\n",
      "                                                                     \"items\": \"File\"}\u001b[0m\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33mWorkflow checker warning:\n",
      "analysis-workflows/definitions/pipelines/alignment_exome.cwl:38:9:   Source 'final_name' of type\n",
      "                                                                     [\"null\", \"string\"] may be\n",
      "                                                                     incompatible\n",
      "analysis-workflows/definitions/pipelines/alignment_exome.cwl:109:13:   with sink 'final_name' of\n",
      "                                                                       type \"string\"\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [workflow ] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow ] starting step alignment\n",
      "\u001b[1;30mINFO\u001b[0m [step alignment] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] starting step align\n",
      "\u001b[1;30mINFO\u001b[0m [step align] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow align] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow align] starting step align_and_tag\n",
      "\u001b[1;30mINFO\u001b[0m [step align_and_tag] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job align_and_tag] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job align_and_tag] /private/tmp/docker_tmp04wmjmrr$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp04wmjmrr,target=/SVAvuq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp4_tmfgme,target=/tmp \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/2895499223.bam,target=/var/lib/cwl/stg1e58f3bc-d307-45ba-a4d0-9c45bb7c857a/2895499223.bam,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stg5a1dc648-9a8d-4b3d-a315-eba536f9bd10/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stg5a1dc648-9a8d-4b3d-a315-eba536f9bd10/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stg5a1dc648-9a8d-4b3d-a315-eba536f9bd10/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stg5a1dc648-9a8d-4b3d-a315-eba536f9bd10/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stg5a1dc648-9a8d-4b3d-a315-eba536f9bd10/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stg5a1dc648-9a8d-4b3d-a315-eba536f9bd10/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stg5a1dc648-9a8d-4b3d-a315-eba536f9bd10/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stg5a1dc648-9a8d-4b3d-a315-eba536f9bd10/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --workdir=/SVAvuq \\\n",
      "    --read-only=true \\\n",
      "    --log-driver=none \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/SVAvuq \\\n",
      "    --cidfile=/private/tmp/docker_tmpnme30rt1/20211018105652-913570.cid \\\n",
      "    mgibio/alignment_helper-cwl:1.1.0 \\\n",
      "    /bin/bash \\\n",
      "    sequence_alignment_helper.sh \\\n",
      "    -b \\\n",
      "    /var/lib/cwl/stg1e58f3bc-d307-45ba-a4d0-9c45bb7c857a/2895499223.bam \\\n",
      "    -g \\\n",
      "    '@RG\tID:2895499223\tPU:H7HY2CCXX.3\tSM:H_NJ-HCC1395-HCC1395\tLB:H_NJ-HCC1395-HCC1395-lg24-lib1\tPL:Illumina\tCN:WUGSC' \\\n",
      "    -r \\\n",
      "    /var/lib/cwl/stg5a1dc648-9a8d-4b3d-a315-eba536f9bd10/hla_and_brca_genes.fa \\\n",
      "    -n \\\n",
      "    8 > /private/tmp/docker_tmp04wmjmrr/refAlign.bam\n",
      "samblaster: Version 0.1.24\n",
      "samblaster: Inputting from stdin\n",
      "samblaster: Outputting to stdout\n",
      "15:56:55.782 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/opt/picard-2.18.1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Oct 18 15:56:56 UTC 2021] SamToFastq INPUT=/var/lib/cwl/stg1e58f3bc-d307-45ba-a4d0-9c45bb7c857a/2895499223.bam FASTQ=/dev/stdout INTERLEAVE=true INCLUDE_NON_PF_READS=true    OUTPUT_PER_RG=false COMPRESS_OUTPUTS_PER_RG=false RG_TAG=PU RE_REVERSE=true CLIPPING_MIN_LENGTH=0 READ1_TRIM=0 READ2_TRIM=0 INCLUDE_NON_PRIMARY_ALIGNMENTS=false VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Mon Oct 18 15:56:56 UTC 2021] Executing as ?@8b703157d83a on Linux 5.10.47-linuxkit amd64; OpenJDK 64-Bit Server VM 1.8.0_252-8u252-b09-1~16.04-b09; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.18.1-SNAPSHOT\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[Mon Oct 18 15:56:56 UTC 2021] picard.sam.SamToFastq done. Elapsed time: 0.02 minutes.\n",
      "Runtime.totalMemory()=78643200\n",
      "[M::process] read 4802 sequences (725102 bp)...\n",
      "[M::process] 0 single-end sequences; 4802 paired-end sequences\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 2144, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (165, 209, 260)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 450)\n",
      "[M::mem_pestat] mean and std.dev: (216.35, 67.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 545)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_process_seqs] Processed 4802 reads in 1.558 CPU sec, 0.204 real sec\n",
      "samblaster: Loaded 2 header sequence entries.\n",
      "[main] Version: 0.7.15-r1140\n",
      "[main] CMD: /usr/local/bin/bwa mem -K 100000000 -t 8 -Y -p -R @RG\tID:2895499223\tPU:H7HY2CCXX.3\tSM:H_NJ-HCC1395-HCC1395\tLB:H_NJ-HCC1395-HCC1395-lg24-lib1\tPL:Illumina\tCN:WUGSC /var/lib/cwl/stg5a1dc648-9a8d-4b3d-a315-eba536f9bd10/hla_and_brca_genes.fa /dev/stdin\n",
      "[main] Real time: 3.707 sec; CPU: 2.132 sec\n",
      "samblaster: Marked 0 of 2401 (0.00%) read ids as duplicates using 2008k memory in 0.005S CPU seconds and 4S wall time.\n",
      "\u001b[1;30mINFO\u001b[0m [job align_and_tag] Max memory used: 1356MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job align_and_tag] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step align_and_tag] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow align] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step align] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow align_2] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow align_2] starting step align_and_tag_2\n",
      "\u001b[1;30mINFO\u001b[0m [step align_and_tag_2] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job align_and_tag_2] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job align_and_tag_2] /private/tmp/docker_tmppi5zoakt$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmppi5zoakt,target=/SVAvuq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpvdcej0yu,target=/tmp \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/2895499237.bam,target=/var/lib/cwl/stg5b11ff25-bf79-43d5-a9d8-189898c49c50/2895499237.bam,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stg7b77819e-a16d-4456-977a-c2b7fa6cd4a4/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stg7b77819e-a16d-4456-977a-c2b7fa6cd4a4/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stg7b77819e-a16d-4456-977a-c2b7fa6cd4a4/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stg7b77819e-a16d-4456-977a-c2b7fa6cd4a4/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stg7b77819e-a16d-4456-977a-c2b7fa6cd4a4/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stg7b77819e-a16d-4456-977a-c2b7fa6cd4a4/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stg7b77819e-a16d-4456-977a-c2b7fa6cd4a4/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stg7b77819e-a16d-4456-977a-c2b7fa6cd4a4/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --workdir=/SVAvuq \\\n",
      "    --read-only=true \\\n",
      "    --log-driver=none \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/SVAvuq \\\n",
      "    --cidfile=/private/tmp/docker_tmpl9v96jwi/20211018105657-581740.cid \\\n",
      "    mgibio/alignment_helper-cwl:1.1.0 \\\n",
      "    /bin/bash \\\n",
      "    sequence_alignment_helper.sh \\\n",
      "    -b \\\n",
      "    /var/lib/cwl/stg5b11ff25-bf79-43d5-a9d8-189898c49c50/2895499237.bam \\\n",
      "    -g \\\n",
      "    '@RG\tID:2895499237\tPU:H7HY2CCXX.4\tSM:H_NJ-HCC1395-HCC1395\tLB:H_NJ-HCC1395-HCC1395-lg24-lib1\tPL:Illumina\tCN:WUGSC' \\\n",
      "    -r \\\n",
      "    /var/lib/cwl/stg7b77819e-a16d-4456-977a-c2b7fa6cd4a4/hla_and_brca_genes.fa \\\n",
      "    -n \\\n",
      "    8 > /private/tmp/docker_tmppi5zoakt/refAlign.bam\n",
      "samblaster: Version 0.1.24\n",
      "samblaster: Inputting from stdin\n",
      "samblaster: Outputting to stdout\n",
      "15:56:59.946 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/opt/picard-2.18.1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Oct 18 15:57:00 UTC 2021] SamToFastq INPUT=/var/lib/cwl/stg5b11ff25-bf79-43d5-a9d8-189898c49c50/2895499237.bam FASTQ=/dev/stdout INTERLEAVE=true INCLUDE_NON_PF_READS=true    OUTPUT_PER_RG=false COMPRESS_OUTPUTS_PER_RG=false RG_TAG=PU RE_REVERSE=true CLIPPING_MIN_LENGTH=0 READ1_TRIM=0 READ2_TRIM=0 INCLUDE_NON_PRIMARY_ALIGNMENTS=false VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Mon Oct 18 15:57:00 UTC 2021] Executing as ?@8959d62c0061 on Linux 5.10.47-linuxkit amd64; OpenJDK 64-Bit Server VM 1.8.0_252-8u252-b09-1~16.04-b09; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.18.1-SNAPSHOT\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[Mon Oct 18 15:57:01 UTC 2021] picard.sam.SamToFastq done. Elapsed time: 0.02 minutes.\n",
      "Runtime.totalMemory()=78643200\n",
      "[M::process] read 4772 sequences (720572 bp)...\n",
      "[M::process] 0 single-end sequences; 4772 paired-end sequences\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 2137, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (166, 208, 262)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 454)\n",
      "[M::mem_pestat] mean and std.dev: (217.95, 69.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 550)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 4772 reads in 0.864 CPU sec, 0.114 real sec\n",
      "samblaster: Loaded 2 header sequence entries.\n",
      "[main] Version: 0.7.15-r1140\n",
      "[main] CMD: /usr/local/bin/bwa mem -K 100000000 -t 8 -Y -p -R @RG\tID:2895499237\tPU:H7HY2CCXX.4\tSM:H_NJ-HCC1395-HCC1395\tLB:H_NJ-HCC1395-HCC1395-lg24-lib1\tPL:Illumina\tCN:WUGSC /var/lib/cwl/stg7b77819e-a16d-4456-977a-c2b7fa6cd4a4/hla_and_brca_genes.fa /dev/stdin\n",
      "[main] Real time: 3.154 sec; CPU: 1.367 sec\n",
      "samblaster: Marked 0 of 2386 (0.00%) read ids as duplicates using 1768k memory in 0.005S CPU seconds and 3S wall time.\n",
      "\u001b[1;30mINFO\u001b[0m [job align_and_tag_2] Max memory used: 1471MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job align_and_tag_2] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step align_and_tag_2] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow align_2] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step align] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] starting step merge\n",
      "\u001b[1;30mINFO\u001b[0m [step merge] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job merge] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job merge] /private/tmp/docker_tmp2ccon_0e$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp2ccon_0e,target=/SVAvuq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpgl0239k2,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp04wmjmrr/refAlign.bam,target=/var/lib/cwl/stg3f2d5191-15fd-4da1-b15d-50b93ca6bbc5/refAlign.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmppi5zoakt/refAlign.bam,target=/var/lib/cwl/stgc705fea5-592b-46d0-bc9c-c5989c74cefa/refAlign.bam,readonly \\\n",
      "    --workdir=/SVAvuq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/SVAvuq \\\n",
      "    --cidfile=/private/tmp/docker_tmpsr7vbx6o/20211018105701-950198.cid \\\n",
      "    mgibio/bam-merge:0.1 \\\n",
      "    /usr/bin/perl \\\n",
      "    merge.pl \\\n",
      "    -t \\\n",
      "    4 \\\n",
      "    -n \\\n",
      "    final \\\n",
      "    /var/lib/cwl/stg3f2d5191-15fd-4da1-b15d-50b93ca6bbc5/refAlign.bam \\\n",
      "    /var/lib/cwl/stgc705fea5-592b-46d0-bc9c-c5989c74cefa/refAlign.bam\n",
      "15:57:04.339 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/opt/picard-2.18.1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[Mon Oct 18 15:57:04 UTC 2021] MergeSamFiles INPUT=[/var/lib/cwl/stg3f2d5191-15fd-4da1-b15d-50b93ca6bbc5/refAlign.bam, /var/lib/cwl/stgc705fea5-592b-46d0-bc9c-c5989c74cefa/refAlign.bam] OUTPUT=final SORT_ORDER=unsorted ASSUME_SORTED=true USE_THREADING=true VALIDATION_STRINGENCY=LENIENT    MERGE_SEQUENCE_DICTIONARIES=false VERBOSITY=INFO QUIET=false COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Mon Oct 18 15:57:04 UTC 2021] Executing as ?@aa46fd5ff0cf on Linux 5.10.47-linuxkit amd64; OpenJDK 64-Bit Server VM 1.8.0_181-8u181-b13-0ubuntu0.16.04.1-b13; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.18.1-SNAPSHOT\n",
      "Ignoring SAM validation error: ERROR: File /var/lib/cwl/stg3f2d5191-15fd-4da1-b15d-50b93ca6bbc5/refAlign.bam, Error parsing SAM header. Problem parsing @PG key:value pair ID:2895499223 clashes with ID:bwa. Line:\n",
      "@PG\tID:bwa\tPN:bwa\tVN:0.7.15-r1140\tCL:/usr/local/bin/bwa mem -K 100000000 -t 8 -Y -p -R @RG\tID:2895499223\tPU:H7HY2CCXX.3\tSM:H_NJ-HCC1395-HCC1395\tLB:H_NJ-HCC1395-HCC1395-lg24-lib1\tPL:Illumina\tCN:WUGSC /var/lib/cwl/stg5a1dc648-9a8d-4b3d-a315-eba536f9bd10/hla_and_brca_genes.fa /dev/stdin\n",
      "Ignoring SAM validation error: ERROR: File /var/lib/cwl/stgc705fea5-592b-46d0-bc9c-c5989c74cefa/refAlign.bam, Error parsing SAM header. Problem parsing @PG key:value pair ID:2895499237 clashes with ID:bwa. Line:\n",
      "@PG\tID:bwa\tPN:bwa\tVN:0.7.15-r1140\tCL:/usr/local/bin/bwa mem -K 100000000 -t 8 -Y -p -R @RG\tID:2895499237\tPU:H7HY2CCXX.4\tSM:H_NJ-HCC1395-HCC1395\tLB:H_NJ-HCC1395-HCC1395-lg24-lib1\tPL:Illumina\tCN:WUGSC /var/lib/cwl/stg7b77819e-a16d-4456-977a-c2b7fa6cd4a4/hla_and_brca_genes.fa /dev/stdin\n",
      "INFO\t2021-10-18 15:57:04\tMergeSamFiles\tInput files are in same order as output so sorting to temp directory is not needed.\n",
      "INFO\t2021-10-18 15:57:05\tMergeSamFiles\tFinished reading inputs.\n",
      "[Mon Oct 18 15:57:05 UTC 2021] picard.sam.MergeSamFiles done. Elapsed time: 0.02 minutes.\n",
      "Runtime.totalMemory()=78643200\n",
      "\u001b[1;30mINFO\u001b[0m [job merge] Max memory used: 211MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job merge] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step merge] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] starting step name_sort\n",
      "\u001b[1;30mINFO\u001b[0m [step name_sort] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job name_sort] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job name_sort] /private/tmp/docker_tmpesu0os_x$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpesu0os_x,target=/SVAvuq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpwo3qma7o,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp2ccon_0e/final,target=/var/lib/cwl/stg9ece422c-1f64-42b7-b79d-a63dce673011/final,readonly \\\n",
      "    --workdir=/SVAvuq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/SVAvuq \\\n",
      "    --cidfile=/private/tmp/docker_tmpv4yfnfb9/20211018105705-765868.cid \\\n",
      "    mgibio/sambamba-cwl:0.6.4 \\\n",
      "    /usr/bin/sambamba \\\n",
      "    sort \\\n",
      "    -t \\\n",
      "    8 \\\n",
      "    -m \\\n",
      "    22G \\\n",
      "    -n \\\n",
      "    -o \\\n",
      "    final.NameSorted.bam \\\n",
      "    /var/lib/cwl/stg9ece422c-1f64-42b7-b79d-a63dce673011/final\n",
      "\u001b[1;30mINFO\u001b[0m [job name_sort] Max memory used: 0MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job name_sort] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step name_sort] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] starting step mark_duplicates_and_sort\n",
      "\u001b[1;30mINFO\u001b[0m [step mark_duplicates_and_sort] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job mark_duplicates_and_sort] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job mark_duplicates_and_sort] /private/tmp/docker_tmpukvdlk98$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpukvdlk98,target=/SVAvuq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpcf2y11q9,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpesu0os_x/final.NameSorted.bam,target=/var/lib/cwl/stg93dbdec8-aa49-4a52-a035-29761c40e604/final.NameSorted.bam,readonly \\\n",
      "    --workdir=/SVAvuq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/SVAvuq \\\n",
      "    --cidfile=/private/tmp/docker_tmpnpvarp9d/20211018105707-019771.cid \\\n",
      "    mgibio/mark_duplicates-cwl:1.0.1 \\\n",
      "    /bin/bash \\\n",
      "    markduplicates_helper.sh \\\n",
      "    /var/lib/cwl/stg93dbdec8-aa49-4a52-a035-29761c40e604/final.NameSorted.bam \\\n",
      "    8 \\\n",
      "    MarkedSorted.bam \\\n",
      "    final.NameSorted.mark_dups_metrics.txt \\\n",
      "    queryname\n",
      "15:57:09.448 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/opt/picard-2.18.1/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "INFO\t2021-10-18 15:57:09\tMarkDuplicates\tStart of doWork freeMemory: 72612960; totalMemory: 78643200; maxMemory: 15271460864\n",
      "INFO\t2021-10-18 15:57:09\tMarkDuplicates\tReading input file and constructing read end information.\n",
      "INFO\t2021-10-18 15:57:09\tMarkDuplicates\tWill retain up to 55331379 data points before spilling to disk.\n",
      "INFO\t2021-10-18 15:57:12\tMarkDuplicates\tRead 9576 records. 0 pairs never matched.\n",
      "INFO\t2021-10-18 15:57:13\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 95062416; totalMemory: 543162368; maxMemory: 15271460864\n",
      "INFO\t2021-10-18 15:57:13\tMarkDuplicates\tWill retain up to 477233152 duplicate indices before spilling to disk.\n",
      "INFO\t2021-10-18 15:57:16\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
      "INFO\t2021-10-18 15:57:16\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
      "INFO\t2021-10-18 15:57:16\tMarkDuplicates\tSorting list of duplicate records.\n",
      "INFO\t2021-10-18 15:57:17\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 539321528; totalMemory: 4361551872; maxMemory: 15271460864\n",
      "INFO\t2021-10-18 15:57:17\tMarkDuplicates\tMarking 475 records as duplicates.\n",
      "INFO\t2021-10-18 15:57:17\tMarkDuplicates\tFound 71 optical duplicate clusters.\n",
      "INFO\t2021-10-18 15:57:17\tMarkDuplicates\tReads are assumed to be ordered by: queryname\n",
      "INFO\t2021-10-18 15:57:17\tMarkDuplicates\tBefore output close freeMemory: 4356759656; totalMemory: 4361551872; maxMemory: 15271460864\n",
      "INFO\t2021-10-18 15:57:17\tMarkDuplicates\tAfter output close freeMemory: 4356834352; totalMemory: 4361551872; maxMemory: 15271460864\n",
      "\u001b[1;30mINFO\u001b[0m [job mark_duplicates_and_sort] Max memory used: 14267MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job mark_duplicates_and_sort] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step mark_duplicates_and_sort] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] starting step bqsr\n",
      "\u001b[1;30mINFO\u001b[0m [step bqsr] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job bqsr] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job bqsr] /private/tmp/docker_tmprs12uy0d$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmprs12uy0d,target=/SVAvuq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmptyhmem24,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpukvdlk98/MarkedSorted.bam,target=/var/lib/cwl/stgd0d7754f-2989-4b0d-a1b1-cca2c60f97ba/MarkedSorted.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpukvdlk98/MarkedSorted.bam.bai,target=/var/lib/cwl/stgd0d7754f-2989-4b0d-a1b1-cca2c60f97ba/MarkedSorted.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_known_indels.vcf.gz,target=/var/lib/cwl/stg3bb72bbf-a05f-4a7a-8e8c-20de96960846/hla_and_brca_genes_known_indels.vcf.gz,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_known_indels.vcf.gz.tbi,target=/var/lib/cwl/stg3bb72bbf-a05f-4a7a-8e8c-20de96960846/hla_and_brca_genes_known_indels.vcf.gz.tbi,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_mills.vcf.gz,target=/var/lib/cwl/stg6167b9e8-33d8-4502-aa2f-5e5a3efea260/hla_and_brca_genes_mills.vcf.gz,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_mills.vcf.gz.tbi,target=/var/lib/cwl/stg6167b9e8-33d8-4502-aa2f-5e5a3efea260/hla_and_brca_genes_mills.vcf.gz.tbi,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_dbsnp.vcf.gz,target=/var/lib/cwl/stg1c7dd296-db2d-4f77-8a4d-564c33e9b6c9/hla_and_brca_genes_dbsnp.vcf.gz,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_dbsnp.vcf.gz.tbi,target=/var/lib/cwl/stg1c7dd296-db2d-4f77-8a4d-564c33e9b6c9/hla_and_brca_genes_dbsnp.vcf.gz.tbi,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stg7444ef95-9f02-4ea6-8adf-9ba8543b57a0/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stg7444ef95-9f02-4ea6-8adf-9ba8543b57a0/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stg7444ef95-9f02-4ea6-8adf-9ba8543b57a0/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stg7444ef95-9f02-4ea6-8adf-9ba8543b57a0/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stg7444ef95-9f02-4ea6-8adf-9ba8543b57a0/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stg7444ef95-9f02-4ea6-8adf-9ba8543b57a0/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stg7444ef95-9f02-4ea6-8adf-9ba8543b57a0/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stg7444ef95-9f02-4ea6-8adf-9ba8543b57a0/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --workdir=/SVAvuq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/SVAvuq \\\n",
      "    --cidfile=/private/tmp/docker_tmpumu3valh/20211018105721-403239.cid \\\n",
      "    broadinstitute/gatk:4.1.8.1 \\\n",
      "    /gatk/gatk \\\n",
      "    --java-options \\\n",
      "    -Xmx16g \\\n",
      "    BaseRecalibrator \\\n",
      "    -O \\\n",
      "    /SVAvuq/bqsr.table \\\n",
      "    --known-sites \\\n",
      "    /var/lib/cwl/stg3bb72bbf-a05f-4a7a-8e8c-20de96960846/hla_and_brca_genes_known_indels.vcf.gz \\\n",
      "    --known-sites \\\n",
      "    /var/lib/cwl/stg6167b9e8-33d8-4502-aa2f-5e5a3efea260/hla_and_brca_genes_mills.vcf.gz \\\n",
      "    -L \\\n",
      "    chr6 \\\n",
      "    -L \\\n",
      "    chr17 \\\n",
      "    --known-sites \\\n",
      "    /var/lib/cwl/stg1c7dd296-db2d-4f77-8a4d-564c33e9b6c9/hla_and_brca_genes_dbsnp.vcf.gz \\\n",
      "    -R \\\n",
      "    /var/lib/cwl/stg7444ef95-9f02-4ea6-8adf-9ba8543b57a0/hla_and_brca_genes.fa \\\n",
      "    -I \\\n",
      "    /var/lib/cwl/stgd0d7754f-2989-4b0d-a1b1-cca2c60f97ba/MarkedSorted.bam\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15:57:26.876 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/gatk/gatk-package-4.1.8.1-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "Oct 18, 2021 3:57:27 PM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine\n",
      "INFO: Failed to detect whether we are running on Google Compute Engine.\n",
      "15:57:27.475 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
      "15:57:27.476 INFO  BaseRecalibrator - The Genome Analysis Toolkit (GATK) v4.1.8.1\n",
      "15:57:27.476 INFO  BaseRecalibrator - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "15:57:27.476 INFO  BaseRecalibrator - Executing as ?@8eaf51a749a7 on Linux v5.10.47-linuxkit amd64\n",
      "15:57:27.477 INFO  BaseRecalibrator - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_242-8u242-b08-0ubuntu3~18.04-b08\n",
      "15:57:27.477 INFO  BaseRecalibrator - Start Date/Time: October 18, 2021 3:57:26 PM GMT\n",
      "15:57:27.477 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
      "15:57:27.478 INFO  BaseRecalibrator - ------------------------------------------------------------\n",
      "15:57:27.479 INFO  BaseRecalibrator - HTSJDK Version: 2.23.0\n",
      "15:57:27.479 INFO  BaseRecalibrator - Picard Version: 2.22.8\n",
      "15:57:27.479 INFO  BaseRecalibrator - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "15:57:27.479 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "15:57:27.479 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "15:57:27.479 INFO  BaseRecalibrator - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "15:57:27.479 INFO  BaseRecalibrator - Deflater: IntelDeflater\n",
      "15:57:27.479 INFO  BaseRecalibrator - Inflater: IntelInflater\n",
      "15:57:27.479 INFO  BaseRecalibrator - GCS max retries/reopens: 20\n",
      "15:57:27.480 INFO  BaseRecalibrator - Requester pays: disabled\n",
      "15:57:27.480 INFO  BaseRecalibrator - Initializing engine\n",
      "15:57:27.950 INFO  FeatureManager - Using codec VCFCodec to read file file:///var/lib/cwl/stg3bb72bbf-a05f-4a7a-8e8c-20de96960846/hla_and_brca_genes_known_indels.vcf.gz\n",
      "15:57:28.005 INFO  FeatureManager - Using codec VCFCodec to read file file:///var/lib/cwl/stg6167b9e8-33d8-4502-aa2f-5e5a3efea260/hla_and_brca_genes_mills.vcf.gz\n",
      "15:57:28.034 INFO  FeatureManager - Using codec VCFCodec to read file file:///var/lib/cwl/stg1c7dd296-db2d-4f77-8a4d-564c33e9b6c9/hla_and_brca_genes_dbsnp.vcf.gz\n",
      "15:57:28.062 INFO  IntervalArgumentCollection - Processing 254063420 bp from intervals\n",
      "15:57:28.070 INFO  BaseRecalibrator - Done initializing engine\n",
      "15:57:28.080 INFO  BaseRecalibrationEngine - The covariates being used here: \n",
      "15:57:28.081 INFO  BaseRecalibrationEngine - \tReadGroupCovariate\n",
      "15:57:28.081 INFO  BaseRecalibrationEngine - \tQualityScoreCovariate\n",
      "15:57:28.081 INFO  BaseRecalibrationEngine - \tContextCovariate\n",
      "15:57:28.081 INFO  BaseRecalibrationEngine - \tCycleCovariate\n",
      "15:57:28.086 INFO  ProgressMeter - Starting traversal\n",
      "15:57:28.086 INFO  ProgressMeter -        Current Locus  Elapsed Minutes       Reads Processed     Reads/Minute\n",
      "15:57:29.314 INFO  BaseRecalibrator - 51 read(s) filtered by: MappingQualityNotZeroReadFilter \n",
      "0 read(s) filtered by: MappingQualityAvailableReadFilter \n",
      "0 read(s) filtered by: MappedReadFilter \n",
      "0 read(s) filtered by: NotSecondaryAlignmentReadFilter \n",
      "939 read(s) filtered by: NotDuplicateReadFilter \n",
      "0 read(s) filtered by: PassesVendorQualityCheckReadFilter \n",
      "0 read(s) filtered by: WellformedReadFilter \n",
      "990 total reads filtered\n",
      "15:57:29.317 INFO  ProgressMeter -       chr17:43104094              0.0                  8586         419511.4\n",
      "15:57:29.317 INFO  ProgressMeter - Traversal complete. Processed 8586 total reads in 0.0 minutes.\n",
      "15:57:29.399 INFO  BaseRecalibrator - Calculating quantized quality scores...\n",
      "15:57:29.413 INFO  BaseRecalibrator - Writing recalibration report...\n",
      "15:58:13.095 INFO  BaseRecalibrator - ...done!\n",
      "15:58:13.095 INFO  BaseRecalibrator - BaseRecalibrator was able to recalibrate 8586 reads\n",
      "15:58:13.095 INFO  BaseRecalibrator - Shutting down engine\n",
      "[October 18, 2021 3:58:13 PM GMT] org.broadinstitute.hellbender.tools.walkers.bqsr.BaseRecalibrator done. Elapsed time: 0.77 minutes.\n",
      "Runtime.totalMemory()=365428736\n",
      "Tool returned:\n",
      "SUCCESS\n",
      "Using GATK jar /gatk/gatk-package-4.1.8.1-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xmx16g -jar /gatk/gatk-package-4.1.8.1-local.jar BaseRecalibrator -O /SVAvuq/bqsr.table --known-sites /var/lib/cwl/stg3bb72bbf-a05f-4a7a-8e8c-20de96960846/hla_and_brca_genes_known_indels.vcf.gz --known-sites /var/lib/cwl/stg6167b9e8-33d8-4502-aa2f-5e5a3efea260/hla_and_brca_genes_mills.vcf.gz -L chr6 -L chr17 --known-sites /var/lib/cwl/stg1c7dd296-db2d-4f77-8a4d-564c33e9b6c9/hla_and_brca_genes_dbsnp.vcf.gz -R /var/lib/cwl/stg7444ef95-9f02-4ea6-8adf-9ba8543b57a0/hla_and_brca_genes.fa -I /var/lib/cwl/stgd0d7754f-2989-4b0d-a1b1-cca2c60f97ba/MarkedSorted.bam\n",
      "\u001b[1;30mINFO\u001b[0m [job bqsr] Max memory used: 1400MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job bqsr] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step bqsr] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] starting step apply_bqsr\n",
      "\u001b[1;30mINFO\u001b[0m [step apply_bqsr] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job apply_bqsr] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job apply_bqsr] /private/tmp/docker_tmp9wk_j5p_$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp9wk_j5p_,target=/SVAvuq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpor_iqtqa,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpukvdlk98/MarkedSorted.bam,target=/var/lib/cwl/stg43ef1119-044f-4651-87b3-f2adb71c481d/MarkedSorted.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpukvdlk98/MarkedSorted.bam.bai,target=/var/lib/cwl/stg43ef1119-044f-4651-87b3-f2adb71c481d/MarkedSorted.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmprs12uy0d/bqsr.table,target=/var/lib/cwl/stg4f9371fe-2ce2-4ba8-8888-b12240804f43/bqsr.table,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stg74268655-42dc-4c71-aa9d-80274643c5fe/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stg74268655-42dc-4c71-aa9d-80274643c5fe/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stg74268655-42dc-4c71-aa9d-80274643c5fe/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stg74268655-42dc-4c71-aa9d-80274643c5fe/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stg74268655-42dc-4c71-aa9d-80274643c5fe/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stg74268655-42dc-4c71-aa9d-80274643c5fe/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stg74268655-42dc-4c71-aa9d-80274643c5fe/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stg74268655-42dc-4c71-aa9d-80274643c5fe/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --workdir=/SVAvuq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/SVAvuq \\\n",
      "    --cidfile=/private/tmp/docker_tmpkq02wd_8/20211018105813-374127.cid \\\n",
      "    broadinstitute/gatk:4.1.8.1 \\\n",
      "    /gatk/gatk \\\n",
      "    --java-options \\\n",
      "    -Xmx16g \\\n",
      "    ApplyBQSR \\\n",
      "    -O \\\n",
      "    /SVAvuq/final.bam \\\n",
      "    --static-quantized-quals \\\n",
      "    10 \\\n",
      "    --static-quantized-quals \\\n",
      "    20 \\\n",
      "    --static-quantized-quals \\\n",
      "    30 \\\n",
      "    -R \\\n",
      "    /var/lib/cwl/stg74268655-42dc-4c71-aa9d-80274643c5fe/hla_and_brca_genes.fa \\\n",
      "    -I \\\n",
      "    /var/lib/cwl/stg43ef1119-044f-4651-87b3-f2adb71c481d/MarkedSorted.bam \\\n",
      "    -bqsr \\\n",
      "    /var/lib/cwl/stg4f9371fe-2ce2-4ba8-8888-b12240804f43/bqsr.table\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15:58:18.316 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/gatk/gatk-package-4.1.8.1-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "Oct 18, 2021 3:58:18 PM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine\n",
      "INFO: Failed to detect whether we are running on Google Compute Engine.\n",
      "15:58:18.894 INFO  ApplyBQSR - ------------------------------------------------------------\n",
      "15:58:18.896 INFO  ApplyBQSR - The Genome Analysis Toolkit (GATK) v4.1.8.1\n",
      "15:58:18.896 INFO  ApplyBQSR - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "15:58:18.896 INFO  ApplyBQSR - Executing as ?@c1eb68976ca9 on Linux v5.10.47-linuxkit amd64\n",
      "15:58:18.896 INFO  ApplyBQSR - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_242-8u242-b08-0ubuntu3~18.04-b08\n",
      "15:58:18.897 INFO  ApplyBQSR - Start Date/Time: October 18, 2021 3:58:18 PM GMT\n",
      "15:58:18.897 INFO  ApplyBQSR - ------------------------------------------------------------\n",
      "15:58:18.897 INFO  ApplyBQSR - ------------------------------------------------------------\n",
      "15:58:18.898 INFO  ApplyBQSR - HTSJDK Version: 2.23.0\n",
      "15:58:18.898 INFO  ApplyBQSR - Picard Version: 2.22.8\n",
      "15:58:18.898 INFO  ApplyBQSR - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "15:58:18.898 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "15:58:18.898 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "15:58:18.898 INFO  ApplyBQSR - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "15:58:18.898 INFO  ApplyBQSR - Deflater: IntelDeflater\n",
      "15:58:18.898 INFO  ApplyBQSR - Inflater: IntelInflater\n",
      "15:58:18.898 INFO  ApplyBQSR - GCS max retries/reopens: 20\n",
      "15:58:18.898 INFO  ApplyBQSR - Requester pays: disabled\n",
      "15:58:18.899 INFO  ApplyBQSR - Initializing engine\n",
      "15:58:19.358 INFO  ApplyBQSR - Done initializing engine\n",
      "15:58:19.396 INFO  ProgressMeter - Starting traversal\n",
      "15:58:19.396 INFO  ProgressMeter -        Current Locus  Elapsed Minutes       Reads Processed     Reads/Minute\n",
      "15:58:20.521 INFO  ApplyBQSR - 0 read(s) filtered by: WellformedReadFilter \n",
      "\n",
      "15:58:20.522 INFO  ProgressMeter -       chr17:43104823              0.0                  9576         510720.0\n",
      "15:58:20.522 INFO  ProgressMeter - Traversal complete. Processed 9576 total reads in 0.0 minutes.\n",
      "15:58:20.580 INFO  ApplyBQSR - Shutting down engine\n",
      "[October 18, 2021 3:58:20 PM GMT] org.broadinstitute.hellbender.tools.walkers.bqsr.ApplyBQSR done. Elapsed time: 0.04 minutes.\n",
      "Runtime.totalMemory()=250609664\n",
      "Using GATK jar /gatk/gatk-package-4.1.8.1-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xmx16g -jar /gatk/gatk-package-4.1.8.1-local.jar ApplyBQSR -O /SVAvuq/final.bam --static-quantized-quals 10 --static-quantized-quals 20 --static-quantized-quals 30 -R /var/lib/cwl/stg74268655-42dc-4c71-aa9d-80274643c5fe/hla_and_brca_genes.fa -I /var/lib/cwl/stg43ef1119-044f-4651-87b3-f2adb71c481d/MarkedSorted.bam -bqsr /var/lib/cwl/stg4f9371fe-2ce2-4ba8-8888-b12240804f43/bqsr.table\n",
      "\u001b[1;30mINFO\u001b[0m [job apply_bqsr] Max memory used: 950MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job apply_bqsr] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step apply_bqsr] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] starting step index_bam\n",
      "\u001b[1;30mINFO\u001b[0m [step index_bam] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job index_bam] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job index_bam] /private/tmp/docker_tmp7ahhx9xi$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp7ahhx9xi,target=/SVAvuq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpu10l8h_z,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp9wk_j5p_/final.bai,target=/var/lib/cwl/stgf1798654-a194-49a3-9b83-2c73ed289a7a/final.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp9wk_j5p_/final.bam,target=/SVAvuq/final.bam,readonly \\\n",
      "    --workdir=/SVAvuq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/SVAvuq \\\n",
      "    --cidfile=/private/tmp/docker_tmpxrrqe5pd/20211018105821-220079.cid \\\n",
      "    quay.io/biocontainers/samtools:1.11--h6270b1f_0 \\\n",
      "    /bin/sh \\\n",
      "    -c \\\n",
      "    '/usr/local/bin/samtools' 'index' '/SVAvuq/final.bam' '/SVAvuq/final.bam.bai'  &&  'cp' 'final.bam.bai' '/SVAvuq/final.bai'\n",
      "\u001b[1;30mINFO\u001b[0m [job index_bam] Max memory used: 0MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job index_bam] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step index_bam] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow alignment] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step alignment] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow ] starting step qc\n",
      "\u001b[1;30mINFO\u001b[0m [step qc] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] starting step collect_insert_size_metrics\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_insert_size_metrics] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job collect_insert_size_metrics] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_insert_size_metrics] /private/tmp/docker_tmpbrrtq1dn$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpbrrtq1dn,target=/SVAvuq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpb7890f8i,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp9wk_j5p_/final.bam,target=/var/lib/cwl/stg252a88cc-28ee-4c25-84e6-e1bc9995c8aa/final.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp7ahhx9xi/final.bam.bai,target=/var/lib/cwl/stg252a88cc-28ee-4c25-84e6-e1bc9995c8aa/final.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp7ahhx9xi/final.bai,target=/var/lib/cwl/stg252a88cc-28ee-4c25-84e6-e1bc9995c8aa/final.bai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stgaa675270-fbe0-42d7-a563-13717f22bdcf/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stgaa675270-fbe0-42d7-a563-13717f22bdcf/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stgaa675270-fbe0-42d7-a563-13717f22bdcf/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stgaa675270-fbe0-42d7-a563-13717f22bdcf/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stgaa675270-fbe0-42d7-a563-13717f22bdcf/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stgaa675270-fbe0-42d7-a563-13717f22bdcf/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stgaa675270-fbe0-42d7-a563-13717f22bdcf/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stgaa675270-fbe0-42d7-a563-13717f22bdcf/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --workdir=/SVAvuq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/SVAvuq \\\n",
      "    --cidfile=/private/tmp/docker_tmpwrtj1rgw/20211018105822-505251.cid \\\n",
      "    broadinstitute/picard:2.23.6 \\\n",
      "    /usr/bin/java \\\n",
      "    -Xmx16g \\\n",
      "    -jar \\\n",
      "    /usr/picard/picard.jar \\\n",
      "    CollectInsertSizeMetrics \\\n",
      "    O= \\\n",
      "    /SVAvuq/final.InsertSizeMetrics.txt \\\n",
      "    H= \\\n",
      "    /SVAvuq/final.InsertSizeHistogram.pdf \\\n",
      "    I= \\\n",
      "    /var/lib/cwl/stg252a88cc-28ee-4c25-84e6-e1bc9995c8aa/final.bam \\\n",
      "    METRIC_ACCUMULATION_LEVEL= \\\n",
      "    LIBRARY \\\n",
      "    REFERENCE_SEQUENCE= \\\n",
      "    /var/lib/cwl/stgaa675270-fbe0-42d7-a563-13717f22bdcf/hla_and_brca_genes.fa\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO\t2021-10-18 15:58:24\tCollectInsertSizeMetrics\t\n",
      "\n",
      "********** NOTE: Picard's command line syntax is changing.\n",
      "**********\n",
      "********** For more information, please see:\n",
      "********** https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
      "**********\n",
      "********** The command line looks like this in the new syntax:\n",
      "**********\n",
      "**********    CollectInsertSizeMetrics -O  /SVAvuq/final.InsertSizeMetrics.txt -H  /SVAvuq/final.InsertSizeHistogram.pdf -I  /var/lib/cwl/stg252a88cc-28ee-4c25-84e6-e1bc9995c8aa/final.bam -METRIC_ACCUMULATION_LEVEL  LIBRARY -REFERENCE_SEQUENCE  /var/lib/cwl/stgaa675270-fbe0-42d7-a563-13717f22bdcf/hla_and_brca_genes.fa\n",
      "**********\n",
      "\n",
      "\n",
      "15:58:25.056 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/usr/picard/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Oct 18 15:58:25 UTC 2021] CollectInsertSizeMetrics HISTOGRAM_FILE=/SVAvuq/final.InsertSizeHistogram.pdf METRIC_ACCUMULATION_LEVEL=[ALL_READS, LIBRARY] INPUT=/var/lib/cwl/stg252a88cc-28ee-4c25-84e6-e1bc9995c8aa/final.bam OUTPUT=/SVAvuq/final.InsertSizeMetrics.txt REFERENCE_SEQUENCE=/var/lib/cwl/stgaa675270-fbe0-42d7-a563-13717f22bdcf/hla_and_brca_genes.fa    DEVIATIONS=10.0 MINIMUM_PCT=0.05 INCLUDE_DUPLICATES=false ASSUME_SORTED=true STOP_AFTER=0 VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Mon Oct 18 15:58:25 UTC 2021] Executing as ?@bc460a9496be on Linux 5.10.47-linuxkit amd64; OpenJDK 64-Bit Server VM 11.0.8+10-post-Debian-1deb10u1; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.23.6-SNAPSHOT\n",
      "INFO\t2021-10-18 15:58:28\tRExecutor\tExecuting R script via command: Rscript /tmp/?/script10718127358846378514.R /SVAvuq/final.InsertSizeMetrics.txt /SVAvuq/final.InsertSizeHistogram.pdf final.bam\n",
      "INFO\t2021-10-18 15:58:29\tProcessExecutor\tnull device \n",
      "INFO\t2021-10-18 15:58:29\tProcessExecutor\t          1 \n",
      "[Mon Oct 18 15:58:29 UTC 2021] picard.analysis.CollectInsertSizeMetrics done. Elapsed time: 0.07 minutes.\n",
      "Runtime.totalMemory()=264241152\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_insert_size_metrics] Max memory used: 1087MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_insert_size_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_insert_size_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] starting step collect_detailed_hs_metrics\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_detailed_hs_metrics] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow collect_detailed_hs_metrics] start\n",
      "\u001b[1;30mINFO\u001b[0m [workflow collect_detailed_hs_metrics] starting step collect_summary_hs_metrics\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job step collect_summary_hs_metrics] Notice: scattering over empty input in 'bait_intervals', 'target_intervals', 'output_prefix'.  All outputs will be empty.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_summary_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow collect_detailed_hs_metrics] starting step collect_per_base_hs_metrics\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_per_base_hs_metrics] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job collect_per_base_hs_metrics] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_per_base_hs_metrics] /private/tmp/docker_tmp0g5z64jq$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp0g5z64jq,target=/SVAvuq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpsokg3nch,target=/tmp \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_target.interval_list,target=/var/lib/cwl/stg8ae76fd9-9c03-4f67-91cd-ca1c5e3f9fab/hla_and_brca_genes_target.interval_list,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp9wk_j5p_/final.bam,target=/var/lib/cwl/stg7d1bb6b5-96a2-439f-8588-d38ce0a869af/final.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp7ahhx9xi/final.bam.bai,target=/var/lib/cwl/stg7d1bb6b5-96a2-439f-8588-d38ce0a869af/final.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp7ahhx9xi/final.bai,target=/var/lib/cwl/stg7d1bb6b5-96a2-439f-8588-d38ce0a869af/final.bai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stg969d5b85-585d-4681-a9c8-cdfb900731fa/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stg969d5b85-585d-4681-a9c8-cdfb900731fa/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stg969d5b85-585d-4681-a9c8-cdfb900731fa/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stg969d5b85-585d-4681-a9c8-cdfb900731fa/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stg969d5b85-585d-4681-a9c8-cdfb900731fa/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stg969d5b85-585d-4681-a9c8-cdfb900731fa/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stg969d5b85-585d-4681-a9c8-cdfb900731fa/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stg969d5b85-585d-4681-a9c8-cdfb900731fa/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --workdir=/SVAvuq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/SVAvuq \\\n",
      "    --cidfile=/private/tmp/docker_tmpk09i7wc5/20211018105829-600910.cid \\\n",
      "    broadinstitute/picard:2.23.6 \\\n",
      "    /usr/bin/java \\\n",
      "    -Xmx48g \\\n",
      "    -jar \\\n",
      "    /usr/picard/picard.jar \\\n",
      "    CollectHsMetrics \\\n",
      "    O= \\\n",
      "    /SVAvuq/final.base-clinvar-HsMetrics.txt \\\n",
      "    BAIT_INTERVALS= \\\n",
      "    /var/lib/cwl/stg8ae76fd9-9c03-4f67-91cd-ca1c5e3f9fab/hla_and_brca_genes_target.interval_list \\\n",
      "    I= \\\n",
      "    /var/lib/cwl/stg7d1bb6b5-96a2-439f-8588-d38ce0a869af/final.bam \\\n",
      "    METRIC_ACCUMULATION_LEVEL= \\\n",
      "    ALL_READS \\\n",
      "    MINIMUM_BASE_QUALITY= \\\n",
      "    1 \\\n",
      "    MINIMUM_MAPPING_QUALITY= \\\n",
      "    1 \\\n",
      "    PER_BASE_COVERAGE= \\\n",
      "    final.base-clinvar-PerBaseCoverage.txt \\\n",
      "    R= \\\n",
      "    /var/lib/cwl/stg969d5b85-585d-4681-a9c8-cdfb900731fa/hla_and_brca_genes.fa \\\n",
      "    TARGET_INTERVALS= \\\n",
      "    /var/lib/cwl/stg8ae76fd9-9c03-4f67-91cd-ca1c5e3f9fab/hla_and_brca_genes_target.interval_list\n",
      "INFO\t2021-10-18 15:58:31\tCollectHsMetrics\t\n",
      "\n",
      "********** NOTE: Picard's command line syntax is changing.\n",
      "**********\n",
      "********** For more information, please see:\n",
      "********** https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
      "**********\n",
      "********** The command line looks like this in the new syntax:\n",
      "**********\n",
      "**********    CollectHsMetrics -O  /SVAvuq/final.base-clinvar-HsMetrics.txt -BAIT_INTERVALS  /var/lib/cwl/stg8ae76fd9-9c03-4f67-91cd-ca1c5e3f9fab/hla_and_brca_genes_target.interval_list -I  /var/lib/cwl/stg7d1bb6b5-96a2-439f-8588-d38ce0a869af/final.bam -METRIC_ACCUMULATION_LEVEL  ALL_READS -MINIMUM_BASE_QUALITY  1 -MINIMUM_MAPPING_QUALITY  1 -PER_BASE_COVERAGE  final.base-clinvar-PerBaseCoverage.txt -R  /var/lib/cwl/stg969d5b85-585d-4681-a9c8-cdfb900731fa/hla_and_brca_genes.fa -TARGET_INTERVALS  /var/lib/cwl/stg8ae76fd9-9c03-4f67-91cd-ca1c5e3f9fab/hla_and_brca_genes_target.interval_list\n",
      "**********\n",
      "\n",
      "\n",
      "15:58:31.839 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/usr/picard/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Oct 18 15:58:32 UTC 2021] CollectHsMetrics BAIT_INTERVALS=[/var/lib/cwl/stg8ae76fd9-9c03-4f67-91cd-ca1c5e3f9fab/hla_and_brca_genes_target.interval_list] TARGET_INTERVALS=[/var/lib/cwl/stg8ae76fd9-9c03-4f67-91cd-ca1c5e3f9fab/hla_and_brca_genes_target.interval_list] INPUT=/var/lib/cwl/stg7d1bb6b5-96a2-439f-8588-d38ce0a869af/final.bam OUTPUT=/SVAvuq/final.base-clinvar-HsMetrics.txt METRIC_ACCUMULATION_LEVEL=[ALL_READS] PER_BASE_COVERAGE=final.base-clinvar-PerBaseCoverage.txt MINIMUM_MAPPING_QUALITY=1 MINIMUM_BASE_QUALITY=1 REFERENCE_SEQUENCE=/var/lib/cwl/stg969d5b85-585d-4681-a9c8-cdfb900731fa/hla_and_brca_genes.fa    NEAR_DISTANCE=250 CLIP_OVERLAPPING_READS=true INCLUDE_INDELS=false COVERAGE_CAP=200 SAMPLE_SIZE=10000 ALLELE_FRACTION=[0.001, 0.005, 0.01, 0.02, 0.05, 0.1, 0.2, 0.3, 0.5] VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Mon Oct 18 15:58:32 UTC 2021] Executing as ?@339d0b2d998e on Linux 5.10.47-linuxkit amd64; OpenJDK 64-Bit Server VM 11.0.8+10-post-Debian-1deb10u1; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.23.6-SNAPSHOT\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO\t2021-10-18 15:58:32\tTheoreticalSensitivity\tCreating Roulette Wheel\n",
      "INFO\t2021-10-18 15:58:32\tTheoreticalSensitivity\tCalculating quality sums from quality sampler\n",
      "INFO\t2021-10-18 15:58:32\tTheoreticalSensitivity\t0 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:33\tTheoreticalSensitivity\t1000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:34\tTheoreticalSensitivity\t2000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:35\tTheoreticalSensitivity\t3000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:36\tTheoreticalSensitivity\t4000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:37\tTheoreticalSensitivity\t5000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:38\tTheoreticalSensitivity\t6000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:39\tTheoreticalSensitivity\t7000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:40\tTheoreticalSensitivity\t8000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:41\tTheoreticalSensitivity\t9000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:42\tTheoreticalSensitivity\tCalculating theoretical het sensitivity\n",
      "INFO\t2021-10-18 15:58:43\tTargetMetricsCollector\tCalculating GC metrics\n",
      "[Mon Oct 18 15:58:43 UTC 2021] picard.analysis.directed.CollectHsMetrics done. Elapsed time: 0.19 minutes.\n",
      "Runtime.totalMemory()=150994944\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_per_base_hs_metrics] Max memory used: 665MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_per_base_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_per_base_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow collect_detailed_hs_metrics] starting step collect_per_target_hs_metrics\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_per_target_hs_metrics] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job collect_per_target_hs_metrics] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_per_target_hs_metrics] /private/tmp/docker_tmpzfuw3iks$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpzfuw3iks,target=/SVAvuq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpaurml49y,target=/tmp \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_target.interval_list,target=/var/lib/cwl/stg016eb90d-d883-4b68-9802-ab3285ac5077/hla_and_brca_genes_target.interval_list,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp9wk_j5p_/final.bam,target=/var/lib/cwl/stg0d7965ce-b296-422c-b64f-7a3ccff027d0/final.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp7ahhx9xi/final.bam.bai,target=/var/lib/cwl/stg0d7965ce-b296-422c-b64f-7a3ccff027d0/final.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp7ahhx9xi/final.bai,target=/var/lib/cwl/stg0d7965ce-b296-422c-b64f-7a3ccff027d0/final.bai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stg8f8ab82b-16d9-4500-8a10-5ef2193ab2e6/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stg8f8ab82b-16d9-4500-8a10-5ef2193ab2e6/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stg8f8ab82b-16d9-4500-8a10-5ef2193ab2e6/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stg8f8ab82b-16d9-4500-8a10-5ef2193ab2e6/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stg8f8ab82b-16d9-4500-8a10-5ef2193ab2e6/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stg8f8ab82b-16d9-4500-8a10-5ef2193ab2e6/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stg8f8ab82b-16d9-4500-8a10-5ef2193ab2e6/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stg8f8ab82b-16d9-4500-8a10-5ef2193ab2e6/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --workdir=/SVAvuq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/SVAvuq \\\n",
      "    --cidfile=/private/tmp/docker_tmp6twd8h3w/20211018105843-721240.cid \\\n",
      "    broadinstitute/picard:2.23.6 \\\n",
      "    /usr/bin/java \\\n",
      "    -Xmx48g \\\n",
      "    -jar \\\n",
      "    /usr/picard/picard.jar \\\n",
      "    CollectHsMetrics \\\n",
      "    O= \\\n",
      "    /SVAvuq/final.target-acmg_genes-HsMetrics.txt \\\n",
      "    BAIT_INTERVALS= \\\n",
      "    /var/lib/cwl/stg016eb90d-d883-4b68-9802-ab3285ac5077/hla_and_brca_genes_target.interval_list \\\n",
      "    I= \\\n",
      "    /var/lib/cwl/stg0d7965ce-b296-422c-b64f-7a3ccff027d0/final.bam \\\n",
      "    METRIC_ACCUMULATION_LEVEL= \\\n",
      "    ALL_READS \\\n",
      "    MINIMUM_BASE_QUALITY= \\\n",
      "    1 \\\n",
      "    MINIMUM_MAPPING_QUALITY= \\\n",
      "    1 \\\n",
      "    PER_TARGET_COVERAGE= \\\n",
      "    final.target-acmg_genes-PerTargetCoverage.txt \\\n",
      "    R= \\\n",
      "    /var/lib/cwl/stg8f8ab82b-16d9-4500-8a10-5ef2193ab2e6/hla_and_brca_genes.fa \\\n",
      "    TARGET_INTERVALS= \\\n",
      "    /var/lib/cwl/stg016eb90d-d883-4b68-9802-ab3285ac5077/hla_and_brca_genes_target.interval_list\n",
      "INFO\t2021-10-18 15:58:45\tCollectHsMetrics\t\n",
      "\n",
      "********** NOTE: Picard's command line syntax is changing.\n",
      "**********\n",
      "********** For more information, please see:\n",
      "********** https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
      "**********\n",
      "********** The command line looks like this in the new syntax:\n",
      "**********\n",
      "**********    CollectHsMetrics -O  /SVAvuq/final.target-acmg_genes-HsMetrics.txt -BAIT_INTERVALS  /var/lib/cwl/stg016eb90d-d883-4b68-9802-ab3285ac5077/hla_and_brca_genes_target.interval_list -I  /var/lib/cwl/stg0d7965ce-b296-422c-b64f-7a3ccff027d0/final.bam -METRIC_ACCUMULATION_LEVEL  ALL_READS -MINIMUM_BASE_QUALITY  1 -MINIMUM_MAPPING_QUALITY  1 -PER_TARGET_COVERAGE  final.target-acmg_genes-PerTargetCoverage.txt -R  /var/lib/cwl/stg8f8ab82b-16d9-4500-8a10-5ef2193ab2e6/hla_and_brca_genes.fa -TARGET_INTERVALS  /var/lib/cwl/stg016eb90d-d883-4b68-9802-ab3285ac5077/hla_and_brca_genes_target.interval_list\n",
      "**********\n",
      "\n",
      "\n",
      "15:58:45.635 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/usr/picard/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Oct 18 15:58:45 UTC 2021] CollectHsMetrics BAIT_INTERVALS=[/var/lib/cwl/stg016eb90d-d883-4b68-9802-ab3285ac5077/hla_and_brca_genes_target.interval_list] TARGET_INTERVALS=[/var/lib/cwl/stg016eb90d-d883-4b68-9802-ab3285ac5077/hla_and_brca_genes_target.interval_list] INPUT=/var/lib/cwl/stg0d7965ce-b296-422c-b64f-7a3ccff027d0/final.bam OUTPUT=/SVAvuq/final.target-acmg_genes-HsMetrics.txt METRIC_ACCUMULATION_LEVEL=[ALL_READS] PER_TARGET_COVERAGE=final.target-acmg_genes-PerTargetCoverage.txt MINIMUM_MAPPING_QUALITY=1 MINIMUM_BASE_QUALITY=1 REFERENCE_SEQUENCE=/var/lib/cwl/stg8f8ab82b-16d9-4500-8a10-5ef2193ab2e6/hla_and_brca_genes.fa    NEAR_DISTANCE=250 CLIP_OVERLAPPING_READS=true INCLUDE_INDELS=false COVERAGE_CAP=200 SAMPLE_SIZE=10000 ALLELE_FRACTION=[0.001, 0.005, 0.01, 0.02, 0.05, 0.1, 0.2, 0.3, 0.5] VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Mon Oct 18 15:58:45 UTC 2021] Executing as ?@c3872f34a5ca on Linux 5.10.47-linuxkit amd64; OpenJDK 64-Bit Server VM 11.0.8+10-post-Debian-1deb10u1; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.23.6-SNAPSHOT\n",
      "INFO\t2021-10-18 15:58:46\tTheoreticalSensitivity\tCreating Roulette Wheel\n",
      "INFO\t2021-10-18 15:58:46\tTheoreticalSensitivity\tCalculating quality sums from quality sampler\n",
      "INFO\t2021-10-18 15:58:46\tTheoreticalSensitivity\t0 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:47\tTheoreticalSensitivity\t1000 sampling iterations completed\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO\t2021-10-18 15:58:48\tTheoreticalSensitivity\t2000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:49\tTheoreticalSensitivity\t3000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:49\tTheoreticalSensitivity\t4000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:50\tTheoreticalSensitivity\t5000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:51\tTheoreticalSensitivity\t6000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:52\tTheoreticalSensitivity\t7000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:53\tTheoreticalSensitivity\t8000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:54\tTheoreticalSensitivity\t9000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:58:55\tTheoreticalSensitivity\tCalculating theoretical het sensitivity\n",
      "INFO\t2021-10-18 15:58:56\tTargetMetricsCollector\tCalculating GC metrics\n",
      "[Mon Oct 18 15:58:56 UTC 2021] picard.analysis.directed.CollectHsMetrics done. Elapsed time: 0.17 minutes.\n",
      "Runtime.totalMemory()=150994944\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_per_target_hs_metrics] Max memory used: 647MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_per_target_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_per_target_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow collect_detailed_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_detailed_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] starting step collect_alignment_summary_metrics\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_alignment_summary_metrics] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job collect_alignment_summary_metrics] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_alignment_summary_metrics] /private/tmp/docker_tmpk1rr170i$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpk1rr170i,target=/SVAvuq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp34e4z9u0,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp9wk_j5p_/final.bam,target=/var/lib/cwl/stg6c0413e0-7822-4228-aff8-a3cee8560eeb/final.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp7ahhx9xi/final.bam.bai,target=/var/lib/cwl/stg6c0413e0-7822-4228-aff8-a3cee8560eeb/final.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp7ahhx9xi/final.bai,target=/var/lib/cwl/stg6c0413e0-7822-4228-aff8-a3cee8560eeb/final.bai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stg9a6891cd-c71a-413d-a1a7-78f23bb88f66/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stg9a6891cd-c71a-413d-a1a7-78f23bb88f66/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stg9a6891cd-c71a-413d-a1a7-78f23bb88f66/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stg9a6891cd-c71a-413d-a1a7-78f23bb88f66/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stg9a6891cd-c71a-413d-a1a7-78f23bb88f66/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stg9a6891cd-c71a-413d-a1a7-78f23bb88f66/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stg9a6891cd-c71a-413d-a1a7-78f23bb88f66/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stg9a6891cd-c71a-413d-a1a7-78f23bb88f66/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --workdir=/SVAvuq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/SVAvuq \\\n",
      "    --cidfile=/private/tmp/docker_tmpeqztgmuj/20211018105856-292343.cid \\\n",
      "    broadinstitute/picard:2.23.6 \\\n",
      "    /usr/bin/java \\\n",
      "    -Xmx16g \\\n",
      "    -jar \\\n",
      "    /usr/picard/picard.jar \\\n",
      "    CollectAlignmentSummaryMetrics \\\n",
      "    OUTPUT= \\\n",
      "    /SVAvuq/final.AlignmentSummaryMetrics.txt \\\n",
      "    INPUT= \\\n",
      "    /var/lib/cwl/stg6c0413e0-7822-4228-aff8-a3cee8560eeb/final.bam \\\n",
      "    METRIC_ACCUMULATION_LEVEL= \\\n",
      "    LIBRARY \\\n",
      "    REFERENCE_SEQUENCE= \\\n",
      "    /var/lib/cwl/stg9a6891cd-c71a-413d-a1a7-78f23bb88f66/hla_and_brca_genes.fa\n",
      "INFO\t2021-10-18 15:58:57\tCollectAlignmentSummaryMetrics\t\n",
      "\n",
      "********** NOTE: Picard's command line syntax is changing.\n",
      "**********\n",
      "********** For more information, please see:\n",
      "********** https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
      "**********\n",
      "********** The command line looks like this in the new syntax:\n",
      "**********\n",
      "**********    CollectAlignmentSummaryMetrics -OUTPUT  /SVAvuq/final.AlignmentSummaryMetrics.txt -INPUT  /var/lib/cwl/stg6c0413e0-7822-4228-aff8-a3cee8560eeb/final.bam -METRIC_ACCUMULATION_LEVEL  LIBRARY -REFERENCE_SEQUENCE  /var/lib/cwl/stg9a6891cd-c71a-413d-a1a7-78f23bb88f66/hla_and_brca_genes.fa\n",
      "**********\n",
      "\n",
      "\n",
      "15:58:58.478 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/usr/picard/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Oct 18 15:58:58 UTC 2021] CollectAlignmentSummaryMetrics METRIC_ACCUMULATION_LEVEL=[ALL_READS, LIBRARY] INPUT=/var/lib/cwl/stg6c0413e0-7822-4228-aff8-a3cee8560eeb/final.bam OUTPUT=/SVAvuq/final.AlignmentSummaryMetrics.txt REFERENCE_SEQUENCE=/var/lib/cwl/stg9a6891cd-c71a-413d-a1a7-78f23bb88f66/hla_and_brca_genes.fa    MAX_INSERT_SIZE=100000 EXPECTED_PAIR_ORIENTATIONS=[FR] ADAPTER_SEQUENCE=[AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT, AGATCGGAAGAGCTCGTATGCCGTCTTCTGCTTG, AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT, AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG, AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT, AGATCGGAAGAGCACACGTCTGAACTCCAGTCACNNNNNNNNATCTCGTATGCCGTCTTCTGCTTG] IS_BISULFITE_SEQUENCED=false COLLECT_ALIGNMENT_INFORMATION=true ASSUME_SORTED=true STOP_AFTER=0 VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Mon Oct 18 15:58:58 UTC 2021] Executing as ?@8694106041d4 on Linux 5.10.47-linuxkit amd64; OpenJDK 64-Bit Server VM 11.0.8+10-post-Debian-1deb10u1; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.23.6-SNAPSHOT\n",
      "[Mon Oct 18 15:59:02 UTC 2021] picard.analysis.CollectAlignmentSummaryMetrics done. Elapsed time: 0.06 minutes.\n",
      "Runtime.totalMemory()=264241152\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_alignment_summary_metrics] Max memory used: 1081MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_alignment_summary_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_alignment_summary_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] starting step collect_roi_hs_metrics\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_roi_hs_metrics] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job collect_roi_hs_metrics] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_roi_hs_metrics] /private/tmp/docker_tmpk5s1o794$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpk5s1o794,target=/SVAvuq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpe4wzi5aq,target=/tmp \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_bait.interval_list,target=/var/lib/cwl/stg37cc34e8-0eef-45e8-a28d-4f74309408cd/hla_and_brca_genes_bait.interval_list,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp9wk_j5p_/final.bam,target=/var/lib/cwl/stg1afeaf88-697a-4fc9-bd95-30ffae680aa7/final.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp7ahhx9xi/final.bam.bai,target=/var/lib/cwl/stg1afeaf88-697a-4fc9-bd95-30ffae680aa7/final.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp7ahhx9xi/final.bai,target=/var/lib/cwl/stg1afeaf88-697a-4fc9-bd95-30ffae680aa7/final.bai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stg6c04854e-ef22-4b8d-b1e2-cad0f736768a/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stg6c04854e-ef22-4b8d-b1e2-cad0f736768a/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stg6c04854e-ef22-4b8d-b1e2-cad0f736768a/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stg6c04854e-ef22-4b8d-b1e2-cad0f736768a/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stg6c04854e-ef22-4b8d-b1e2-cad0f736768a/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stg6c04854e-ef22-4b8d-b1e2-cad0f736768a/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stg6c04854e-ef22-4b8d-b1e2-cad0f736768a/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stg6c04854e-ef22-4b8d-b1e2-cad0f736768a/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_target.interval_list,target=/var/lib/cwl/stgaeea3667-8f2d-454c-946e-fa780ce44a69/hla_and_brca_genes_target.interval_list,readonly \\\n",
      "    --workdir=/SVAvuq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/SVAvuq \\\n",
      "    --cidfile=/private/tmp/docker_tmpo7w80mfp/20211018105902-557982.cid \\\n",
      "    broadinstitute/picard:2.23.6 \\\n",
      "    /usr/bin/java \\\n",
      "    -Xmx48g \\\n",
      "    -jar \\\n",
      "    /usr/picard/picard.jar \\\n",
      "    CollectHsMetrics \\\n",
      "    O= \\\n",
      "    /SVAvuq/final.roi-HsMetrics.txt \\\n",
      "    BAIT_INTERVALS= \\\n",
      "    /var/lib/cwl/stg37cc34e8-0eef-45e8-a28d-4f74309408cd/hla_and_brca_genes_bait.interval_list \\\n",
      "    I= \\\n",
      "    /var/lib/cwl/stg1afeaf88-697a-4fc9-bd95-30ffae680aa7/final.bam \\\n",
      "    METRIC_ACCUMULATION_LEVEL= \\\n",
      "    ALL_READS \\\n",
      "    MINIMUM_BASE_QUALITY= \\\n",
      "    1 \\\n",
      "    MINIMUM_MAPPING_QUALITY= \\\n",
      "    1 \\\n",
      "    R= \\\n",
      "    /var/lib/cwl/stg6c04854e-ef22-4b8d-b1e2-cad0f736768a/hla_and_brca_genes.fa \\\n",
      "    TARGET_INTERVALS= \\\n",
      "    /var/lib/cwl/stgaeea3667-8f2d-454c-946e-fa780ce44a69/hla_and_brca_genes_target.interval_list\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO\t2021-10-18 15:59:03\tCollectHsMetrics\t\n",
      "\n",
      "********** NOTE: Picard's command line syntax is changing.\n",
      "**********\n",
      "********** For more information, please see:\n",
      "********** https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
      "**********\n",
      "********** The command line looks like this in the new syntax:\n",
      "**********\n",
      "**********    CollectHsMetrics -O  /SVAvuq/final.roi-HsMetrics.txt -BAIT_INTERVALS  /var/lib/cwl/stg37cc34e8-0eef-45e8-a28d-4f74309408cd/hla_and_brca_genes_bait.interval_list -I  /var/lib/cwl/stg1afeaf88-697a-4fc9-bd95-30ffae680aa7/final.bam -METRIC_ACCUMULATION_LEVEL  ALL_READS -MINIMUM_BASE_QUALITY  1 -MINIMUM_MAPPING_QUALITY  1 -R  /var/lib/cwl/stg6c04854e-ef22-4b8d-b1e2-cad0f736768a/hla_and_brca_genes.fa -TARGET_INTERVALS  /var/lib/cwl/stgaeea3667-8f2d-454c-946e-fa780ce44a69/hla_and_brca_genes_target.interval_list\n",
      "**********\n",
      "\n",
      "\n",
      "15:59:04.563 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/usr/picard/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Mon Oct 18 15:59:04 UTC 2021] CollectHsMetrics BAIT_INTERVALS=[/var/lib/cwl/stg37cc34e8-0eef-45e8-a28d-4f74309408cd/hla_and_brca_genes_bait.interval_list] TARGET_INTERVALS=[/var/lib/cwl/stgaeea3667-8f2d-454c-946e-fa780ce44a69/hla_and_brca_genes_target.interval_list] INPUT=/var/lib/cwl/stg1afeaf88-697a-4fc9-bd95-30ffae680aa7/final.bam OUTPUT=/SVAvuq/final.roi-HsMetrics.txt METRIC_ACCUMULATION_LEVEL=[ALL_READS] MINIMUM_MAPPING_QUALITY=1 MINIMUM_BASE_QUALITY=1 REFERENCE_SEQUENCE=/var/lib/cwl/stg6c04854e-ef22-4b8d-b1e2-cad0f736768a/hla_and_brca_genes.fa    NEAR_DISTANCE=250 CLIP_OVERLAPPING_READS=true INCLUDE_INDELS=false COVERAGE_CAP=200 SAMPLE_SIZE=10000 ALLELE_FRACTION=[0.001, 0.005, 0.01, 0.02, 0.05, 0.1, 0.2, 0.3, 0.5] VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Mon Oct 18 15:59:04 UTC 2021] Executing as ?@36c1e2bd89d1 on Linux 5.10.47-linuxkit amd64; OpenJDK 64-Bit Server VM 11.0.8+10-post-Debian-1deb10u1; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.23.6-SNAPSHOT\n",
      "INFO\t2021-10-18 15:59:05\tTheoreticalSensitivity\tCreating Roulette Wheel\n",
      "INFO\t2021-10-18 15:59:05\tTheoreticalSensitivity\tCalculating quality sums from quality sampler\n",
      "INFO\t2021-10-18 15:59:05\tTheoreticalSensitivity\t0 sampling iterations completed\n",
      "INFO\t2021-10-18 15:59:06\tTheoreticalSensitivity\t1000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:59:07\tTheoreticalSensitivity\t2000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:59:08\tTheoreticalSensitivity\t3000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:59:08\tTheoreticalSensitivity\t4000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:59:09\tTheoreticalSensitivity\t5000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:59:10\tTheoreticalSensitivity\t6000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:59:11\tTheoreticalSensitivity\t7000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:59:12\tTheoreticalSensitivity\t8000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:59:13\tTheoreticalSensitivity\t9000 sampling iterations completed\n",
      "INFO\t2021-10-18 15:59:14\tTheoreticalSensitivity\tCalculating theoretical het sensitivity\n",
      "INFO\t2021-10-18 15:59:15\tTargetMetricsCollector\tCalculating GC metrics\n",
      "[Mon Oct 18 15:59:15 UTC 2021] picard.analysis.directed.CollectHsMetrics done. Elapsed time: 0.18 minutes.\n",
      "Runtime.totalMemory()=150994944\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_roi_hs_metrics] Max memory used: 622MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job collect_roi_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step collect_roi_hs_metrics] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] starting step samtools_flagstat\n",
      "\u001b[1;30mINFO\u001b[0m [step samtools_flagstat] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job samtools_flagstat] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job samtools_flagstat] /private/tmp/docker_tmpcl6yd9g6$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpcl6yd9g6,target=/SVAvuq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp_5zmufm1,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp9wk_j5p_/final.bam,target=/var/lib/cwl/stgc2785c32-230e-4dca-aaf4-ba9a600a0c58/final.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp7ahhx9xi/final.bam.bai,target=/var/lib/cwl/stgc2785c32-230e-4dca-aaf4-ba9a600a0c58/final.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp7ahhx9xi/final.bai,target=/var/lib/cwl/stgc2785c32-230e-4dca-aaf4-ba9a600a0c58/final.bai,readonly \\\n",
      "    --workdir=/SVAvuq \\\n",
      "    --read-only=true \\\n",
      "    --log-driver=none \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/SVAvuq \\\n",
      "    --cidfile=/private/tmp/docker_tmpebjcwl6f/20211018105915-224123.cid \\\n",
      "    quay.io/biocontainers/samtools:1.11--h6270b1f_0 \\\n",
      "    /usr/local/bin/samtools \\\n",
      "    flagstat \\\n",
      "    /var/lib/cwl/stgc2785c32-230e-4dca-aaf4-ba9a600a0c58/final.bam > /private/tmp/docker_tmpcl6yd9g6/final.bam.flagstat\n",
      "\u001b[1;30mINFO\u001b[0m [job samtools_flagstat] Max memory used: 0MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job samtools_flagstat] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step samtools_flagstat] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] starting step select_variants\n",
      "\u001b[1;30mINFO\u001b[0m [step select_variants] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job select_variants] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job select_variants] /private/tmp/docker_tmpvufs9vcj$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpvufs9vcj,target=/SVAvuq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmptd3qdq42,target=/tmp \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_target.interval_list,target=/var/lib/cwl/stg9ef62f94-5c20-45c9-8137-3b6fbf62d3c6/hla_and_brca_genes_target.interval_list,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa,target=/var/lib/cwl/stg69f07560-4efb-4317-8ddf-ea95741944fe/hla_and_brca_genes.fa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.fai,target=/var/lib/cwl/stg69f07560-4efb-4317-8ddf-ea95741944fe/hla_and_brca_genes.fa.fai,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.dict,target=/var/lib/cwl/stg69f07560-4efb-4317-8ddf-ea95741944fe/hla_and_brca_genes.dict,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.amb,target=/var/lib/cwl/stg69f07560-4efb-4317-8ddf-ea95741944fe/hla_and_brca_genes.fa.amb,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.ann,target=/var/lib/cwl/stg69f07560-4efb-4317-8ddf-ea95741944fe/hla_and_brca_genes.fa.ann,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.bwt,target=/var/lib/cwl/stg69f07560-4efb-4317-8ddf-ea95741944fe/hla_and_brca_genes.fa.bwt,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.pac,target=/var/lib/cwl/stg69f07560-4efb-4317-8ddf-ea95741944fe/hla_and_brca_genes.fa.pac,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes.fa.sa,target=/var/lib/cwl/stg69f07560-4efb-4317-8ddf-ea95741944fe/hla_and_brca_genes.fa.sa,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_omni.vcf.gz,target=/var/lib/cwl/stg76080261-9780-4b81-978b-15092c68fc25/hla_and_brca_genes_omni.vcf.gz,readonly \\\n",
      "    --mount=type=bind,source=/Users/jwalker/git/bfx_workshop/lectures/week_05/hla_and_brca_genes_omni.vcf.gz.tbi,target=/var/lib/cwl/stg76080261-9780-4b81-978b-15092c68fc25/hla_and_brca_genes_omni.vcf.gz.tbi,readonly \\\n",
      "    --workdir=/SVAvuq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/SVAvuq \\\n",
      "    --cidfile=/private/tmp/docker_tmpwnvo9q1w/20211018105916-263772.cid \\\n",
      "    broadinstitute/gatk:4.1.8.1 \\\n",
      "    /gatk/gatk \\\n",
      "    --java-options \\\n",
      "    -Xmx4g \\\n",
      "    SelectVariants \\\n",
      "    -O \\\n",
      "    /SVAvuq/select_variants.vcf.gz \\\n",
      "    -R \\\n",
      "    /var/lib/cwl/stg69f07560-4efb-4317-8ddf-ea95741944fe/hla_and_brca_genes.fa \\\n",
      "    --variant \\\n",
      "    /var/lib/cwl/stg76080261-9780-4b81-978b-15092c68fc25/hla_and_brca_genes_omni.vcf.gz \\\n",
      "    -L \\\n",
      "    /var/lib/cwl/stg9ef62f94-5c20-45c9-8137-3b6fbf62d3c6/hla_and_brca_genes_target.interval_list\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15:59:20.847 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/gatk/gatk-package-4.1.8.1-local.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "Oct 18, 2021 3:59:21 PM shaded.cloud_nio.com.google.auth.oauth2.ComputeEngineCredentials runningOnComputeEngine\n",
      "INFO: Failed to detect whether we are running on Google Compute Engine.\n",
      "15:59:21.338 INFO  SelectVariants - ------------------------------------------------------------\n",
      "15:59:21.339 INFO  SelectVariants - The Genome Analysis Toolkit (GATK) v4.1.8.1\n",
      "15:59:21.339 INFO  SelectVariants - For support and documentation go to https://software.broadinstitute.org/gatk/\n",
      "15:59:21.339 INFO  SelectVariants - Executing as ?@c3ad63af7801 on Linux v5.10.47-linuxkit amd64\n",
      "15:59:21.339 INFO  SelectVariants - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_242-8u242-b08-0ubuntu3~18.04-b08\n",
      "15:59:21.339 INFO  SelectVariants - Start Date/Time: October 18, 2021 3:59:20 PM GMT\n",
      "15:59:21.339 INFO  SelectVariants - ------------------------------------------------------------\n",
      "15:59:21.339 INFO  SelectVariants - ------------------------------------------------------------\n",
      "15:59:21.340 INFO  SelectVariants - HTSJDK Version: 2.23.0\n",
      "15:59:21.340 INFO  SelectVariants - Picard Version: 2.22.8\n",
      "15:59:21.340 INFO  SelectVariants - HTSJDK Defaults.COMPRESSION_LEVEL : 2\n",
      "15:59:21.340 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false\n",
      "15:59:21.340 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true\n",
      "15:59:21.341 INFO  SelectVariants - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false\n",
      "15:59:21.341 INFO  SelectVariants - Deflater: IntelDeflater\n",
      "15:59:21.341 INFO  SelectVariants - Inflater: IntelInflater\n",
      "15:59:21.341 INFO  SelectVariants - GCS max retries/reopens: 20\n",
      "15:59:21.341 INFO  SelectVariants - Requester pays: disabled\n",
      "15:59:21.341 INFO  SelectVariants - Initializing engine\n",
      "15:59:21.701 INFO  FeatureManager - Using codec VCFCodec to read file file:///var/lib/cwl/stg76080261-9780-4b81-978b-15092c68fc25/hla_and_brca_genes_omni.vcf.gz\n",
      "15:59:21.766 INFO  FeatureManager - Using codec IntervalListCodec to read file file:///var/lib/cwl/stg9ef62f94-5c20-45c9-8137-3b6fbf62d3c6/hla_and_brca_genes_target.interval_list\n",
      "15:59:21.794 INFO  IntervalArgumentCollection - Processing 10741 bp from intervals\n",
      "15:59:21.801 INFO  SelectVariants - Done initializing engine\n",
      "15:59:21.830 INFO  ProgressMeter - Starting traversal\n",
      "15:59:21.830 INFO  ProgressMeter -        Current Locus  Elapsed Minutes    Variants Processed  Variants/Minute\n",
      "15:59:21.850 INFO  ProgressMeter -             unmapped              0.0                     2           6000.0\n",
      "15:59:21.851 INFO  ProgressMeter - Traversal complete. Processed 2 total variants in 0.0 minutes.\n",
      "15:59:21.876 INFO  SelectVariants - Shutting down engine\n",
      "[October 18, 2021 3:59:21 PM GMT] org.broadinstitute.hellbender.tools.walkers.variantutils.SelectVariants done. Elapsed time: 0.02 minutes.\n",
      "Runtime.totalMemory()=230686720\n",
      "Using GATK jar /gatk/gatk-package-4.1.8.1-local.jar\n",
      "Running:\n",
      "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xmx4g -jar /gatk/gatk-package-4.1.8.1-local.jar SelectVariants -O /SVAvuq/select_variants.vcf.gz -R /var/lib/cwl/stg69f07560-4efb-4317-8ddf-ea95741944fe/hla_and_brca_genes.fa --variant /var/lib/cwl/stg76080261-9780-4b81-978b-15092c68fc25/hla_and_brca_genes_omni.vcf.gz -L /var/lib/cwl/stg9ef62f94-5c20-45c9-8137-3b6fbf62d3c6/hla_and_brca_genes_target.interval_list\n",
      "\u001b[1;30mINFO\u001b[0m [job select_variants] Max memory used: 873MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job select_variants] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step select_variants] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] starting step verify_bam_id\n",
      "\u001b[1;30mINFO\u001b[0m [step verify_bam_id] start\n",
      "\u001b[1;30mWARNING\u001b[0m \u001b[33m[job verify_bam_id] Skipping Docker software container '--memory' limit despite presence of ResourceRequirement with ramMin and/or ramMax setting. Consider running with --strict-memory-limit for increased portability assurance.\u001b[0m\n",
      "\u001b[1;30mINFO\u001b[0m [job verify_bam_id] /private/tmp/docker_tmpfan7kx7o$ docker \\\n",
      "    run \\\n",
      "    -i \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpfan7kx7o,target=/SVAvuq \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpzukvw66i,target=/tmp \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp9wk_j5p_/final.bam,target=/var/lib/cwl/stg03fa164d-7297-450d-ac8b-f2b942bead4a/final.bam,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp7ahhx9xi/final.bam.bai,target=/var/lib/cwl/stg03fa164d-7297-450d-ac8b-f2b942bead4a/final.bam.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmp7ahhx9xi/final.bai,target=/var/lib/cwl/stg03fa164d-7297-450d-ac8b-f2b942bead4a/final.bai,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpvufs9vcj/select_variants.vcf.gz,target=/var/lib/cwl/stg6bad9e54-391d-4848-8385-e4d9405f3a3f/select_variants.vcf.gz,readonly \\\n",
      "    --mount=type=bind,source=/private/tmp/docker_tmpvufs9vcj/select_variants.vcf.gz.tbi,target=/var/lib/cwl/stg6bad9e54-391d-4848-8385-e4d9405f3a3f/select_variants.vcf.gz.tbi,readonly \\\n",
      "    --workdir=/SVAvuq \\\n",
      "    --read-only=true \\\n",
      "    --user=503:20 \\\n",
      "    --rm \\\n",
      "    --env=TMPDIR=/tmp \\\n",
      "    --env=HOME=/SVAvuq \\\n",
      "    --cidfile=/private/tmp/docker_tmpun5qybou/20211018105922-298465.cid \\\n",
      "    mgibio/verify_bam_id-cwl:1.1.3 \\\n",
      "    /usr/local/bin/verifyBamID \\\n",
      "    --out \\\n",
      "    /SVAvuq/final.VerifyBamId \\\n",
      "    --bam \\\n",
      "    /var/lib/cwl/stg03fa164d-7297-450d-ac8b-f2b942bead4a/final.bam \\\n",
      "    --vcf \\\n",
      "    /var/lib/cwl/stg6bad9e54-391d-4848-8385-e4d9405f3a3f/select_variants.vcf.gz\n",
      "\n",
      "Available Options\n",
      "                             Input Files : --vcf [/var/lib/cwl/stg6bad9e54-391d-4848-8385-e4d9405f3a3f/select_variants.vcf.gz],\n",
      "                                           --bam [/var/lib/cwl/stg03fa164d-7297-450d-ac8b-f2b942bead4a/final.bam],\n",
      "                                           --bai [], --subset [], --smID []\n",
      "                    VCF analysis options : --genoError [1.0e-03],\n",
      "                                           --minAF [0.01],\n",
      "                                           --minCallRate [0.50]\n",
      "   Individuals to compare with chip data : --site, --self, --best\n",
      "          Chip-free optimization options : --free-none, --free-mix [ON],\n",
      "                                           --free-refBias, --free-full\n",
      "          With-chip optimization options : --chip-none, --chip-mix [ON],\n",
      "                                           --chip-refBias, --chip-full\n",
      "                    BAM analysis options : --ignoreRG, --ignoreOverlapPair,\n",
      "                                           --noEOF, --precise, --minMapQ [10],\n",
      "                                           --maxDepth [20], --minQ [13],\n",
      "                                           --maxQ [40], --grid [0.05]\n",
      "                 Modeling Reference Bias : --refRef [1.00], --refHet [0.50],\n",
      "                                           --refAlt [0.00]\n",
      "                          Output options : --out [/SVAvuq/final.VerifyBamId],\n",
      "                                           --verbose\n",
      "                               PhoneHome : --noPhoneHome,\n",
      "                                           --phoneHomeThinning [50]\n",
      "\n",
      "verifyBamID 1.1.3 -- verify identity and purity of sequence data\n",
      "(c) 2010-2014 Hyun Min Kang, Goo Jun, and Goncalo Abecasis\n",
      "\n",
      "\n",
      "\u0007WARNING - \n",
      "--self option was autotomatically turned on by default. Specify --best option if you wanted to check across all possible samples in the VCF\n",
      "\u001b[1;30mINFO\u001b[0m [job verify_bam_id] Max memory used: 0MiB\n",
      "\u001b[1;30mINFO\u001b[0m [job verify_bam_id] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step verify_bam_id] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow qc] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [step qc] completed success\n",
      "\u001b[1;30mINFO\u001b[0m [workflow ] completed success\n",
      "{\n",
      "    \"alignment_summary_metrics\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.AlignmentSummaryMetrics.txt\",\n",
      "        \"basename\": \"final.AlignmentSummaryMetrics.txt\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$f8921c5abddf35c200f0c007c0df2caf072adc82\",\n",
      "        \"size\": 3867,\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.AlignmentSummaryMetrics.txt\"\n",
      "    },\n",
      "    \"bam\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.bam\",\n",
      "        \"basename\": \"final.bam\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$99ac10712918de4442bf117c82604498d0588139\",\n",
      "        \"size\": 940416,\n",
      "        \"secondaryFiles\": [\n",
      "            {\n",
      "                \"basename\": \"final.bam.bai\",\n",
      "                \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.bam.bai\",\n",
      "                \"class\": \"File\",\n",
      "                \"checksum\": \"sha1$56300824919a6b7fc73950872e5a2476d2c035fa\",\n",
      "                \"size\": 36864,\n",
      "                \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.bam.bai\"\n",
      "            },\n",
      "            {\n",
      "                \"basename\": \"final.bai\",\n",
      "                \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.bai\",\n",
      "                \"class\": \"File\",\n",
      "                \"checksum\": \"sha1$56300824919a6b7fc73950872e5a2476d2c035fa\",\n",
      "                \"size\": 36864,\n",
      "                \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.bai\"\n",
      "            }\n",
      "        ],\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.bam\"\n",
      "    },\n",
      "    \"flagstats\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.bam.flagstat\",\n",
      "        \"basename\": \"final.bam.flagstat\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$e8e9abd47d957b6cf0d133f6410689d8601fd644\",\n",
      "        \"size\": 403,\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.bam.flagstat\"\n",
      "    },\n",
      "    \"hs_metrics\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.roi-HsMetrics.txt\",\n",
      "        \"basename\": \"final.roi-HsMetrics.txt\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$a4e6416a0247df8e399af419cec54b88cc556515\",\n",
      "        \"size\": 4319,\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.roi-HsMetrics.txt\"\n",
      "    },\n",
      "    \"insert_size_histogram\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.InsertSizeHistogram.pdf\",\n",
      "        \"basename\": \"final.InsertSizeHistogram.pdf\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$202a0760cd52ee1ee802aff7230ac2aa3f126038\",\n",
      "        \"size\": 15424,\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.InsertSizeHistogram.pdf\"\n",
      "    },\n",
      "    \"insert_size_metrics\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.InsertSizeMetrics.txt\",\n",
      "        \"basename\": \"final.InsertSizeMetrics.txt\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$9d0babd024382f37dc075caf4284d5893e99449d\",\n",
      "        \"size\": 5049,\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.InsertSizeMetrics.txt\"\n",
      "    },\n",
      "    \"mark_duplicates_metrics\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.NameSorted.mark_dups_metrics.txt\",\n",
      "        \"basename\": \"final.NameSorted.mark_dups_metrics.txt\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$070c2bc0051b9def1ef681ec585712150520e4ae\",\n",
      "        \"size\": 2881,\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.NameSorted.mark_dups_metrics.txt\"\n",
      "    },\n",
      "    \"per_base_coverage_metrics\": [\n",
      "        {\n",
      "            \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.base-clinvar-PerBaseCoverage.txt\",\n",
      "            \"basename\": \"final.base-clinvar-PerBaseCoverage.txt\",\n",
      "            \"class\": \"File\",\n",
      "            \"checksum\": \"sha1$82f3c64abc78accdce331d680e6379be89513099\",\n",
      "            \"size\": 267090,\n",
      "            \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.base-clinvar-PerBaseCoverage.txt\"\n",
      "        }\n",
      "    ],\n",
      "    \"per_base_hs_metrics\": [\n",
      "        {\n",
      "            \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.base-clinvar-HsMetrics.txt\",\n",
      "            \"basename\": \"final.base-clinvar-HsMetrics.txt\",\n",
      "            \"class\": \"File\",\n",
      "            \"checksum\": \"sha1$84ff1b2071ee70d1b44ecbf8d2673eaadea31396\",\n",
      "            \"size\": 4383,\n",
      "            \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.base-clinvar-HsMetrics.txt\"\n",
      "        }\n",
      "    ],\n",
      "    \"per_target_coverage_metrics\": [\n",
      "        {\n",
      "            \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.target-acmg_genes-PerTargetCoverage.txt\",\n",
      "            \"basename\": \"final.target-acmg_genes-PerTargetCoverage.txt\",\n",
      "            \"class\": \"File\",\n",
      "            \"checksum\": \"sha1$f3a311c613e20a416ab08591244dc19661916946\",\n",
      "            \"size\": 5112,\n",
      "            \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.target-acmg_genes-PerTargetCoverage.txt\"\n",
      "        }\n",
      "    ],\n",
      "    \"per_target_hs_metrics\": [\n",
      "        {\n",
      "            \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.target-acmg_genes-HsMetrics.txt\",\n",
      "            \"basename\": \"final.target-acmg_genes-HsMetrics.txt\",\n",
      "            \"class\": \"File\",\n",
      "            \"checksum\": \"sha1$833ab59fd0a1c4d032ea97713be2648c5a112d5e\",\n",
      "            \"size\": 4397,\n",
      "            \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.target-acmg_genes-HsMetrics.txt\"\n",
      "        }\n",
      "    ],\n",
      "    \"summary_hs_metrics\": [],\n",
      "    \"verify_bam_id_depth\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.VerifyBamId.depthSM\",\n",
      "        \"basename\": \"final.VerifyBamId.depthSM\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$4121abf4f28a3b184b38098d89502f4233106fee\",\n",
      "        \"size\": 544,\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.VerifyBamId.depthSM\"\n",
      "    },\n",
      "    \"verify_bam_id_metrics\": {\n",
      "        \"location\": \"file:///Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.VerifyBamId.selfSM\",\n",
      "        \"basename\": \"final.VerifyBamId.selfSM\",\n",
      "        \"class\": \"File\",\n",
      "        \"checksum\": \"sha1$1de97523d5e30f6248cd743bf2bd0d2882177052\",\n",
      "        \"size\": 228,\n",
      "        \"path\": \"/Users/jwalker/git/bfx_workshop/lectures/week_05/tumor/final.VerifyBamId.selfSM\"\n",
      "    }\n",
      "}\n",
      "\u001b[1;30mINFO\u001b[0m Final process status is success\n"
     ]
    }
   ],
   "source": [
    "!cwltool --outdir tumor analysis-workflows/definitions/pipelines/alignment_exome.cwl tumor_inputs.yaml"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c97787e9",
   "metadata": {},
   "source": [
    "Load the tumor/final.bam and normal/final.bam in IGV: ex. chr17:7,675,089"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
